---
title: "SAR11_October_2024"
author: "Sarah Tucker"
date: "2024-10-24"
output: html_document
---

Build a set of high quality genomes
Take all the isolate genomes from Pelagibacteraceae published in Freel et al., in prep and those that are previously published and select only those that are high quality. Use checkM to make sure that you have the most highly complete genomes. We are using a threshold of >95% completion, <5 contamination, and fewer than 50 contigs.


```{python}


cd /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/final_genomes
mkdir updated
for f in *.fa; 
do 
base=$(basename "$f")
prefix=$(echo "$base" | sed s/".fa$"/""/)
sed 's/^>c_*./>'$prefix-'/' $f > updated/${prefix}.fa 
done

cd updated/ 

for files in *.fa 
do 
cat ${files} >> SAR11_April2024_bycontig.fa 
done


```



Expanded diversity observed within new SAR11 isolates
Let's first build a tree with outgroups, SAR11 isolates from other families, and all the high quality genomes from Pelagibacteraceae to see where they fall 
```{bash}
#copy over hmm for alphaproteobacteria specifically 
cd /tank/tucker_data/SAR11_2024/May_2024
cp -r Alphaproteobacterial_SCGs /tank/tucker_data/SAR11_2024/Oct_2024/

#Note isolates copies over from desktop final_genomes folder on 10/24/2024

cd /tank/tucker_data/SAR11_2024/Oct_2024/tree_building/final_genomes
for files in *.fa 
do 
cat ${files} >> SAR11_Oct2024_isolatestree_outgroup.fa 
done

anvi-gen-contigs-database -f SAR11_Oct2024_isolatestree_outgroup.fa -o SAR11_Oct2024_isolatestree-CONTIGS.db -n "SAR11 Isolates Tree" --num-threads 35

anvi-run-hmms -c SAR11_Oct2024_isolatestree-CONTIGS.db -H /tank/tucker_data/SAR11_2024/Oct_2024/tree_builiding/Alphaproteobacterial_SCGs/ --num-threads 40

bowtie2-build SAR11_Oct2024_isolatestree_outgroup.fa SAR11_Oct2024_isolatestree --threads 5


for sample in `awk '{print $1}' samples_test.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x SAR11_Oct2024_isolatestree\
            -1 /tank/tucker_data/01_QC/$sample-QUALITY_PASSED_R1.fastq.gz \
            -2 /tank/tucker_data/01_QC/$sample-QUALITY_PASSED_R2.fastq.gz \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

for sample in `awk '{print $1}' samples_test.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi
    anvi-profile -c SAR11_Oct2024_isolatestree-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --num-threads 35 \
                 -o $sample
done

anvi-merge */PROFILE.db -o SAR11_OCt2024_isolatestree-MERGED -c SAR11_Oct2024_isolatestree-CONTIGS.db


for split_name in `sqlite3 SAR11_Oct2024_isolatestree-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > SAR11_Oct2024_isolatestree-GENOME-COLLECTION.txt

anvi-import-collection SAR11_Oct2024_isolatestree-GENOME-COLLECTION.txt -c SAR11_Oct2024_isolatestree-CONTIGS.db -p SAR11_OCt2024_isolatestree-MERGED/PROFILE.db -C Genomes



anvi-get-sequences-for-hmm-hits -c SAR11_Oct2024_isolatestree-CONTIGS.db \
                                -p SAR11_OCt2024_isolatestree-MERGED/PROFILE.db\
                                -C Genomes \
                                --hmm-source Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --get-aa-sequences \
                                --concatenate-genes \
                                --return-best-hit \
                                -o Phylogeny_Oct2024/concatenated-Oct2024-alphaproteobacteria165_out.fa

####come back to this 
trimal -in Phylogeny_Oct2024/concatenated-Oct2024-alphaproteobacteria165_out.fa \
       -out Phylogeny_Oct2024/concatenated-Oct2024-alphaproteobacteria165_out-clean.fa \
       -gt 0.50


##trying this one on ROSA
clusterize "iqtree -s concatenated-Oct2024-alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 80 -B 1000" -n 80 -j Nov_tree -t 14-0:00:00

```



####the main contig db
```{r}
anvi-gen-contigs-database -f SAR11_Oct2024_isolates.fa -o SAR11_Oct2024_bycontig-CONTIGS.db -n "SAR11 Oct 2024" --num-threads 35


anvi-run-hmms -c SAR11_Oct2024_bycontig-CONTIGS.db --num-threads 35 

anvi-run-ncbi-cogs -c SAR11_Oct2024_bycontig-CONTIGS.db --num-threads 35 

anvi-run-kegg-kofams -c SAR11_Oct2024_bycontig-CONTIGS.db -T 35

anvi-run-hmms -c SAR11_Oct2024_bycontig-CONTIGS.db -H /tank/tucker_data/SAR11_2024/June_2024/Alphaproteobacterial_SCGs/ --num-threads 30

bowtie2-build SAR11_Oct2024_isolates.fa SAR11-Oct2024 --threads 35

while read sample r1 r2;
do
 if [ "$sample" == "sample" ]; then continue; fi
    bowtie2 --threads 35 \
            -x SAR11-Oct2024\
            -1 $r1 \
            -2 $r2 \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done < <(tail -n+2 samples_Aloha_KByT_mende_additional.txt)


for sample in `awk '{print $1}' samples_Aloha_KByT_mende_additional.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi
    anvi-profile -c SAR11_Oct2024_bycontig-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done

anvi-merge */PROFILE.db -o SAR11_Oct2024_bycontig-MERGED -c SAR11_Oct2024_bycontig-CONTIGS.db
anvi-run-hmms -c SAR11_Oct2024_bycontig-CONTIGS.db --num-threads 35 --also-scan-trnas --just-do-it


for split_name in `sqlite3 SAR11_Oct2024_bycontig-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > SAR11_Oct2024_bycontig-GENOME-COLLECTION.txt


anvi-import-collection SAR11_Oct2024_bycontig-GENOME-COLLECTION.txt -c SAR11_Oct2024_bycontig-CONTIGS.db -p SAR11_Oct2024_bycontig-MERGED/PROFILE.db -C Genomes

anvi-summarize -c SAR11_Oct2024_bycontig-CONTIGS.db -p SAR11_Oct2024_bycontig-MERGED/PROFILE.db -C Genomes --init-gene-coverages -o SAR11_Oct2024-by-contig-SUMMARY


anvi-gen-genomes-storage -i SAR11-Oct2024-bycontig_internal_genomes.txt -o SAR11_Oct2024-by-contig-PAN-GENOMES.db


anvi-pan-genome -g SAR11_Oct2024-by-contig-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 2 --project-name SAR11_Oct2024-by-contig-PAN --num-threads 35

anvi-meta-pan-genome -i SAR11-Oct2024-bycontig_internal_genomes.txt -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db --fraction-of-median-coverage 0.25

#add some information to the pangenome 
anvi-import-misc-data Genera_genome_info_layer_Oct2024.txt -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db --target-data-table layers

#Take a look  
anvi-display-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db


###Let's define the core pangenome and the singletons for Pelgaibacteraceae
##While in the pangenome, we need to bin Core Pelagibacteraceae and Singleton Pelagibacteraceae. For core use the search gene cluster function to look for Min number of genomes gene clusters occurs == 93 and for singleton use, Max number of genomes gene cluster occurs==1 and name the collection of the two bins Core_Pelagibacteraceae and Singleton_Pelagibacteraceae. There should be 784 gene clusters that are core to Pelagibacteraceae and 4201 gene clusters that are singletons
#total 8242 gene clusters

# summarize this to get output that we can bring into R 
anvi-summarize -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -C Core_Singleton --init-gene-coverages -o Core_Singleton_Pelagibacteraceae_SAR11_Oct2024

anvi-export-table SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db --table SAR11_Oct2024_gene_cluster_presence_absence

###have ANI data ready 
anvi-compute-genome-similarity --internal-genomes SAR11-Oct2024-bycontig_internal_genomes.txt --program fastANI --output-dir SAR11-by-contig-Oct2024-fastANI --num-threads 25 -p  SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db


```


###Characterisitics 
```{r}
characteristics=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/Core_Singleton_Pelagibacteraceae_SAR11_Oct2024/misc_data_layers/default.txt")
names(characteristics)
#layers= genome
##Add in genera data

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
tax=Taxonomy%>%dplyr::rename("layers"="samples")
characteristics=left_join(characteristics, tax, by="layers")

###by Genera
Count_Genomes=characteristics %>% group_by(Genus) %>% tally()
Count_Genomes

total_length=characteristics  %>% group_by(Genus)%>%summarise(total_length_mean=round(mean(total_length),0), total_length_sd=round(sd(total_length),0))
total_length

gc_content=characteristics  %>% group_by(Genus)%>%summarise(gc_content_mean=round(mean(gc_content),3), gc_content_sd=round(sd(gc_content),3))


###NOTE CHECKM is a better measure for this
percent_completion=characteristics  %>% group_by(Genus)%>%summarise(percent_completion_mean=round(mean(percent_completion),2), percent_completion_sd=round(sd(percent_completion),2))
percent_completion

###NOTE CHECKM is a better measure for this
percent_redundancy=characteristics  %>% group_by(Genus)%>%summarise(percent_redundancy_mean=round(mean(percent_redundancy),2), percent_redundancy_sd=round(sd(percent_redundancy),2))
percent_redundancy

num_genes=characteristics  %>% group_by(Genus)%>%summarise(num_genes_mean=round(mean(num_genes),0), num_genes_sd=round(sd(num_genes),0))
num_genes

num_gene_clusters=characteristics  %>% group_by(Genus)%>%summarise(num_gene_clusters_mean=round(mean(num_gene_clusters),0), num_gene_clusters_sd=round(sd(num_gene_clusters),0))
num_gene_clusters

singleton_gene_clusters=characteristics  %>% group_by(Genus)%>%summarise(singleton_gene_clusters_mean=round(mean(singleton_gene_clusters),0), singleton_gene_clusters_sd=round(sd(singleton_gene_clusters),0))
singleton_gene_clusters

library(dplyr)

all_metrics=cbind(Count_Genomes, total_length, gc_content, percent_completion, percent_redundancy, num_genes, num_gene_clusters, singleton_gene_clusters)
all_metrics
row.names(all_metrics) <- all_metrics$Genus
all_metrics[1] <- NULL
all_metrics <- all_metrics[!names(all_metrics) %in% c("Genus")]
write.csv(all_metrics, "output/by_contig_all_metrics_genus.csv")

###By Pelagibacteraceae
Count_Genomes=characteristics %>% tally()
Count_Genomes

total_length=characteristics%>%summarise(total_length_mean=round(mean(total_length),0), total_length_sd=round(sd(total_length),0))
total_length

gc_content=characteristics%>%summarise(gc_content_mean=round(mean(gc_content),3), gc_content_sd=round(sd(gc_content),3))

percent_completion=characteristics%>%summarise(percent_completion_mean=round(mean(percent_completion),2), percent_completion_sd=round(sd(percent_completion),2))
percent_completion

percent_redundancy=characteristics%>%summarise(percent_redundancy_mean=round(mean(percent_redundancy),2), percent_redundancy_sd=round(sd(percent_redundancy),2))
percent_redundancy

num_genes=characteristics%>%summarise(num_genes_mean=round(mean(num_genes),0), num_genes_sd=round(sd(num_genes),0))
num_genes

num_gene_clusters=characteristics%>%summarise(num_gene_clusters_mean=round(mean(num_gene_clusters),0), num_gene_clusters_sd=round(sd(num_gene_clusters),0))
num_gene_clusters

singleton_gene_clusters=characteristics %>%summarise(singleton_gene_clusters_mean=round(mean(singleton_gene_clusters),0), singleton_gene_clusters_sd=round(sd(singleton_gene_clusters),0))
singleton_gene_clusters

library(dplyr)
library(tidyverse)

Pelagibacteracae=cbind(Count_Genomes, total_length, gc_content, percent_completion, percent_redundancy, num_genes, num_gene_clusters, singleton_gene_clusters)

Pelagibacteracae
row.names(Pelagibacteracae) <- "Pelagibacteracae"

all_metrics2=rbind(Pelagibacteracae, all_metrics)

write.csv(all_metrics2, "output/all_metrics_by_contig_Pelagibacteracae.csv")
```



#bring in checkm info with other information from anvio to create Supp File 1a 
```{r}
characteristics=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_summary.txt")
names(characteristics)
characteristics=characteristics%>%dplyr::rename("layers"="bins")
characteristics=characteristics%>%dplyr::select(!c("percent_redundancy","percent_completion"))
#layers= genome
##Add in genera data

##note this could be updated to include the location and accession info
Taxonomy=read.delim("Location_accession_Genera_genome_info_layer_Oct2024.txt")
tax=Taxonomy%>%dplyr::rename("layers"="samples")
characteristics=left_join(tax,characteristics,  by="layers")

checkm=read.delim("checkm_final_genomes_organized.txt")
checkm=checkm%>%dplyr::select("layers", "Completeness","Contamination")

Sup_file_1=left_join(characteristics,checkm,  by="layers")

write.csv(Sup_file_1, "output/Sup_file_ia.csv")

```





#########################Figure 1 tree building
############################################################
###http://bioconductor.org/packages/release/bioc/vignettes/ggtreeExtra/inst/doc/ggtreeExtra.html

###creating a schematic tree with cut branches & learning to root
### to get bootstrap support and for making main figure and add on for supplemental 
###Figure 1b
```{r}
library(seecolor)


source_location=read.csv("source_location_genomes_october.csv")
source_location=source_location%>%dplyr::select("bins", "Source.Location2","Genus",)


source_location3 <- source_location
rownames(source_location3) <-source_location[,1]
source_location2 <- as.data.frame(source_location3[,-1])


colors_tog=c("Ia.1.I" ="#808000", "Ia.3.I" ="#eee8aa" , "Ia.3.II"  = "#87ceeb",  "Ia.3.III"="#adff2f"  ,  "Ia.3.IV"="#800000", "Ia.3.V"="#ba55d3","Ia.3.VI"="#0000cd","Ia.4.II"="#ff8c00" , "Ia.4.RS39"="#ffd700", "Ib.1"="#dda0dd", "Ia.4.1483"="#F91EAE","previous" ="#9FE2BF", "new" ="#4682B4" , "other_locations"  = "#ccdeea")



colorsgenus<- c("Ia.1.I" ="#808000", "Ia.3.I" ="#eee8aa" , "Ia.3.II"  = "#87ceeb",  "Ia.3.III"="#adff2f"  ,  "Ia.3.IV"="#800000", "Ia.3.V"="#ba55d3","Ia.3.VI"="#0000cd","Ia.4.II"="#ff8c00" , "Ia.4.RS39"="#ffd700", "Ib.1"="#dda0dd", "Ia.4.1483"="#ff1493") 
print_color(colorsgenus)

colorssource<- c("previous" ="#9FE2BF", "new" ="#4682B4" , "other_locations"  = "#ccdeea") 
print_color(colorssource)

packageVersion("pheatmap")
library(fpc)
library(treeio)
library(ape)
library(ggplot2)
library(ggtree)
library(phytools)

#note treefile from Iqtree is a newick, but if make edits in Figtree first you will lose how data is stored so lets do all the steps in R

tr <- treeio::read.newick("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/concatenated-Oct2024-alphaproteobacteria165_out-clean.fa.treefile", node.label='label')

#ggtree(tr) + geom_nodelab(geom='label', aes(label=label, subset=support > 80))

##see the labels
ggtree(tr) + geom_text(aes(label=node), hjust=-.3)+geom_tiplab(size=2,align=TRUE,family="arial")
ggsave("output/test.svg")
##re-root
#https://bioconnector.github.io/workshops/r-ggtree.html

#122
tree2 <- phytools::reroot(tr, node = 122, edgelabel = TRUE)

p <- ggtree(tree2, right=TRUE) +   geom_tiplab(size=2,align=TRUE,family="arial")+ geom_nodelab(geom='label', aes(label=label))
p

ggtree(tree2) + geom_text(aes(label=node), hjust=-.3)+geom_tiplab(size=2,align=TRUE,family="arial")


###alll but pelagibacteraceae
toremove=c("HIMB1517","HIMB1565","HIMB58","HIMB114","IMCC9063","LSUCC0530", "LSUCC0261","LSUCC0664","LSUCC0723","Thalassospira_profundimaris_WP0211", "Thalassospira_sp_A40_3")


#tree3<-drop.tip(tree2,tree2$tip.label[grep("HIMB1517","HIMB1565","HIMB58","HIMB114","IMCC9063","LSUCC0530","HTCC1002","HTCC1013","HTCC1016","HTCC1040","HTCC1062","NP1","HTCC9565","HTCC7211","HTCC7214","HTCC7217",tree2$tip.label)])
tree3<-drop.tip(tree2,tree2$tip.label[grep(toremove,tree2$tip.label)])
tree3=drop.tip(tree2, toremove, trim.internal = TRUE)
test_tree=ggtree(tree3)+   geom_tiplab(size=2,align=TRUE)+ geom_nodelab(size=0)

test_tree
write.tree(tree3, "/Users/sarahtucker/Documents/KByt/Diel_tRNA/SAR11_Pangraph/SAR11_Structural_Pangenome/Pelagibacteraceae_tree_dropped_othersubclades_outgroup.newick") 
#class(tree2$node.label)

#extract tip order 
x <- paste("Taxa order:", 
        paste0(get_taxa_name(test_tree), collapse=', '))
write.csv(x, "output/order_tree.csv")


###Figure 1b
kk=gheatmap(test_tree, source_location2, offset=0.3, width=1, font.size=2, 
        colnames = FALSE) +scale_fill_manual(values= colors_tog )+ theme(legend.position='right')+ geom_tiplab(size=2,align=TRUE)+geom_nodepoint(aes(color=as.numeric(label) > 90))
kk
ggsave("output/figure_1b_draft.svg")
#geom_nodelab(geom='label', aes(label=label))


?gheatmap()
??geom_nodepoint()

#geom_nodelab(geom='label', aes(label=label))+geom_label2(aes(label=label, subset = !is.na(as.numeric(label)) & as.numeric(label) > 80))



```









##what about ANI

```{r}
library(reshape2)
ani=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11-by-contig-Oct2024-fastANI/fastANI_ani.txt")
ani[ani == 1.000000] <- NA
long_ani=melt(ani)
names(long_ani)


ani_Pelagibacter_min=long_ani%>%summarise(ani_min=round(min(value, na.rm = TRUE),3), ani_mean=round(mean(value, na.rm = TRUE),3))

###Min ANI by Genus
Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)
Genus_Tax=Taxonomy%>%dplyr::select("Genus", "samples")%>%dplyr::rename( "key"= "samples")
genus_ani=left_join(ani, Genus_Tax)
long_ani2=melt(genus_ani)

Genus_Tax=Taxonomy%>%dplyr::select("Genus", "samples")%>%dplyr::rename("variable"= "samples", "Genus_variable"="Genus")

long_ani3=left_join(long_ani2, Genus_Tax)


min_ani_genera=long_ani3%>%dplyr::filter(Genus_variable==Genus)%>%dplyr::group_by(Genus_variable)%>%dplyr::summarise(ani_min=round(min(value, na.rm = TRUE)*100,1), ani_mean=round(mean(value, na.rm = TRUE)*100,1),ani_sd=round(sd(value, na.rm = TRUE)*100,1), n=n())



min_ani_genera=long_ani3%>%dplyr::summarise(ani_min=round(min(value, na.rm = TRUE)*100,1), ani_mean=round(mean(value, na.rm = TRUE)*100,1),ani_sd=round(sd(value, na.rm = TRUE)*100,1), n=n())

###average of averages
avg_min_ani=min_ani_genera%>%dplyr::filter(Genus_variable!="Ia.4.1483")%>%dplyr::summarise(mean_mean=round(mean(ani_mean, na.rm = TRUE),3), sd_mean=round(sd(ani_mean, na.rm = TRUE),3))

###ANI input comes from metagenomic section 
library(reshape2)
library(tidyverse)
ani=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11-by-contig-Oct2024-fastANI/fastANI_ani.txt", row.names = 1)
ani[ani >= 0.999] <- NA

ani2=ani
ani2$genome_name=rownames(ani)
long_ani=melt(ani2)

ani_min=long_ani%>%summarise(ani_min=round(min(value, na.rm = TRUE),3), ani_mean=round(mean(value, na.rm = TRUE),3), ani_max=round(max(value, na.rm = TRUE),3))

###Min ANI by MinorClade
Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Genus_Tax=Taxonomy%>%dplyr::select("Genus", "samples")%>%dplyr::rename("variable"= "samples", "Genus_variable"="Genus")
long_ani2=left_join(long_ani, Genus_Tax)

names(long_ani2)

Genus_Tax2=Taxonomy%>%dplyr::select("Genus", "samples")%>%dplyr::rename(  "genome_name"="samples")

Genus_ani=left_join(long_ani2, Genus_Tax2)
view(Genus_ani)
Genus_ani2=Genus_ani
Genus_ani2$per=Genus_ani2$value*100

min_ani_genera=Genus_ani%>%dplyr::filter(Genus_variable==Genus)%>%dplyr::group_by(Genus_variable)%>%dplyr::summarise(ani_min=round(min(value, na.rm = TRUE),3), ani_mean=round(mean(value, na.rm = TRUE),3))
names(Genus_ani)

##note the order comes from out/order_tree.csv

order_genome=factor(Genus_ani2$genome_name,level=c("RS40","HIMB2305","HTCC9565","NP1","HTCC1040","HTCC1016","HTCC1002","HTCC1062","HTCC1013","HIMB1527","HIMB1321","HIMB1549","HIMB1505","HIMB4","HIMB1412","HIMB1420","HIMB1427","HIMB1564","HIMB1520","HIMB5","HIMB2187","HIMB2226","HIMB1436","RS39","HIMB1483","HIMB2204","HIMB2215","HIMB2200","HIMB1402","HIMB1437","HIMB1863","HIMB2104","HTCC8051","HTCC9022","HTCC7214","HTCC7211","HTCC7217","FZCC0015","HIMB83","HIMB1456","HIMB2201","HIMB1444","HIMB2250","HIMB1409","HIMB1413","HIMB1709","HIMB1623","HIMB1765","HIMB1485","HIMB1715","HIMB1488","HIMB1506","HIMB2211","HIMB1782","HIMB1701","HIMB1509","HIMB1593","HIMB1685","HIMB1559","HIMB1597","HIMB1587","HIMB1495","HIMB1494","HIMB1490","HIMB1491","HIMB122","HIMB1631","HIMB1577","HIMB140","HIMB1611","HIMB1758","HIMB1430","HIMB1746","HIMB1542","HIMB1695","HIMB1573","HIMB1513","HIMB1521","HIMB1507","HIMB1518","HIMB1770","HIMB1493","HIMB1723","HIMB1662","HIMB1702","HIMB1556","HIMB1641","HIMB1552","HIMB1636","HIMB1526","HIMB1748","HIMB1710"))


order_variable=factor(Genus_ani2$variable, level=c("RS40","HIMB2305","HTCC9565","NP1","HTCC1040","HTCC1016","HTCC1002","HTCC1062","HTCC1013","HIMB1527","HIMB1321","HIMB1549","HIMB1505","HIMB4","HIMB1412","HIMB1420","HIMB1427","HIMB1564","HIMB1520","HIMB5","HIMB2187","HIMB2226","HIMB1436","RS39","HIMB1483","HIMB2204","HIMB2215","HIMB2200","HIMB1402","HIMB1437","HIMB1863","HIMB2104","HTCC8051","HTCC9022","HTCC7214","HTCC7211","HTCC7217","FZCC0015","HIMB83","HIMB1456","HIMB2201","HIMB1444","HIMB2250","HIMB1409","HIMB1413","HIMB1709","HIMB1623","HIMB1765","HIMB1485","HIMB1715","HIMB1488","HIMB1506","HIMB2211","HIMB1782","HIMB1701","HIMB1509","HIMB1593","HIMB1685","HIMB1559","HIMB1597","HIMB1587","HIMB1495","HIMB1494","HIMB1490","HIMB1491","HIMB122","HIMB1631","HIMB1577","HIMB140","HIMB1611","HIMB1758","HIMB1430","HIMB1746","HIMB1542","HIMB1695","HIMB1573","HIMB1513","HIMB1521","HIMB1507","HIMB1518","HIMB1770","HIMB1493","HIMB1723","HIMB1662","HIMB1702","HIMB1556","HIMB1641","HIMB1552","HIMB1636","HIMB1526","HIMB1748","HIMB1710"))


Genus_ani3=Genus_ani2
Genus_ani3[is.na(Genus_ani3)] <- 100
class(Genus_ani2$value)

pdf("output/ani_sar11.pdf", height=12.5, width=16)
geom.text.size.ani=2
ani_all=ggplot(Genus_ani3, aes(x = order_genome, y = order_variable, fill = per)) +
    geom_tile(aes(fill = per)) + geom_text(aes(label = round(per, 0)), size=geom.text.size.ani) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) +
  scale_fill_stepsn(colors =c("white","#999999","#6666FF","#FFFF66", "#C70039"),
                    breaks = c(1,75,84, 90,97, 99, 100))+ labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=7)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=7),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_blank(), legend.text=element_text(size=7), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")

ani_all
dev.off()



kk+ani_all
ggsave('output/ANI_tree.svg',width=21, height=14)
```

Now that we have built the Pelagibacteraceae pangenome, we can understand what is unique and shared among our different SAR11 genera. We already know that there is a large shared core among all 92 Pelagibacteraceae and a LOT of singletons (>4,000), but does a given genus of SAR11 maintain a unique set of genes not shared with other SAR11 genera? Do some sets of genera share more than others?

To answer these questions, I brought the summary table into R and characterized what was "core" and "singletons" to each genus.



```{r}

library("tidyverse")
library(reshape2)
library(tidyverse)

getwd()

Gene_Cluster_Summary=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/Core_Singleton_Pelagibacteraceae_SAR11_Oct2024/SAR11_Oct2024-by-contig-PAN_gene_clusters_summary.txt")
length(unique(Gene_Cluster_Summary$gene_cluster_id))
GC_add_on=Gene_Cluster_Summary%>%dplyr::select("gene_cluster_id", "KOfam", "KOfam_ACC" ,  "COG20_FUNCTION_ACC", "COG20_FUNCTION" )%>%distinct(gene_cluster_id, .keep_all=TRUE)



names(Gene_Cluster_Summary)
Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)
count_genera=Taxonomy%>%group_by(Genus)%>%tally()
Taxonomy=Taxonomy%>%rename_all(recode,"samples"="genome_name")
names(Taxonomy)
Tax = Taxonomy[,!(names(Taxonomy) %in% c("Numerical", "Species", "Type"))]

Gene_Cluster_Summary_Tax=left_join(Gene_Cluster_Summary, Tax, "genome_name")
Core_Pelagibacteraceae=subset(Gene_Cluster_Summary_Tax, num_genomes_gene_cluster_has_hits==92)
core=distinct(Core_Pelagibacteraceae,gene_cluster_id, .keep_all = TRUE)
length(unique(core$gene_cluster_id))
core=core%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
core$genus_specific="NA"
core$Environment="NA"
core$Pangenome="Core"
core$Enviro_Specific="NA"
write.csv(core, "output/core_pelagi_conventional.csv")


Singleton_Pelagibacteraceae=subset(Gene_Cluster_Summary_Tax, num_genomes_gene_cluster_has_hits==1)
sing=distinct(Singleton_Pelagibacteraceae,gene_cluster_id, .keep_all = TRUE)
length(unique(sing$gene_cluster_id))
sing=sing%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
sing$genus_specific="NA"
sing$Environment="NA"
sing$Pangenome="Singleton"
sing$Enviro_Specific="NA"
write.csv(sing, "output/sing_pelagi_conventional.csv")

###lets define the core and singletons at the genus level 
Ia.4.RS39=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.4.RS39")
ngenomes=length(unique(Ia.4.RS39$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.4.RS39$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.RS39%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether these gene clusters are single copy or not 
test$SCG ="single_copy"
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.RS39<- "present"
data$bin_Ia.4.RS39[data$sum==ngenomes] <- "Core"
data$bin_Ia.4.RS39[data$sum==1] <- "Singleton"
data$Ia.4.RS39_bin_SCG=paste(data$bin_Ia.4.RS39, data$SCG, sep="_")

gg=data%>%dplyr::count(Ia.4.RS39_bin_SCG)
to_compare_Ia.4.RS39=data

write.csv(data, "output/Ia.4.RS39_bin_SCG.csv")

Ia.4.RS39_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.4.RS39", "Ia.4.RS39_bin_SCG"))



###Ia.4.II, n=7
Ia.4.II=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.4.II")
ngenomes=length(unique(Ia.4.II$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.4.II$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.II%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
View(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.II<- "present"
data$bin_Ia.4.II[data$sum==ngenomes] <- "Core"
data$bin_Ia.4.II[data$sum==1] <- "Singleton"

data$Ia.4.II_bin_SCG=paste(data$bin_Ia.4.II, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.4.II_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.4.II_bin_SCG.csv")

names(data)

Ia.4.II_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.4.II", "Ia.4.II_bin_SCG"))


###Ia.1.I, n=7
Ia.1.I=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.1.I")
ngenomes=length(unique(Ia.1.I$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.1.I$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.1.I%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.1.I<- "present"
data$bin_Ia.1.I[data$sum==ngenomes] <- "Core"
data$bin_Ia.1.I[data$sum==1] <- "Singleton"

data$Ia.1.I_bin_SCG=paste(data$bin_Ia.1.I, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.1.I_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.1.I_bin_SCG.csv")
names(data)

Ia.1.I_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.1.I", "Ia.1.I_bin_SCG"))


###Ia.3.III, n=5
Ia.3.III=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.III")
ngenomes=length(unique(Ia.3.III$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.III$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.III%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.III<- "present"
data$bin_Ia.3.III[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.III[data$sum==1] <- "Singleton"

data$Ia.3.III_bin_SCG=paste(data$bin_Ia.3.III, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.III_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.III_bin_SCG.csv")

names(data)

Ia.3.III_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.III", "Ia.3.III_bin_SCG"))




###Ia.3.II, n=6
Ia.3.II=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.II")
ngenomes=length(unique(Ia.3.II$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.II$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.II%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.II<- "present"
data$bin_Ia.3.II[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.II[data$sum==1] <- "Singleton"

data$Ia.3.II_bin_SCG=paste(data$bin_Ia.3.II, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.II_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.II_bin_SCG.csv")
names(data)

Ia.3.II_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.II", "Ia.3.II_bin_SCG"))






###Ia.3.I, n=3
Ia.3.I=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.I")
ngenomes=length(unique(Ia.3.I$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.I$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.I%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)

###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.I<- "present"
data$bin_Ia.3.I[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.I[data$sum==1] <- "Singleton"

data$Ia.3.I_bin_SCG=paste(data$bin_Ia.3.I, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.I_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.I_bin_SCG.csv")
names(data)

Ia.3.I_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.I", "Ia.3.I_bin_SCG"))





###Ia.3.IV, n=2
Ia.3.IV=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.IV")
ngenomes=length(unique(Ia.3.IV$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.IV$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.IV%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.IV <- "present"
data$bin_Ia.3.IV[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.IV[data$sum==1] <- "Singleton"

data$Ia.3.IV_bin_SCG=paste(data$bin_Ia.3.IV, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.IV_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.IV_bin_SCG.csv")

names(data)

Ia.3.IV_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.IV", "Ia.3.IV_bin_SCG"))




###Ia.3.V, n=9
Ia.3.V=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.V")
ngenomes=length(unique(Ia.3.V$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.V$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.V%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.V="present"
data$bin_Ia.3.V[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.V[data$sum==1] <- "Singleton"

data$Ia.3.V_bin_SCG=paste(data$bin_Ia.3.V, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.V_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.V_bin_SCG.csv")
names(data)

Ia.3.V_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.V", "Ia.3.V_bin_SCG"))




###Ia.3.VI, n=47
Ia.3.VI=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.3.VI")
ngenomes=length(unique(Ia.3.VI$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.3.VI$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.VI%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
okay
View(okay)

###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.VI="present"
data$bin_Ia.3.VI[data$sum==ngenomes] <- "Core"
data$bin_Ia.3.VI[data$sum==1] <- "Singleton"

data$Ia.3.VI_bin_SCG=paste(data$bin_Ia.3.VI, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.VI_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.3.VI_bin_SCG.csv")
names(data)

Ia.3.VI_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.3.VI", "Ia.3.VI_bin_SCG"))




###Ib.1, n=2
Ib.1=subset(Gene_Cluster_Summary_Tax, Genus=="Ib.1")
ngenomes=length(unique(Ib.1$genome_name))
ngenomes
##total pangenome size
length(unique(Ib.1$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ib.1%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "gene_cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "gene_cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ib.1="present"
data$bin_Ib.1[data$sum==ngenomes] <- "Core"
data$bin_Ib.1[data$sum==1] <- "Singleton"

data$Ib.1_bin_SCG=paste(data$bin_Ib.1, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ib.1_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ib.1_bin_SCG.csv")

Ib.1_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ib.1", "Ib.1_bin_SCG"))


#####

###Ia.4.1483, n=1
Ia.4.1483=subset(Gene_Cluster_Summary_Tax, Genus=="Ia.4.1483")
ngenomes=length(unique(Ia.4.1483$genome_name))
ngenomes
##total pangenome size
length(unique(Ia.4.1483$gene_cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.1483%>%dcast(gene_cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))

test3=test

test3$SCG[test3$HIMB1483 > 1] <-  "multiple_copy"
###new way
data=test3
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.1483="present"
data$Ia.4.1483_bin_SCG=paste(data$bin_Ia.4.1483, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.4.1483_bin_SCG)%>%dplyr::summarise(n = n())
okay
write.csv(data, "output/Ia.4.1483_bin_SCG.csv")

Ia.4.1483_bin_info=data%>%dplyr::select(c("gene_cluster_id", "bin_Ia.4.1483", "Ia.4.1483_bin_SCG"))


```



##find genus specific GCs

```{r}

a=left_join(Gene_Cluster_Summary_Tax, Ib.1_bin_info, by="gene_cluster_id")
b=left_join(a, Ia.3.VI_bin_info, by="gene_cluster_id")
c=left_join(b, Ia.3.V_bin_info, by="gene_cluster_id")
d=left_join(c, Ia.3.IV_bin_info, by="gene_cluster_id")
e=left_join(d, Ia.3.I_bin_info, by="gene_cluster_id")
f=left_join(e, Ia.3.II_bin_info, by="gene_cluster_id")
g=left_join(f, Ia.3.III_bin_info, by="gene_cluster_id")
h=left_join(g, Ia.1.I_bin_info, by="gene_cluster_id")
i=left_join(h, Ia.4.II_bin_info, by="gene_cluster_id")
j=left_join(i, Ia.4.RS39_bin_info, by="gene_cluster_id")
pangenome_SAR11=left_join(j, Ia.4.1483_bin_info, by="gene_cluster_id")
dim(pangenome_SAR11)
length(unique(pangenome_SAR11$gene_cluster_id))

#write.csv(pangenome_SAR11, "output/by_contigs_all_bins_Oct_Gene_Cluster_Summary_Tax.csv")
#pangenome_SAR11=read.csv("output/by_contigs_all_bins_Oct_Gene_Cluster_Summary_Tax.csv")

names(pangenome_SAR11)
library(dplyr)
compare_cores=pangenome_SAR11%>%dplyr::select(c("gene_cluster_id", "bin_name","bin_Ib.1", "bin_Ia.3.VI", "bin_Ia.3.V", "bin_Ia.3.IV", "bin_Ia.3.I","bin_Ia.3.II",  "bin_Ia.3.II", "bin_Ia.3.III","bin_Ia.1.I", "bin_Ia.4.II", "bin_Ia.4.RS39", "bin_Ia.4.1483" ))

unique(compare_cores$bin_Ia.4.1483)
dim(compare_cores)
length(unique(compare_cores$gene_cluster_id))


compare_cores3=distinct(compare_cores, gene_cluster_id, .keep_all = TRUE)
length(unique(compare_cores3$gene_cluster_id))
compare_cores4=compare_cores3%>%dplyr::select("bin_name","bin_Ib.1", "bin_Ia.3.VI", "bin_Ia.3.V", "bin_Ia.3.IV", "bin_Ia.3.I","bin_Ia.3.II",  "bin_Ia.3.II", "bin_Ia.3.III","bin_Ia.1.I", "bin_Ia.4.II", "bin_Ia.4.RS39","bin_Ia.4.1483","gene_cluster_id")

compare_cores4[is.na(compare_cores4)] <- "absent"
names(compare_cores4)


test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "Core" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#95 unique Ib.1 genes
Ib.1_specific=test$gene_cluster_id
length(Ib.1_specific)


unique_Ib.1=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ib.1=distinct(unique_Ib.1,gene_cluster_id, .keep_all = TRUE)
names(unique_Ib.1)
unique_Ib.1=unique_Ib.1%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION,aa_sequence)
unique_Ib.1$genus_specific="unique_Ib.1"
#95
write.csv(unique_Ib.1, "output/unique_Ib.1.csv")


test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "Core" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent"& bin_Ia.4.1483== "absent" )
#1 unique Ia.3.VIgenes
Ia.3.VI_specific=test$gene_cluster_id
length(Ia.3.VI_specific)


unique_Ia.3.VI=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.VI=distinct(unique_Ia.3.VI,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.VI)
unique_Ia.3.VI=unique_Ia.3.VI%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.VI$genus_specific="unique_Ia.3.VI"
write.csv(unique_Ia.3.VI, "output/unique_Ia.3.VI.csv")

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "Core" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#8 unique Ia.3.V genes
Ia.3.V_specific=test$gene_cluster_id
length(Ia.3.V_specific)


unique_Ia.3.V=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.V=distinct(unique_Ia.3.V,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.V)
unique_Ia.3.V=unique_Ia.3.V%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.V$genus_specific="unique_Ia.3.V"
write.csv(unique_Ia.3.V, "output/unique_Ia.3.V.csv")


test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "Core" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#3 unique Ia.3.IV genes
Ia.3.IV_specific=test$gene_cluster_id
length(Ia.3.IV_specific)


unique_Ia.3.IV=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.IV=distinct(unique_Ia.3.IV,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.IV)
unique_Ia.3.IV=unique_Ia.3.IV%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.IV$genus_specific="unique_Ia.3.IV"
write.csv(unique_Ia.3.IV, "output/unique_Ia.3.IV.csv")




test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "Core" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#15 unique Ia.3.III genes
Ia.3.III_specific=test$gene_cluster_id
length(Ia.3.III_specific)


unique_Ia.3.III=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.III=distinct(unique_Ia.3.III,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.III)
unique_Ia.3.III=unique_Ia.3.III%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.III$genus_specific="unique_Ia.3.III"
write.csv(unique_Ia.3.III, "output/unique_Ia.3.III.csv")



test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "Core" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#16 unique Ia.3.II genes
Ia.3.II_specific=test$gene_cluster_id
length(Ia.3.II_specific)

unique_Ia.3.II=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.II=distinct(unique_Ia.3.II,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.II)
unique_Ia.3.II=unique_Ia.3.II%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.II$genus_specific="unique_Ia.3.II"
write.csv(unique_Ia.3.II, "output/unique_Ia.3.II.csv")



test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "Core" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#27 unique Ia.3.I genes
Ia.3.I_specific=test$gene_cluster_id
length(Ia.3.I_specific)


unique_Ia.3.I=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.3.I=distinct(unique_Ia.3.I,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.3.I)
unique_Ia.3.I=unique_Ia.3.I%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.3.I$genus_specific="unique_Ia.3.I"
write.csv(unique_Ia.3.I, "output/unique_Ia.3.I.csv")

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "Core" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
#18 unique Ia.1.I genes
Ia.1.I_specific=test$gene_cluster_id
length(Ia.1.I_specific)


unique_Ia.1.I=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.1.I=distinct(unique_Ia.1.I,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.1.I)
unique_Ia.1.I=unique_Ia.1.I%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.1.I$genus_specific="unique_Ia.1.I"
write.csv(unique_Ia.1.I, "output/unique_Ia.1.I.csv")

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "Core" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
#4 unique Ia.4.II genes
Ia.4.II_specific=test$gene_cluster_id
length(Ia.4.II_specific)

unique_Ia.4.II=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.4.II=distinct(unique_Ia.4.II,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.4.II)
unique_Ia.4.II=unique_Ia.4.II%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.4.II$genus_specific="unique_Ia.4.II"
write.csv(unique_Ia.4.II, "output/unique_Ia.4.II.csv")

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "Core" & bin_Ia.4.1483== "absent" )
Ia.4.RS39_specific=test$gene_cluster_id
#18 unique Ia.4.RS39 genes
length(Ia.4.RS39_specific)


unique_Ia.4.RS39=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.4.RS39=distinct(unique_Ia.4.RS39,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.4.RS39)
unique_Ia.4.RS39=unique_Ia.4.RS39%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.4.RS39$genus_specific="unique_Ia.4.RS39"
write.csv(unique_Ia.4.RS39, "output/unique_Ia.4.RS39.csv")



test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "present" )
Ia.4.1483_specific=test$gene_cluster_id
#18 unique Ia.4.1483 genes
length(Ia.4.1483_specific)


unique_Ia.4.1483=left_join(test, Gene_Cluster_Summary, by="gene_cluster_id")
unique_Ia.4.1483=distinct(unique_Ia.4.1483,gene_cluster_id, .keep_all = TRUE)
names(unique_Ia.4.1483)
unique_Ia.4.1483=unique_Ia.4.1483%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
unique_Ia.4.1483$genus_specific="unique_Ia.4.1483"
write.csv(unique_Ia.4.1483, "output/unique_Ia.4.1483.csv")

###Extended Data Table 3
genus_specific_GCs=rbind(unique_Ia.4.RS39, unique_Ia.4.II, unique_Ia.1.I, unique_Ia.3.I, unique_Ia.3.II, unique_Ia.3.III, unique_Ia.3.IV, unique_Ia.3.V, unique_Ia.3.VI, unique_Ib.1)

genus_specific_GCs$Environment="NA"

genus_specific_GCs$Environment="NA"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.VI"] <- "Coastal"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.II"] <- "Coastal"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.III"] <- "Coastal"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.V"] <- "Offshore"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.4.II"] <- "Offshore"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.4.1483"] <- "Offshore"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.4.RS39"] <- "Offshore"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ib.1"] <- "Offshore"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.I"] <- "Rare"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.3.IV"] <- "Rare"
genus_specific_GCs$Environment[genus_specific_GCs$genus_specific=="unique_Ia.1.I"] <- "Rare"
genus_specific_GCs$Pangenome="genus_specific"


genus_specific_GCs=genus_specific_GCs%>%unite(Enviro_Specific, c('Pangenome', 'Environment'), sep='_', remove = FALSE)

genus_specific_GCs=genus_specific_GCs%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence, genus_specific, Environment, Pangenome, Enviro_Specific)
write.csv(genus_specific_GCs, "output/genus_specific_GCs.csv")
info=genus_specific_GCs%>%group_by(genus_specific)%>%count()
```

##find GCs that are multilineage core

```{r}
strict_multilineage_core=compare_cores4%>%dplyr:: filter(bin_Ib.1 %in% c("absent", "Core") & bin_Ia.3.VI%in% c("absent", "Core") & bin_Ia.3.V%in% c("absent", "Core") & bin_Ia.3.IV%in% c("absent", "Core") & bin_Ia.3.III %in% c("absent", "Core") & bin_Ia.3.II %in% c("absent", "Core") & bin_Ia.3.I  %in% c("absent", "Core") & bin_Ia.1.I%in% c("absent", "Core") & bin_Ia.4.II%in% c("absent", "Core") & bin_Ia.4.RS39 %in% c("absent", "Core") &  bin_Ia.4.1483%in% c( "present","absent"))
view(strict_multilineage_core)



strict_multilineage_core=strict_multilineage_core%>% dplyr::filter(bin_name !="Core_Pelagibacteraceae")
strict_multilineage_core=strict_multilineage_core%>% dplyr::filter(bin_name !="Singleton_Pelagibacteraceae")
##here the genus specific are still included 
## multilineage is 388-206= 182

compare_cores8=strict_multilineage_core[ , !names(strict_multilineage_core) %in% c("bin_name", "gene_cluster_id")] 
compare_cores9<- data.frame(lapply(compare_cores8, function(x) {gsub("Core", "1", x)  }))
compare_cores10<- data.frame(lapply(compare_cores9, function(x) {gsub("absent", "0", x)  }))
compare_cores11<- data.frame(lapply(compare_cores10, function(x) {gsub("present", "1", x)  }))


##need to remove the genus-specific from this list of multiple-genus specific
listy=genus_specific_GCs$gene_cluster_id
strict_multilineage_core_not_genus=strict_multilineage_core%>% dplyr::filter(! gene_cluster_id %in% listy)


strict_multilineage_core_not_genus=left_join(strict_multilineage_core_not_genus, Gene_Cluster_Summary, by="gene_cluster_id")
strict_multilineage_core_not_genus=distinct(strict_multilineage_core_not_genus,gene_cluster_id, .keep_all = TRUE)
names(strict_multilineage_core_not_genus)
strict_multilineage_core_not_genus=strict_multilineage_core_not_genus%>%dplyr::select(bin_Ib.1, bin_Ia.3.I, bin_Ia.3.VI, bin_Ia.3.V, bin_Ia.3.IV, bin_Ia.3.III, bin_Ia.3.II, bin_Ia.4.1483, bin_Ia.4.II, bin_Ia.4.RS39, bin_Ia.1.I, gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)

write.csv(strict_multilineage_core_not_genus,"output/strict_multilineage_conventional_with_bins.csv")
df=strict_multilineage_core_not_genus
df$Enviro="NA"
df$bin_Ib.1_remake="NA"
df$bin_Ib.1_remake[df$bin_Ib.1=="Core"] <- "Ib.1"
df$bin_Ia.3.I_remake="NA"
df$bin_Ia.3.I_remake[df$bin_Ia.3.I=="Core"] <- "Ia.3.I"
df$bin_Ia.3.VI_remake="NA"
df$bin_Ia.3.VI_remake[df$bin_Ia.3.VI=="Core"] <- "Ia.3.VI"
df$bin_Ia.3.V_remake="NA"
df$bin_Ia.3.V_remake[df$bin_Ia.3.V=="Core"] <- "Ia.3.V"
df$bin_Ia.3.IV_remake="NA"
df$bin_Ia.3.IV_remake[df$bin_Ia.3.IV=="Core"] <- "Ia.3.IV"
df$bin_Ia.3.III_remake="NA"
df$bin_Ia.3.III_remake[df$bin_Ia.3.III=="Core"] <- "Ia.3.III"
df$bin_Ia.3.II_remake="NA"
df$bin_Ia.3.II_remake[df$bin_Ia.3.II=="Core"] <- "Ia.3.II"
df$bin_Ia.4.1483_remake="NA"
df$bin_Ia.4.1483_remake[df$bin_Ia.4.1483=="present"] <- "Ia.4.1483"
df$bin_Ia.4.II_remake="NA"
df$bin_Ia.4.II_remake[df$bin_Ia.4.II=="Core"] <- "Ia.4.II"
df$bin_Ia.4.RS39_remake="NA"
df$bin_Ia.4.RS39_remake[df$bin_Ia.4.RS39=="Core"] <- "Ia.4.RS39"
df$bin_Ia.1.I_remake="NA"
df$bin_Ia.1.I_remake[df$bin_Ia.1.I=="Core"] <- "Ia.1.I"

df2=df%>%dplyr::select(bin_Ib.1_remake, bin_Ia.3.I_remake, bin_Ia.3.VI_remake, bin_Ia.3.V_remake, bin_Ia.3.IV_remake, bin_Ia.3.III_remake, bin_Ia.3.II_remake, bin_Ia.4.1483_remake, bin_Ia.4.II_remake, bin_Ia.4.RS39_remake, bin_Ia.1.I_remake, gene_cluster_id)


# columns to paste together
cols <- c( "bin_Ib.1_remake", "bin_Ia.3.I_remake", "bin_Ia.3.VI_remake", "bin_Ia.3.V_remake", "bin_Ia.3.IV_remake", "bin_Ia.3.III_remake","bin_Ia.3.II_remake", "bin_Ia.4.1483_remake", "bin_Ia.4.II_remake", "bin_Ia.4.RS39_remake", "bin_Ia.1.I_remake" )

# create a new column `x` with the three columns collapsed together
df$genus_specific <- apply( df[ , cols ] , 1 , paste , collapse = "," )




df$bin_Ib.1_remake2="0"
df$bin_Ib.1_remake2[df$bin_Ib.1=="Core"] <- "1"
df$bin_Ia.3.I_remake2="0"
df$bin_Ia.3.I_remake2[df$bin_Ia.3.I=="Core"] <- "1"
df$bin_Ia.3.VI_remake2="0"
df$bin_Ia.3.VI_remake2[df$bin_Ia.3.VI=="Core"] <- "1"
df$bin_Ia.3.V_remake2="0"
df$bin_Ia.3.V_remake2[df$bin_Ia.3.V=="Core"] <- "1"
df$bin_Ia.3.IV_remake2="0"
df$bin_Ia.3.IV_remake2[df$bin_Ia.3.IV=="Core"] <- "1"
df$bin_Ia.3.III_remake2="0"
df$bin_Ia.3.III_remake2[df$bin_Ia.3.III=="Core"] <- "1"
df$bin_Ia.3.II_remake2="0"
df$bin_Ia.3.II_remake2[df$bin_Ia.3.II=="Core"] <- "1"
df$bin_Ia.4.1483_remake2="0"
df$bin_Ia.4.1483_remake2[df$bin_Ia.4.1483=="present"] <- "1"
df$bin_Ia.4.II_remake2="0"
df$bin_Ia.4.II_remake2[df$bin_Ia.4.II=="Core"] <- "1"
df$bin_Ia.4.RS39_remake2="0"
df$bin_Ia.4.RS39_remake2[df$bin_Ia.4.RS39=="Core"] <- "1"
df$bin_Ia.1.I_remake2="0"
df$bin_Ia.1.I_remake2[df$bin_Ia.1.I=="Core"] <- "1"



test <- df%>%dplyr::select( "bin_Ib.1_remake2", "bin_Ia.3.I_remake2", "bin_Ia.3.VI_remake2", "bin_Ia.3.V_remake2", "bin_Ia.3.IV_remake2", "bin_Ia.3.III_remake2","bin_Ia.3.II_remake2", "bin_Ia.4.1483_remake2", "bin_Ia.4.II_remake2", "bin_Ia.4.RS39_remake2", "bin_Ia.1.I_remake2" )

dat <- as.data.frame(sapply( test, as.numeric ))
dat$sum=rowSums(dat)
Enviro_GC=dat
Enviro_GC$gene_cluster_id=df$gene_cluster_id

##find out which are only found rare
test_rare=Enviro_GC%>%dplyr::select(bin_Ia.3.I_remake2,bin_Ia.3.IV_remake2, bin_Ia.1.I_remake2,sum, gene_cluster_id)%>%filter(sum<4)

test_rare$sum_rare=rowSums(test_rare[1:3])

test_rare$Category[test_rare$sum_rare==test_rare$sum] <- "True"
test_rare_true=test_rare%>%filter(Category=="True")
listy=test_rare_true$gene_cluster_id
write.csv(test_rare, "output/GC_ID_pangenome_by_genus_only_multi_rare.csv")
##find out which are only found offshore

test_offshore=Enviro_GC%>%dplyr::filter( bin_Ia.3.VI_remake2 == "0"  & bin_Ia.3.III_remake2 == "0" & bin_Ia.3.II_remake2 == "0")

##find out which are only found coastal
test_coastal=Enviro_GC%>%dplyr::filter( bin_Ia.3.V_remake2 == "0"  & bin_Ia.4.RS39_remake2 == "0" &bin_Ia.4.1483_remake2 == "0"&bin_Ia.4.II_remake2 == "0"&bin_Ib.1_remake2 == "0")


###
test_offshore$Environment="Offshore"
test_offshore=test_offshore%>% dplyr::filter(!gene_cluster_id %in% listy)

write.csv(test_offshore, "output/GC_ID_pangenome_by_genus_multi_offshore.csv")
test_coastal$Environment="Coastal"
test_coastal=test_coastal%>% dplyr::filter(!gene_cluster_id %in% listy)
write.csv(test_coastal, "output/GC_ID_pangenome_by_genus_multi_coastal.csv")

test_rare_final=Enviro_GC%>% dplyr::filter(gene_cluster_id %in% listy)
test_rare_final$Environment="Rare"
tog=rbind(test_offshore, test_coastal)
tog2=rbind(tog, test_rare_final)

table(tog2$Environment)

tog3=tog2%>%dplyr::select("gene_cluster_id", "Environment")

test=left_join(df,tog3, by="gene_cluster_id")



strict_multilineage_core_not_genus=test%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence, genus_specific,Environment)
strict_multilineage_core_not_genus$Pangenome="multilineage"

strict_multilineage_core_not_genus=strict_multilineage_core_not_genus%>%unite(Enviro_Specific, c('Pangenome', 'Environment'), sep='_', remove = FALSE)

strict_multilineage_core_not_genus$Enviro_Specific[strict_multilineage_core_not_genus$Enviro_Specific=="multilineage_NA"]=" "

strict_multilineage_core_not_genus=strict_multilineage_core_not_genus%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence, genus_specific, Environment, Pangenome, Enviro_Specific)


write.csv(strict_multilineage_core_not_genus,"output/strict_multilineage_conventional_genus_enviro.csv")



library(ComplexUpset)
library(patchwork)
Genera = colnames(compare_cores11)
compare_cores11[Genera] = compare_cores11[Genera] == 1
pdf("output/strict_multilineage_specific_Oct2024_wHIMB1483.pdf", width=12)
upset(compare_cores11, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1,max_degree=100)
dev.off()

svg("output/strict_multilineage_specific_Oct_2024wHIMB1483.svg", width=12)
upset(compare_cores11, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1,max_degree=100)
dev.off()
```







##find GCs that are shared among all eleven genera

```{r}

listy=strict_multilineage_core$gene_cluster_id
listy

test=compare_cores4
test=test%>% dplyr::filter(bin_name !="Core_Pelagibacteraceae")
test=test%>% dplyr::filter(bin_name !="Singleton_Pelagibacteraceae")
test=test%>% dplyr::filter(!gene_cluster_id %in% listy)
gc_id=test
8242-784-4201-182
#3075

test<- data.frame(lapply(test, function(x) {gsub("Core", "1", x)  }))
test<- data.frame(lapply(test, function(x) {gsub("Singleton", "1", x)  }))
test<- data.frame(lapply(test, function(x) {gsub("present", "1", x)  }))
test<- data.frame(lapply(test, function(x) {gsub("absent", "0", x)  }))

test=test[ , !names(test) %in% c("bin_name", "gene_cluster_id")] 

dat <- as.data.frame(sapply( test, as.numeric ))
dat$sum=rowSums(dat)



##to get the gene clusters in 11 genera
eleven_GC=dat
eleven_GC$gene_cluster_id=gc_id$gene_cluster_id
eleven_GC=eleven_GC%>%filter(sum==11)



eleven_GC=left_join(eleven_GC, Gene_Cluster_Summary, by="gene_cluster_id")
eleven_GC=distinct(eleven_GC,gene_cluster_id, .keep_all = TRUE)
eleven_GC=eleven_GC%>%dplyr::select(gene_cluster_id,KOfam_ACC,KOfam,COG20_FUNCTION_ACC, COG20_FUNCTION, aa_sequence)
eleven_GC$genus_specific="NA"
eleven_GC$Environment="NA"
eleven_GC$Pangenome="eleven"
eleven_GC$Enviro_Specific="NA"
write.csv(eleven_GC,"output/eleven_GC_conventional.csv")

names(core)
names(sing)
names(eleven_GC)
names(genus_specific_GCs)
names(strict_multilineage_core_not_genus)
tog=rbind(core, sing, eleven_GC, genus_specific_GCs, strict_multilineage_core_not_genus)
write.csv(tog, "intermediate_tables_figures/Sup_File_2a_Conventional_pangenome_info.csv")
tog2=tog%>%dplyr::select(gene_cluster_id,Pangenome,Environment, Enviro_Specific )
write.table(tog2,"output/conventional_pangenome_GC_data_layerOct2024.txt", row.names=FALSE, sep = "\t",quote = FALSE)

```



###
Add this information to the conventional pangeome
###

```{bash}

anvi-import-misc-data conventional_pangenome_GC_data_layerOct2024.txt -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db --target-data-table items


anvi-display-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db

```



###assess whether the number of gene-specific gene-clusters correlates with number of genome representatives 

```{r}
gc_count=read.csv("genus_specific_gc_correlation_Oct2024.csv")
head(gc_count)
hist(gc_count$No_genomes)
hist(gc_count$genus_specific_gc)
hist(log(gc_count$No_genomes))
hist(log(gc_count$genus_specific_gc))
library(ggpmisc)

gc_count_test<-ggplot(gc_count, aes(x=log(No_genomes), y=log(genus_specific_gc))) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log No. genus-specific GCs")+xlab("log No of Genomes")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("")
gc_count_test





##correlation
ggplot(gc_count, aes(x=log(No_genomes), y=log(genus_specific_gc))) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))



```


####
Functional pangenome
####

```{bash}
anvi-estimate-metabolism -c SAR11_Oct2024_bycontig-CONTIGS.db -p SAR11_Oct2024_bycontig-MERGED/PROFILE.db -C Genomes -O estimate_SAR11_Oct2024

##first see which annotation sources are available
anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt  --category-variable Genus --annotation-source COG20_FUNCTION --functional-occurrence-table-output SAR11_Oct2024-COG20_FUNCTION_FUNC_OCCURRENCE.TXT

mkdir functional_enrichment

anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11_Oct2024-COG20_CATEGORY-functional-enrichment.txt  --category-variable Genus --annotation-source COG20_CATEGORY  --functional-occurrence-table-output functional_enrichment/SAR11_Oct2024-COG20_CATEGORY-FUNC_OCCURRENCE.TXT



anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11_Oct2024-COG20-functional-enrichment.txt  --category-variable Genus --annotation-source COG20_CATEGORY  --functional-occurrence-table-output functional_enrichment/SAR11_Oct2024-COG20-FUNC_OCCURRENCE.TXT


anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt  --category-variable Genus --annotation-source KOfam --functional-occurrence-table-output functional_enrichment/SAR11-Oct2024-KOfam-FUNC_OCCURRENCE.TXT


anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11-Oct2024-KEGG_Module-functional-enrichment.txt  --category-variable Genus --annotation-source KEGG_Module --functional-occurrence-table-output functional_enrichment/SAR11-Oct2024-KEGG_Module-FUNC_OCCURRENCE.TXT


anvi-compute-functional-enrichment-in-pan -p SAR11_Oct2024-by-contig-PAN/SAR11_Oct2024-by-contig-PAN-PAN.db -g SAR11_Oct2024-by-contig-PAN-GENOMES.db -o functional_enrichment/SAR11-Oct2024-KEGG_BRITE-functional-enrichment.txt  --category-variable Genus --annotation-source KEGG_BRITE --functional-occurrence-table-output functional_enrichment/SAR11-Oct2024-KEGG_BRITE-FUNC_OCCURRENCE.TXT

####

##Might delete this one :D
anvi-script-gen-function-matrix-across-genomes -i SAR11-Oct2024-bycontig_internal_genomes.txt \
                                               --annotation-source KOfam \
                                               --output-file-prefix func_matrix \
                                               --groups-txt SAR11_Oct2024_groups.txt



###looking at the functional pangenome 
anvi-display-functions --annotation-source KOfam -g SAR11_Oct2024-by-contig-PAN-GENOMES.db --profile-db Oct2024_KOFAM-PROFILE.db

anvi-import-misc-data Genera_genome_info_layer_Oct2024.txt -p Oct2024_KOFAM-PROFILE.db --target-data-table layers

anvi-interactive --profile-db Oct2024_KOFAM-PROFILE.db --manual

```


####
Let's analyze the functional pangenome 


```{r}
pre_abs=read.delim("func_matrix-PRESENCE-ABSENCE.txt", row.names = 1)
pre_abs_summed=pre_abs

sapply(pre_abs_summed, class)

pre_abs_summed=pre_abs_summed[,1:92]


sapply(pre_abs_summed, class)

pre_abs_summed2 <- data.frame(apply(pre_abs_summed, 2, function(x) as.numeric(as.integer(x))))
sapply(pre_abs_summed2, class)
pre_abs_summed_binary2=pre_abs_summed2

pre_abs_summed_binary2$summed=rowSums(pre_abs_summed_binary2)
pre_abs_summed3=pre_abs_summed_binary2%>%dplyr::group_by(summed)%>%dplyr::count(summed)
view(pre_abs_summed3)

##from this we gather that ther are 612 KOfams core to all 92 genomes, and 70 singltons 
##to get the ids of those groups; 1276 total func

```


```{r}
pre_abs=read.delim("func_matrix-PRESENCE-ABSENCE.txt")
pre_abs_summed=pre_abs

sapply(pre_abs_summed, class)

pre_abs_summed=pre_abs_summed[,2:93]


sapply(pre_abs_summed, class)
dim(pre_abs_summed)
pre_abs_summed2 <- data.frame(apply(pre_abs_summed, 2, function(x) as.numeric(as.integer(x))))
sapply(pre_abs_summed2, class)
pre_abs_summed_binary2=pre_abs_summed2

pre_abs_summed_binary2$summed=rowSums(pre_abs_summed_binary2)
pre_abs_summed_binary2$KOfam=pre_abs$KOfam
pre_abs_summed_binary2$key=pre_abs$key
names(pre_abs_summed)
pre_abs_summed_binary3=pre_abs_summed_binary2%>%dplyr::select("key", "KOfam", "summed")
write.csv(pre_abs_summed_binary3, "Func_distribution_SAR11_Oct2024.csv")

pre_abs_summed3=pre_abs_summed_binary2%>%dplyr::group_by(summed)%>%dplyr::count(summed)
view(pre_abs_summed3)
```



```{r}

Func_Cluster_datatable=read.delim("func_matrix-PRESENCE-ABSENCE.txt")

Func_Cluster_Summary=Func_Cluster_datatable%>%rename_all(recode,"key"="Func_Cluster_id")
KOfam_add_on=Func_Cluster_Summary%>%dplyr::select("Func_Cluster_id", "KOfam")

#fun=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/Functional_all_groups_data_items.txt")
#fun_con=left_join(fun, KOfam_add_on, "Func_Cluster_id")
#write.csv(fun_con, "fun_con.csv")
Func_Cluster_Summary <-Func_Cluster_Summary[ , -which(names(Func_Cluster_Summary) %in% c("KOfam"))]


Func_Cluster_Summary$num_genomes_Func_Cluster_has_hits=rowSums(Func_Cluster_Summary[,-1])
Func_Cluster_Summary$total_func_pangenome_bins=""
Func_Cluster_Summary=Func_Cluster_Summary%>%mutate(total_func_pangenome_bins= ifelse(num_genomes_Func_Cluster_has_hits==92, "Core_Pelagibacteraceae", ifelse(num_genomes_Func_Cluster_has_hits==1, "Singleton_Pelagibacteraceae", '')))
names(Func_Cluster_Summary)
Func_Cluster_Summary_short=Func_Cluster_Summary%>%dplyr::select(num_genomes_Func_Cluster_has_hits, total_func_pangenome_bins, Func_Cluster_id)
Func_Cluster_Summary_short=left_join(Func_Cluster_Summary_short, KOfam_add_on, by="Func_Cluster_id")

Core_sing_func=Func_Cluster_Summary_short%>%dplyr::select(Func_Cluster_id, KOfam,total_func_pangenome_bins)
Core_sing_func=Core_sing_func%>%dplyr::rename("Pangenome"="total_func_pangenome_bins")
Core_sing_func=Core_sing_func%>%subset(Pangenome=="Core_Pelagibacteraceae"| Pangenome=="Singleton_Pelagibacteraceae")
Core_sing_func$genus_specific="NA"
Core_sing_func$Environment="NA"
Core_sing_func$Enviro_specific="NA"
write.csv(Core_sing_func, "output/Core_sing_func.csv")


Func_Cluster_Summary_melt=melt(Func_Cluster_Summary,id.vars = c("Func_Cluster_id", "num_genomes_Func_Cluster_has_hits", "total_func_pangenome_bins"))
names(Func_Cluster_Summary_melt)
Func_Cluster_Summary_melt=Func_Cluster_Summary_melt%>%rename_all(recode,"variable"="genome_name")

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)
count_genera=Taxonomy%>%group_by(Genus)%>%tally()
Tax=Taxonomy%>%dplyr::select("samples", "Genus")
Taxonomy=Tax%>%rename_all(recode,"samples"="genome_name")
names(Taxonomy)



Func_Cluster_Summary_Tax=left_join(Func_Cluster_Summary_melt, Taxonomy, "genome_name")
Core_Pelagibacteraceae=subset(Func_Cluster_Summary_Tax, num_genomes_Func_Cluster_has_hits==92)
length(unique(Core_Pelagibacteraceae$Func_Cluster_id))
names(Core_Pelagibacteraceae)

Singleton_Pelagibacteraceae=subset(Func_Cluster_Summary_Tax, num_genomes_Func_Cluster_has_hits==1)
length(unique(Singleton_Pelagibacteraceae$Func_Cluster_id))


view(Func_Cluster_Summary_Tax)

###Ib.1, n=2
Ib.1=subset(Func_Cluster_Summary_Tax, Genus=="Ib.1")
ngenomes=length(unique(Ib.1$genome_name))
ngenomes
##total pangenome size
length(unique(Ib.1$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ib.1%>%dcast(Func_Cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ib.1="present"
data$bin_Ib.1[data$sum==ngenomes] <- "Core"


data$Ib.1_bin_SCG=paste(data$bin_Ib.1, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ib.1_bin_SCG)%>%dplyr::summarise(n = n())
okay
view(okay)
write.csv(data, "output/Ib.1_bin_SCG_Function.csv")

Ib.1_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ib.1", "Ib.1_bin_SCG"))
view(Ib.1_bin_info)
```


```{r}

###lets define the core and singletons at the genus level 
###We currently have HORRIBLE names for genera so bare with me, here is the first "Genus-Ia.4.RS39"
Ia.4.RS39=subset(Func_Cluster_Summary_Tax, Genus=="Ia.4.RS39")
ngenomes=length(unique(Ia.4.RS39$genome_name))
ngenomes
Ia.4.RS39=Ia.4.RS39%>%filter(value>0)
##total pangenome size
length(unique(Ia.4.RS39$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.RS39%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
View(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.RS39<- "present"
data$bin_Ia.4.RS39[data$sum==ngenomes] <- "Core"


data$Ia.4.RS39_bin_SCG=paste(data$bin_Ia.4.RS39, data$SCG, sep="_")

gg=data%>%dplyr::count(Ia.4.RS39_bin_SCG)
to_compare_Ia.4.RS39=data

#write.csv(data, "Ia.4.RS39_bin_SCG.csv")
okay=data%>%dplyr::group_by(Ia.4.RS39_bin_SCG)%>%dplyr::summarise(n = n())

names(data)

Ia.4.RS39_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.4.RS39", "Ia.4.RS39_bin_SCG"))



###Ia.4.II, n=7
Ia.4.II=subset(Func_Cluster_Summary_Tax, Genus=="Ia.4.II")
ngenomes=length(unique(Ia.4.II$genome_name))
ngenomes
Ia.4.II=Ia.4.II%>%filter(value>0)
##total pangenome size
length(unique(Ia.4.II$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.II%>%dcast(Func_Cluster_id~genome_name)
##add a column to understand whether 
test$SCG ="single_copy"
test[is.na(test)] <- 0
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
View(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.II<- "present"
data$bin_Ia.4.II[data$sum==ngenomes] <- "Core"

data$Ia.4.II_bin_SCG=paste(data$bin_Ia.4.II, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.4.II_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.4.II_bin_SCG.csv")

names(data)

Ia.4.II_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.4.II", "Ia.4.II_bin_SCG"))


###Ia.1.I, n=7
Ia.1.I=subset(Func_Cluster_Summary_Tax, Genus=="Ia.1.I")
ngenomes=length(unique(Ia.1.I$genome_name))
ngenomes
Ia.1.I=Ia.1.I%>%filter(value>0)
##total pangenome size
length(unique(Ia.1.I$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.1.I%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.1.I<- "present"
data$bin_Ia.1.I[data$sum==ngenomes] <- "Core"

data$Ia.1.I_bin_SCG=paste(data$bin_Ia.1.I, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.1.I_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.1.I_bin_SCG.csv")
names(data)

Ia.1.I_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.1.I", "Ia.1.I_bin_SCG"))


###Ia.3.III, n=5
Ia.3.III=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.III")
ngenomes=length(unique(Ia.3.III$genome_name))
ngenomes
Ia.3.III=Ia.3.III%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.III$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.III%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.III<- "present"
data$bin_Ia.3.III[data$sum==ngenomes] <- "Core"


data$Ia.3.III_bin_SCG=paste(data$bin_Ia.3.III, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.III_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.III_bin_SCG.csv")

names(data)

Ia.3.III_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.III", "Ia.3.III_bin_SCG"))




###Ia.3.II, n=6
Ia.3.II=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.II")
ngenomes=length(unique(Ia.3.II$genome_name))
ngenomes
Ia.3.II=Ia.3.II%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.II$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.II%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.II<- "present"
data$bin_Ia.3.II[data$sum==ngenomes] <- "Core"


data$Ia.3.II_bin_SCG=paste(data$bin_Ia.3.II, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.II_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.II_bin_SCG.csv")
names(data)

Ia.3.II_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.II", "Ia.3.II_bin_SCG"))






###Ia.3.I, n=3
Ia.3.I=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.I")
ngenomes=length(unique(Ia.3.I$genome_name))
ngenomes
Ia.3.I=Ia.3.I%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.I$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.I%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)

###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.I<- "present"
data$bin_Ia.3.I[data$sum==ngenomes] <- "Core"


data$Ia.3.I_bin_SCG=paste(data$bin_Ia.3.I, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.I_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.I_bin_SCG.csv")
names(data)

Ia.3.I_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.I", "Ia.3.I_bin_SCG"))





###Ia.3.IV, n=2
Ia.3.IV=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.IV")
ngenomes=length(unique(Ia.3.IV$genome_name))
ngenomes
Ia.3.IV=Ia.3.IV%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.IV$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.IV%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.IV <- "present"
data$bin_Ia.3.IV[data$sum==ngenomes] <- "Core"


data$Ia.3.IV_bin_SCG=paste(data$bin_Ia.3.IV, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.IV_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.IV_bin_SCG.csv")

names(data)

Ia.3.IV_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.IV", "Ia.3.IV_bin_SCG"))




###Ia.3.V, n=9
Ia.3.V=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.V")
ngenomes=length(unique(Ia.3.V$genome_name))
ngenomes
Ia.3.V=Ia.3.V%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.V$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.V%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.V="present"
data$bin_Ia.3.V[data$sum==ngenomes] <- "Core"


data$Ia.3.V_bin_SCG=paste(data$bin_Ia.3.V, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.V_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.V_bin_SCG.csv")
names(data)

Ia.3.V_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.V", "Ia.3.V_bin_SCG"))




###Ia.3.VI, n=47
Ia.3.VI=subset(Func_Cluster_Summary_Tax, Genus=="Ia.3.VI")
ngenomes=length(unique(Ia.3.VI$genome_name))
ngenomes
Ia.3.VI=Ia.3.VI%>%filter(value>0)
##total pangenome size
length(unique(Ia.3.VI$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.3.VI%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
okay
View(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.3.VI="present"
data$bin_Ia.3.VI[data$sum==ngenomes] <- "Core"

data$Ia.3.VI_bin_SCG=paste(data$bin_Ia.3.VI, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.3.VI_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.3.VI_bin_SCG.csv")
names(data)

Ia.3.VI_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.3.VI", "Ia.3.VI_bin_SCG"))




###Ib.1, n=2
Ib.1=subset(Func_Cluster_Summary_Tax, Genus=="Ib.1")
ngenomes=length(unique(Ib.1$genome_name))
ngenomes
Ib.1=Ib.1%>%filter(value>0)
##total pangenome size
length(unique(Ib.1$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ib.1%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
length(unique(test$Func_Cluster_id))
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))
test3 <- test %>% mutate(SCG = ifelse(apply(test[,-which(names(test) %in% c("SCG", "Func_Cluster_id"))] > 1, 1, any), "multiple_copy", SCG))
###new way
data=test3
data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))] = ifelse(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))]  >= 1, 1, 0)
data$sum=rowSums(data[,-which(names(data) %in% c("SCG", "Func_Cluster_id"))])
okay=data%>%dplyr::group_by(sum)%>%dplyr::summarise(n = n())
view(okay)
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ib.1="present"
data$bin_Ib.1[data$sum==ngenomes] <- "Core"


data$Ib.1_bin_SCG=paste(data$bin_Ib.1, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ib.1_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ib.1_bin_SCG.csv")

Ib.1_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ib.1", "Ib.1_bin_SCG"))


#####

###Ia.4.1483, n=1
Ia.4.1483=subset(Func_Cluster_Summary_Tax, Genus=="Ia.4.1483")
ngenomes=length(unique(Ia.4.1483$genome_name))
ngenomes
Ia.4.1483=Ia.4.1483%>%filter(value>0)
##total pangenome size
length(unique(Ia.4.1483$Func_Cluster_id))
##to get some core and singletons lets reformat the table
library(reshape2)
test=Ia.4.1483%>%dcast(Func_Cluster_id~genome_name)
test[is.na(test)] <- 0
##add a column to understand whether 
test$SCG ="single_copy"
#test3 <- test %>% mutate(SCG = ifelse(apply(test[,2:5] > 1, 1, any), "multiple_copy", SCG))

test3=test

test3$SCG[test3$HIMB1483 > 1] <-  "multiple_copy"
###new way
data=test3
###Note the core number changes based on the number of genomes present in the Genus
data$bin_Ia.4.1483="present"
data$Ia.4.1483_bin_SCG=paste(data$bin_Ia.4.1483, data$SCG, sep="_")
okay=data%>%dplyr::group_by(Ia.4.1483_bin_SCG)%>%dplyr::summarise(n = n())
okay
#write.csv(data, "Ia.4.1483_bin_SCG.csv")

Ia.4.1483_bin_info=data%>%dplyr::select(c("Func_Cluster_id", "bin_Ia.4.1483", "Ia.4.1483_bin_SCG"))


```


```{r}

a=left_join(Func_Cluster_Summary_Tax, Ib.1_bin_info, by="Func_Cluster_id")
b=left_join(a, Ia.3.VI_bin_info, by="Func_Cluster_id")
c=left_join(b, Ia.3.V_bin_info, by="Func_Cluster_id")
d=left_join(c, Ia.3.IV_bin_info, by="Func_Cluster_id")
e=left_join(d, Ia.3.I_bin_info, by="Func_Cluster_id")
f=left_join(e, Ia.3.II_bin_info, by="Func_Cluster_id")
g=left_join(f, Ia.3.III_bin_info, by="Func_Cluster_id")
h=left_join(g, Ia.1.I_bin_info, by="Func_Cluster_id")
i=left_join(h, Ia.4.II_bin_info, by="Func_Cluster_id")
j=left_join(i, Ia.4.RS39_bin_info, by="Func_Cluster_id")
functional_pangenome_SAR11=left_join(j, Ia.4.1483_bin_info, by="Func_Cluster_id")

#write.csv(functional_pangenome_SAR11, "by_contigs_all_bins_Func_Cluster_Summary_Tax.csv")
#functional_pangenome_SAR11=read.csv("by_contigs_all_bins_Func_Cluster_Summary_Tax.csv")
names(functional_pangenome_SAR11)



compare_cores=functional_pangenome_SAR11%>%dplyr::select(c("Func_Cluster_id", "total_func_pangenome_bins","bin_Ib.1", "bin_Ia.3.VI", "bin_Ia.3.V", "bin_Ia.3.IV", "bin_Ia.3.I","bin_Ia.3.II",  "bin_Ia.3.II", "bin_Ia.3.III","bin_Ia.1.I", "bin_Ia.4.II", "bin_Ia.4.RS39", "bin_Ia.4.1483" ))

unique(compare_cores$bin_Ia.4.1483)



#compare_cores2=subset(compare_cores, bin_name!="Singleton_Pelagibacteraceae"& bin_name!="Core_Pelagibacteraceae")
compare_cores3=distinct(compare_cores, Func_Cluster_id, .keep_all = TRUE)
ugh=compare_cores3%>%filter(total_func_pangenome_bins=="Core_Pelagibacteraceae")

compare_cores4=compare_cores3%>%dplyr::select("total_func_pangenome_bins","bin_Ib.1", "bin_Ia.3.VI", "bin_Ia.3.V", "bin_Ia.3.IV", "bin_Ia.3.I","bin_Ia.3.II",  "bin_Ia.3.II", "bin_Ia.3.III","bin_Ia.1.I", "bin_Ia.4.II", "bin_Ia.4.RS39","bin_Ia.4.1483","Func_Cluster_id")
write.csv(compare_cores4, "output/func_compare_cores4.csv")

compare_cores4[is.na(compare_cores4)] <- "absent"
names(compare_cores4)


#specific Ib.1
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "Core" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#5 unique Ib.1 genes
Ib.1_specific=test$Func_Cluster_id
length(Ib.1_specific)
Ib.1_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ib.1_func$genus_specific="unique_Ib.1"


#specific Ia.3.VI
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "Core" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent"& bin_Ia.4.1483== "absent" )
#0 unique Ia.3.VI genes
Ia.3.VI_specific=test$Func_Cluster_id
length(Ia.3.VI_specific)
Ia.3.VI_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.VI_func$genus_specific="unique_Ia.3.VI"

#specific Ia.3.IV
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "Core" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#0 unique Ia.3.IV genes
Ia.3.IV_specific=test$Func_Cluster_id
length(Ia.3.IV_specific)
Ia.3.IV_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.IV_func$genus_specific="unique_Ia.3.IV"

#specific Ia.3.III
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "Core" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#3 unique Ia.3.III genes
Ia.3.III_specific=test$Func_Cluster_id
length(Ia.3.III_specific)
Ia.3.III_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.III_func$genus_specific="unique_Ia.3.III"



#specific Ia.3.II
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "Core" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#0 unique Ia.3.II genes
Ia.3.II_specific=test$Func_Cluster_id
length(Ia.3.II_specific)
Ia.3.II_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.II_func$genus_specific="unique_Ia.3.II"


#specific Ia.3.I
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "Core" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent")
#3 unique Ia.3.I genes
Ia.3.I_specific=test$Func_Cluster_id
length(Ia.3.I_specific)
Ia.3.I_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.I_func$genus_specific="unique_Ia.3.I"

#specific Ia.1.I
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "Core" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
#0 unique Ia.1.I genes
Ia.1.I_specific=test$Func_Cluster_id
length(Ia.1.I_specific)
Ia.1.I_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.1.I_func$genus_specific="unique_Ia.1.I"

#specific Ia.4.II
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "Core" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
#0 unique Ia.4.II genes
Ia.4.II_specific=test$Func_Cluster_id
length(Ia.4.II_specific)
Ia.4.II_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.4.II_func$genus_specific="unique_Ia.4.II"



#specific Ia.4.RS39
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "Core" & bin_Ia.4.1483== "absent" )
Ia.4.RS39_specific=test$Func_Cluster_id
#0 unique Ia.4.RS39 genes
length(Ia.4.RS39_specific)
Ia.4.RS39_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.4.RS39_func$genus_specific="unique_Ia.4.RS39"




#specific Ia.3.V
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "Core" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
Ia.3.V_specific=test$Func_Cluster_id
#3 unique Ia.3.V genes
Ia.3.V_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.3.V_func$genus_specific="unique_Ia.3.V"


#specific Ia.4.1483
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "present" )
#3 unique Ia.4.1483 genes
Ia.4.1483_specific=test$Func_Cluster_id
Ia.4.1483_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")
Ia.4.1483_func$genus_specific="unique_Ia.4.1483"

###All functions- NOT HIMB1483 is not inclued in this
genus_specific_func_clusters=rbind(Ia.4.RS39_func, Ia.4.II_func, Ia.1.I_func, Ia.3.I_func, Ia.3.II_func, Ia.3.III_func, Ia.3.IV_func, Ia.3.V_func, Ia.3.VI_func, Ib.1_func)

genus_specific_func=genus_specific_func_clusters
genus_specific_func
names(genus_specific_func)

genus_specific_func$Environment="NA"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.VI"] <- "Coastal"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.II"] <- "Coastal"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.III"] <- "Coastal"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.V"] <- "Offshore"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.4.II"] <- "Offshore"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.4.RS39"] <- "Offshore"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ib.1"] <- "Offshore"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.I"] <- "Rare"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.3.IV"] <- "Rare"
genus_specific_func$Environment[genus_specific_func$genus_specific=="unique_Ia.1.I"] <- "Rare"
genus_specific_func$Pangenome="genus_specific"


genus_specific_func=genus_specific_func%>%unite(Enviro_specific, c('Pangenome', 'Environment'), sep='_', remove = FALSE)

genus_specific_func=genus_specific_func%>%dplyr::select("Func_Cluster_id","KOfam","Pangenome","genus_specific", "Environment", "Enviro_specific")

write.csv(genus_specific_func, "output/genus_specific_functional_Kofam_clusters.csv")



####specific tests
##missing_Ia.4.RS39_Ia.4.II
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "Core" & bin_Ia.3.VI == "Core" & bin_Ia.3.V == "Core" & bin_Ia.3.IV == "Core" & bin_Ia.3.III == "Core" & bin_Ia.3.II == "Core" & bin_Ia.3.I == "Core" & bin_Ia.1.I == "Core" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "present" )
absent_Ia.4.RS39_Ia.4.II_specific=test$Func_Cluster_id

missing_Ia.4.RS39_Ia.4.II=left_join(test, KOfam_add_on, by="Func_Cluster_id")

write.csv(missing_Ia.4.RS39_Ia.4.II, "output/missing_Ia.4.RS39_Ia.4.II_KOfam.csv")
#specific Ia.3.VI, Ia.3.II. Ia.3.III
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "Core" & bin_Ia.3.V == "absent"  & bin_Ia.3.III == "Core" & bin_Ia.3.II == "Core"  & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
coastal_specific=test$Func_Cluster_id

coastal_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")

#specific Ia.3.VI, Ia.3.II. Ia.3.III
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI != "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.III != "absent"   & bin_Ia.3.II != "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
coastal_specific=test$Func_Cluster_id

coastal_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")


#specific Ia.3.VI, within coastal
test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI != "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
coastal_specific=test$Func_Cluster_id
coastal_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.III != "absent" & bin_Ia.3.V == "absent" & bin_Ia.3.IV == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "absent" )
coastal_specific=test$Func_Cluster_id
coastal_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")


#specific Ia.3.V and Ia.4.1483

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "absent" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "Core" & bin_Ia.3.IV == "absent" & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" & bin_Ia.3.I == "absent" & bin_Ia.1.I == "absent" & bin_Ia.4.II == "absent" & bin_Ia.4.RS39 == "absent" & bin_Ia.4.1483== "present" )
Ia.3.V_Ia.4.1483_specific=test$Func_Cluster_id

Ia.3.V_Ia.4.1483_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")




#specific offshore

test=compare_cores4%>%dplyr::filter(bin_Ib.1 == "Core" & bin_Ia.3.VI == "absent" & bin_Ia.3.V == "Core"  & bin_Ia.3.III == "absent" & bin_Ia.3.II == "absent" &   bin_Ia.4.II == "Core" & bin_Ia.4.RS39 == "Core" & bin_Ia.4.1483== "present" )
Offshore_specific=test$Func_Cluster_id

Offshore_specific_func=left_join(test, KOfam_add_on, by="Func_Cluster_id")

```

###multilineage functional IDs


```{r}
compare_cores4[is.na(compare_cores4)] <- "absent"
names(compare_cores4)
ugh=compare_cores4%>%filter(total_func_pangenome_bins =="Core_Pelagibacteraceae")

unique(compare_cores4$bin_Ia.4.1483)
strict_multilineage_core=compare_cores4%>%dplyr:: filter(bin_Ib.1 %in% c("absent", "Core") & bin_Ia.3.VI%in% c("absent", "Core") & bin_Ia.3.V%in% c("absent", "Core") & bin_Ia.3.IV%in% c("absent", "Core") & bin_Ia.3.III %in% c("absent", "Core") & bin_Ia.3.II %in% c("absent", "Core") & bin_Ia.3.I  %in% c("absent", "Core") & bin_Ia.1.I%in% c("absent", "Core") & bin_Ia.4.II%in% c("absent", "Core") & bin_Ia.4.RS39 %in% c("absent", "Core") &  bin_Ia.4.1483 %in% c("absent", "present"))
view(strict_multilineage_core)




##note this will remove HIMB1483 genus-specific genes
strict_multilineage_core=strict_multilineage_core%>% dplyr::filter(total_func_pangenome_bins !="Singleton_Pelagibacteraceae")
#strict_multilineage_core=strict_multilineage_core%>% dplyr::filter(total_func_pangenome_bins !="Core_Pelagibacteraceae")
#view(strict_multilineage_core)
#write.csv(strict_multilineage_core, "strict_multilineage_core_functional.csv")
#this gives you multicore specific for the pangenome
#perparing for upset 
compare_cores8=strict_multilineage_core[ , !names(strict_multilineage_core) %in% c( "Func_Cluster_id", "total_func_pangenome_bins")] 
compare_cores9<- data.frame(lapply(compare_cores8, function(x) {gsub("Core", "1", x)  }))
compare_cores10<- data.frame(lapply(compare_cores9, function(x) {gsub("absent", "0", x)  }))
compare_cores11<- data.frame(lapply(compare_cores10, function(x) {gsub("present", "1", x)  }))



library(ComplexUpset)
library(patchwork)
Genera = colnames(compare_cores11)
compare_cores11[Genera] = compare_cores11[Genera] == 1
svg("output/functional_strict_multilineage_specific.svg", width=8, height=6)
upset(compare_cores11, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1)
dev.off()

pdf()
pdf("output/functional_strict_multilineage_specific.pdf", width=6, height=6)
upset(compare_cores11, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1,max_degree=100)
dev.off()





##Find out multilineage 
multilineage_only=strict_multilineage_core%>% dplyr::filter(total_func_pangenome_bins !="Core_Pelagibacteraceae")

listy=genus_specific_func$Func_Cluster_id

multilineage_only=multilineage_only%>% dplyr::filter(!Func_Cluster_id %in% listy)
multilineage_only=left_join(multilineage_only, KOfam_add_on, by="Func_Cluster_id")



####
df=multilineage_only
df$Enviro="NA"
df$bin_Ib.1_remake="NA"
df$bin_Ib.1_remake[df$bin_Ib.1=="Core"] <- "Ib.1"
df$bin_Ia.3.I_remake="NA"
df$bin_Ia.3.I_remake[df$bin_Ia.3.I=="Core"] <- "Ia.3.I"
df$bin_Ia.3.VI_remake="NA"
df$bin_Ia.3.VI_remake[df$bin_Ia.3.VI=="Core"] <- "Ia.3.VI"
df$bin_Ia.3.V_remake="NA"
df$bin_Ia.3.V_remake[df$bin_Ia.3.V=="Core"] <- "Ia.3.V"
df$bin_Ia.3.IV_remake="NA"
df$bin_Ia.3.IV_remake[df$bin_Ia.3.IV=="Core"] <- "Ia.3.IV"
df$bin_Ia.3.III_remake="NA"
df$bin_Ia.3.III_remake[df$bin_Ia.3.III=="Core"] <- "Ia.3.III"
df$bin_Ia.3.II_remake="NA"
df$bin_Ia.3.II_remake[df$bin_Ia.3.II=="Core"] <- "Ia.3.II"
df$bin_Ia.4.1483_remake="NA"
df$bin_Ia.4.1483_remake[df$bin_Ia.4.1483=="present"] <- "Ia.4.1483"
df$bin_Ia.4.II_remake="NA"
df$bin_Ia.4.II_remake[df$bin_Ia.4.II=="Core"] <- "Ia.4.II"
df$bin_Ia.4.RS39_remake="NA"
df$bin_Ia.4.RS39_remake[df$bin_Ia.4.RS39=="Core"] <- "Ia.4.RS39"
df$bin_Ia.1.I_remake="NA"
df$bin_Ia.1.I_remake[df$bin_Ia.1.I=="Core"] <- "Ia.1.I"


# columns to paste together
cols <- c( "bin_Ib.1_remake", "bin_Ia.3.I_remake", "bin_Ia.3.VI_remake", "bin_Ia.3.V_remake", "bin_Ia.3.IV_remake", "bin_Ia.3.III_remake","bin_Ia.3.II_remake", "bin_Ia.4.1483_remake", "bin_Ia.4.II_remake", "bin_Ia.4.RS39_remake", "bin_Ia.1.I_remake" )

# create a new column `x` with the three columns collapsed together
df$genus_specific <- apply( df[ , cols ] , 1 , paste , collapse = "," )



mult2=df%>%dplyr::select("Func_Cluster_id", "KOfam", "genus_specific")
mult2$Pangenome="multilineage"

df$bin_Ib.1_remake2="0"
df$bin_Ib.1_remake2[df$bin_Ib.1=="Core"] <- "1"
df$bin_Ia.3.I_remake2="0"
df$bin_Ia.3.I_remake2[df$bin_Ia.3.I=="Core"] <- "1"
df$bin_Ia.3.VI_remake2="0"
df$bin_Ia.3.VI_remake2[df$bin_Ia.3.VI=="Core"] <- "1"
df$bin_Ia.3.V_remake2="0"
df$bin_Ia.3.V_remake2[df$bin_Ia.3.V=="Core"] <- "1"
df$bin_Ia.3.IV_remake2="0"
df$bin_Ia.3.IV_remake2[df$bin_Ia.3.IV=="Core"] <- "1"
df$bin_Ia.3.III_remake2="0"
df$bin_Ia.3.III_remake2[df$bin_Ia.3.III=="Core"] <- "1"
df$bin_Ia.3.II_remake2="0"
df$bin_Ia.3.II_remake2[df$bin_Ia.3.II=="Core"] <- "1"
df$bin_Ia.4.1483_remake2="0"
df$bin_Ia.4.1483_remake2[df$bin_Ia.4.1483=="present"] <- "1"
df$bin_Ia.4.II_remake2="0"
df$bin_Ia.4.II_remake2[df$bin_Ia.4.II=="Core"] <- "1"
df$bin_Ia.4.RS39_remake2="0"
df$bin_Ia.4.RS39_remake2[df$bin_Ia.4.RS39=="Core"] <- "1"
df$bin_Ia.1.I_remake2="0"
df$bin_Ia.1.I_remake2[df$bin_Ia.1.I=="Core"] <- "1"


test <- df%>%dplyr::select( "bin_Ib.1_remake2", "bin_Ia.3.I_remake2", "bin_Ia.3.VI_remake2", "bin_Ia.3.V_remake2", "bin_Ia.3.IV_remake2", "bin_Ia.3.III_remake2","bin_Ia.3.II_remake2", "bin_Ia.4.1483_remake2", "bin_Ia.4.II_remake2", "bin_Ia.4.RS39_remake2", "bin_Ia.1.I_remake2" )

dat <- as.data.frame(sapply( test, as.numeric ))
dat$sum=rowSums(dat)
Enviro_Func=dat
Enviro_Func$Func_Cluster_id=df$Func_Cluster_id

##find out which are only found environment

test_rare=Enviro_Func%>%dplyr::select(bin_Ia.3.I_remake2,bin_Ia.3.IV_remake2, bin_Ia.1.I_remake2,sum, Func_Cluster_id)%>%filter(sum<4)

test_rare$sum_rare=rowSums(test_rare[1:3])

test_rare$Category[test_rare$sum_rare==test_rare$sum] <- "True"
test_rare_true=test_rare%>%filter(Category=="True")
listy=test_rare_true$Func_Cluster_id
write.csv(test_rare, "output/Func_Cluster_id_multi_rare.csv")
##find out which are only found offshore

test_offshore=Enviro_Func%>%dplyr::filter( bin_Ia.3.VI_remake2 == "0"  & bin_Ia.3.III_remake2 == "0" & bin_Ia.3.II_remake2 == "0")

##find out which are only found coastal
test_coastal=Enviro_Func%>%dplyr::filter( bin_Ia.3.V_remake2 == "0"  & bin_Ia.4.RS39_remake2 == "0" & bin_Ia.4.1483_remake2 == "0"&bin_Ia.4.II_remake2 == "0"&bin_Ib.1_remake2 == "0")


###
test_offshore$Environment="Offshore"
test_offshore=test_offshore%>% dplyr::filter(!Func_Cluster_id %in% listy)

write.csv(test_offshore, "output/Func_Cluster_id_multi_offshore.csv")
test_coastal$Environment="Coastal"
test_coastal=test_coastal%>% dplyr::filter(!Func_Cluster_id %in% listy)
write.csv(test_coastal, "output/Func_Cluster_id_multi_coastal.csv")

test_rare_final=Enviro_Func%>% dplyr::filter(Func_Cluster_id %in% listy)
test_rare_final$Environment="Rare"
tog=rbind(test_offshore, test_coastal)
tog2=rbind(tog, test_rare_final)

table(tog2$Environment)

Environment_multi=tog2%>%dplyr::select("Func_Cluster_id", "Environment")


mult2=left_join(mult2, Environment_multi, by="Func_Cluster_id")

mult2=mult2%>%unite(Enviro_specific, c('Pangenome', 'Environment'), sep='_', remove = FALSE)

mult2$Enviro_specific[mult2$Enviro_specific=="multilineage_NA"]=" "
mult2=mult2%>%dplyr::select("Func_Cluster_id", "KOfam", "Pangenome", "genus_specific", "Environment", "Enviro_specific")

write.csv(mult2, "output/Func_Cluster_id_multilineage_withEnviro.csv")

```







####Intermediate to get eleven 

```{r}

####Intermediate
view(compare_cores4)
unique(compare_cores4$bin_Ia.4.1483)
strict_multilineage_core=compare_cores4%>%dplyr:: filter(bin_Ib.1 %in% c("present", "Core") & bin_Ia.3.VI%in% c("present", "Core") & bin_Ia.3.V%in% c("present", "Core") & bin_Ia.3.IV%in% c("present", "Core") & bin_Ia.3.III %in% c("present", "Core") & bin_Ia.3.II %in% c("present", "Core") & bin_Ia.3.I  %in% c("present", "Core") & bin_Ia.1.I%in% c("present", "Core") & bin_Ia.4.II%in% c("present", "Core") & bin_Ia.4.RS39 %in% c("present", "Core") &  bin_Ia.4.1483 %in% c("present", "present"))
view(strict_multilineage_core)


unique(compare_cores4$total_func_pangenome_bins)

unique(compare_cores4$total_func_pangenome_bins)

test=compare_cores4%>%filter(total_func_pangenome_bins !="Core_Pelagibacteraceae")
test=test%>%filter(total_func_pangenome_bins !="Singleton_Pelagibacteraceae")
test8=test[ , !names(test) %in% c( "total_func_pangenome_bins")] 
test8<- data.frame(lapply(test8, function(x) {gsub("Core", "1", x)  }))
test8<- data.frame(lapply(test8, function(x) {gsub("absent", "0", x)  }))
test8<- data.frame(lapply(test8, function(x) {gsub("present", "1", x)  }))

test10=test8[ , !names(test8) %in% c( "Func_Cluster_id")] 


library(ComplexUpset)
library(patchwork)
Genera = colnames(test10)
test10[Genera] = test10[Genera] == 1
svg("output/functional_strict_multilineage_specific.svg", width=8, height=6)
upset(test10, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1)
dev.off()

pdf()
pdf("output/functional_strict_multilineage_specific.pdf", width=6, height=6)
upset(compare_cores11, Genera, name='Genera', width_ratio=0.1,set_sizes=FALSE,min_size=1,min_degree=1,max_degree=100)
dev.off()

####lets get functions that are shared by 11 genera

sapply(test8, class)
test9=test8%>%
  dplyr::select(-Func_Cluster_id) %>%
  mutate_if(is.character,as.numeric) 
sapply(test9, class)

test9$Func_Cluster_id=test8$Func_Cluster_id

test9$sum=rowSums(test9[1:11])

table(test9$sum)
test9_func=left_join(test9, KOfam_add_on, by="Func_Cluster_id")

eleven_fun=test9_func%>%subset(sum=="11")
eleven_fun=eleven_fun%>%dplyr::select(Func_Cluster_id, KOfam)
eleven_fun$Pangenome="eleven"
eleven_fun$genus_specific="NA"
eleven_fun$Environment="NA"
eleven_fun$Enviro_specific="NA"
write.csv(eleven_fun, "output/eleven_fun.csv")

```



```{r}
names(Core_sing_func)
names(eleven_fun)
names(genus_specific_func)
names(mult2)
tog=rbind(Core_sing_func, eleven_fun, genus_specific_func, mult2)

write.csv(tog, "output/Func_pangeome_enviro_specific.csv")

tog2=tog%>%dplyr::select(Func_Cluster_id,Pangenome,Environment, Enviro_specific )
write.table(tog2,"output/functional_pangenome_data_layerOct2024.txt", row.names=FALSE, sep = "\t",quote = FALSE)


table(tog2$Enviro_specific)

```

###
Add this information to the functional pangenome
###
```{bash}

anvi-import-misc-data functional_pangenome_data_layerOct2024.txt \
                              -p Oct2024_KOFAM-PROFILE.db \
                              --target-data-table items
 
anvi-interactive --profile-db Oct2024_KOFAM-PROFILE.db --manual

```





###What about the distribution of these genomes across samples?
Using the output in SAR11_Oct2024-by-contig-SUMMARY we can get lots of information at a gene level about the coverage of individual genes. 

```{r}
Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)
Genus_names=Taxonomy%>%dplyr::rename_all(recode,"samples"="bin")
Genus_names=Genus_names%>%dplyr::select("bin", "Genus")

write.table(Genus_names,"/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/Genus_names.txt", row.names=FALSE, sep = "\t",quote = FALSE)

```

```{bash}
##re-organize the gene non outlier coverages to be group by genera
cd /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/
mkdir SAR11_Read_Recruitment_By_Genus
IFS=$'\n'; for line in `cat Genus_names.txt | tail -n +2`; do
  bin=$(echo "$line" | awk '{print $1}');
  Genus=$(echo "$line" | awk '{print $2}' | tr -d '\r');      
  dest=~/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Read_Recruitment_By_Genus/$Genus;
  mkdir -p "$dest";
  cp ~/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bin_by_bin/$bin/$bin-gene_non_outlier_coverages.txt "$dest/";
done

```


```{r}

##Pick a gene_non_outlier_coverage file from your summarized data to define the samples in your read recruitment and create a header template for building the updated summed read recruitment data 
template_SCG_v2=read.delim("SAR11_Read_Recruitment_By_Genus/Ia.1.I/HTCC1013-gene_non_outlier_coverages.txt", row.name=1)
template_SCG_v2=template_SCG_v2[1,]
template_SCG_v2
rownames(template_SCG_v2)[1] <- 'delete'


##make sure there is a template for each of the genera
SumsIa.4.II_Ia.4.1483=template_SCG_v2
SumsIa.4.1483=template_SCG_v2
SumsIa.4.RS39=template_SCG_v2
SumsIa.4.II=template_SCG_v2
SumsIa.3.III=template_SCG_v2
SumsIa.3.II=template_SCG_v2
SumsIa.3.VI=template_SCG_v2
SumsIa.3.V=template_SCG_v2
SumsIb.1=template_SCG_v2
SumsIa.1.I=template_SCG_v2
SumsIa.3.IV=template_SCG_v2
SumsIa.3.I=template_SCG_v2

library(tidyverse)
```




##Ia.3.V
```{r}

###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.V_SCG=subset(pangenome_SAR11, Ia.3.V_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.V_SCG=pangenome_SAR11_Ia.3.V_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.V_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.V_SCG=subset(pangenome_SAR11_Ia.3.V_SCG, Genus=="Ia.3.V")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.V/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.V/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.V_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.V_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.V[nrow(SumsIa.3.V) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.V)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.V)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.V_v2=SumsIa.3.V[-1,]

write.csv(SumsIa.3.V_v2, "SumsIa.3.V_v2.csv")


```
##Ia.3.VI
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.VI_SCG=subset(pangenome_SAR11, Ia.3.VI_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.VI_SCG=pangenome_SAR11_Ia.3.VI_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.VI_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.VI_SCG=subset(pangenome_SAR11_Ia.3.VI_SCG, Genus=="Ia.3.VI")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.VI/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.VI/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.VI_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.VI_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.VI[nrow(SumsIa.3.VI) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.VI)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.VI)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.VI_v2=SumsIa.3.VI[-1,]


write.csv(SumsIa.3.VI_v2, "SumsIa.3.VI_v2.csv")


```




##Ia.3.I
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.I_SCG=subset(pangenome_SAR11, Ia.3.I_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.I_SCG=pangenome_SAR11_Ia.3.I_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.I_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.I_SCG=subset(pangenome_SAR11_Ia.3.I_SCG, Genus=="Ia.3.I")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.I/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.I/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.I_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.I_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.I[nrow(SumsIa.3.I) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.I)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.I)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.I_v2=SumsIa.3.I[-1,]

write.csv(SumsIa.3.I_v2, "SumsIa.3.I_v2.csv")


```

##Ia.3.II
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.II_SCG=subset(pangenome_SAR11, Ia.3.II_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.II_SCG=pangenome_SAR11_Ia.3.II_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.II_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.II_SCG=subset(pangenome_SAR11_Ia.3.II_SCG, Genus=="Ia.3.II")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.II/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.II/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.II_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.II_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.II[nrow(SumsIa.3.II) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.II)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.II)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.II_v2=SumsIa.3.II[-1,]

write.csv(SumsIa.3.II_v2, "SumsIa.3.II_v2.csv")


```


##Ia.3.III
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.III_SCG=subset(pangenome_SAR11, Ia.3.III_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.III_SCG=pangenome_SAR11_Ia.3.III_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.III_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.III_SCG=subset(pangenome_SAR11_Ia.3.III_SCG, Genus=="Ia.3.III")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.III/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.III/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.III_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.III_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.III[nrow(SumsIa.3.III) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.III)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.III)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.III_v2=SumsIa.3.III[-1,]

write.csv(SumsIa.3.III_v2, "SumsIa.3.III_v2.csv")


```




##Ia.3.IV
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.3.IV_SCG=subset(pangenome_SAR11, Ia.3.IV_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.3.IV_SCG=pangenome_SAR11_Ia.3.IV_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.IV_bin_SCG", "Genus" )
pangenome_SAR11_Ia.3.IV_SCG=subset(pangenome_SAR11_Ia.3.IV_SCG, Genus=="Ia.3.IV")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.3.IV/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.3.IV/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.3.IV_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.3.IV_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.3.IV[nrow(SumsIa.3.IV) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.3.IV)
##update the name of the row in the template based on the genome name
row.names(SumsIa.3.IV)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.3.IV_v2=SumsIa.3.IV[-1,]

write.csv(SumsIa.3.IV_v2, "SumsIa.3.IV_v2.csv")


```



##Ia.4.II
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.4.II_SCG=subset(pangenome_SAR11, Ia.4.II_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.4.II_SCG=pangenome_SAR11_Ia.4.II_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.II_bin_SCG", "Genus" )
pangenome_SAR11_Ia.4.II_SCG=subset(pangenome_SAR11_Ia.4.II_SCG, Genus=="Ia.4.II")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.4.II/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.4.II/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.4.II_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.II_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.4.II[nrow(SumsIa.4.II) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.4.II)
##update the name of the row in the template based on the genome name
row.names(SumsIa.4.II)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.4.II_v2=SumsIa.4.II[-1,]

write.csv(SumsIa.4.II_v2, "SumsIa.4.II_v2.csv")


```


##Ia.4.RS39
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.4.RS39_SCG=subset(pangenome_SAR11, Ia.4.RS39_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.4.RS39_SCG=pangenome_SAR11_Ia.4.RS39_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.RS39_bin_SCG", "Genus" )
pangenome_SAR11_Ia.4.RS39_SCG=subset(pangenome_SAR11_Ia.4.RS39_SCG, Genus=="Ia.4.RS39")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.4.RS39/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.4.RS39/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.4.RS39_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.RS39_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.4.RS39[nrow(SumsIa.4.RS39) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.4.RS39)
##update the name of the row in the template based on the genome name
row.names(SumsIa.4.RS39)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.4.RS39_v2=SumsIa.4.RS39[-1,]

write.csv(SumsIa.4.RS39_v2, "SumsIa.4.RS39_v2.csv")


```




##Ia.1.I
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ia.1.I_SCG=subset(pangenome_SAR11, Ia.1.I_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.1.I_SCG=pangenome_SAR11_Ia.1.I_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.1.I_bin_SCG", "Genus" )
pangenome_SAR11_Ia.1.I_SCG=subset(pangenome_SAR11_Ia.1.I_SCG, Genus=="Ia.1.I")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.1.I/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.1.I/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.1.I_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.1.I_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.1.I[nrow(SumsIa.1.I) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.1.I)
##update the name of the row in the template based on the genome name
row.names(SumsIa.1.I)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.1.I_v2=SumsIa.1.I[-1,]

write.csv(SumsIa.1.I_v2, "SumsIa.1.I_v2.csv")


```



##Ib.1
```{r}
###read in the pangenome files from the earlier analyses 
pangenome_SAR11

##select only the single copy core genes for the genera and then select only a few columns to keep from the datatable

pangenome_SAR11_Ib.1_SCG=subset(pangenome_SAR11, Ib.1_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ib.1_SCG=pangenome_SAR11_Ib.1_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ib.1_bin_SCG", "Genus" )
pangenome_SAR11_Ib.1_SCG=subset(pangenome_SAR11_Ib.1_SCG, Genus=="Ib.1")


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ib.1/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ib.1/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ib.1_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ib.1_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIb.1[nrow(SumsIb.1) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIb.1)
##update the name of the row in the template based on the genome name
row.names(SumsIb.1)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIb.1_v2=SumsIb.1[-1,]

write.csv(SumsIb.1_v2, "SumsIb.1_v2.csv")


```

##Ia.4.1483- note because there is only a single representative of this genus, I will use the single core copy genes of a its close relative (Ia.4.II) that are present within the Ia.4.1483 to define the core genes of this group for read recruitment
```{r}
pangenome_SAR11_Ia.4.1483_SCG=subset(pangenome_SAR11, Ia.4.1483_bin_SCG=="present_single_copy")
pangenome_SAR11_Ia.4.1483_SCG=subset(pangenome_SAR11_Ia.4.1483_SCG, Ia.4.II_bin_SCG=="Core_single_copy")
pangenome_SAR11_Ia.4.1483_SCG=subset(pangenome_SAR11_Ia.4.1483_SCG, Genus=="Ia.4.1483")
pangenome_SAR11_Ia.4.1483_SCG=pangenome_SAR11_Ia.4.1483_SCG%>%dplyr::select("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.1483_bin_SCG", "Genus" )


###read in each of the read recruitment data files for the genera and set up a loop to read through the read recruitment files 
fileNames <- list.files(path="SAR11_Read_Recruitment_By_Genus/Ia.4.1483/", pattern="*.txt", full.names=FALSE, recursive=FALSE)
fileNames

for (fileName in fileNames) {
  # read original data:
  sample <- read.delim(paste0("SAR11_Read_Recruitment_By_Genus/Ia.4.1483/",fileName))

  # create new data based on contents of original file:
  #join the information from the pangenome with the read recruitment data to only get read recruitment data for the single-copy core genes
sample_SCG=inner_join(pangenome_SAR11_Ia.4.1483_SCG,sample, by="gene_callers_id" )
sample_SCG=column_to_rownames(sample_SCG, "gene_callers_id")
sample_SCG_v2 <- sample_SCG[!names(sample_SCG) %in% c("unique_id", "gene_cluster_id", "gene_callers_id","Ia.4.1483_bin_SCG", "Genus" )]
#so now we have just the read recruitment data per sample per gene caller id that is from the single-copy core genes and we add the sum of the read recruitment to our template for each genome in the loop
SumsIa.4.1483[nrow(SumsIa.4.1483) + 1,] <- colSums(sample_SCG_v2)
hh=dim(SumsIa.4.1483)
##update the name of the row in the template based on the genome name
row.names(SumsIa.4.1483)[hh[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName) 
##add the read recruitment data from the single copy core genes to a sample and genome specific file to save (all the data here)
sample_SCG_v2[nrow(sample_SCG_v2) + 1,] <- colSums(sample_SCG_v2)
gg=dim(sample_SCG_v2)
row.names(sample_SCG_v2)[gg[1]] <- gsub("gene_non_outlier_coverages.txt", "Sum",fileName)
print(sample_SCG_v2)
csvfile<-paste0("SCG_",fileName,".csv")
name<-paste0("SCG_",fileName)
as.data.frame(assign(name, sample_SCG_v2))
write.csv(sample_SCG_v2,csvfile)
}
#save the summed data :)
SumsIa.4.1483_v2=SumsIa.4.1483[-1,]

write.csv(SumsIa.4.1483_v2, "SumsIa.4.1483_v2.csv")


```


Before we bring all the genus-level read recruitment data together, correct for detection


```{r}

test=rbind(SumsIa.4.1483_v2, SumsIa.4.II_v2, SumsIa.4.RS39_v2, SumsIa.3.V_v2, SumsIa.3.VI_v2, SumsIa.3.IV_v2, SumsIa.3.I_v2, SumsIa.3.II_v2, SumsIa.3.III_v2, SumsIa.1.I_v2, SumsIb.1_v2)
test$genomes=rownames(test)

test3=test %>% separate(genomes, c('genomes', 'delete'), sep='-')%>%dplyr::select(!"delete")

Taxonomy=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Taxonomy_Genus_genomes=Taxonomy%>%rename_all(recode,"samples"="genomes")%>%dplyr::select("genomes", "Genus")
test3=left_join(test3, Taxonomy_Genus_genomes, by="genomes")

test3_genus=test3%>%unite("Merged", c(genomes,Genus), sep = "-")

test3_genus=as.data.frame(test3_genus)

test3_genus_genus <- data.frame(test3_genus[,-49], row.names=test3_genus[,49])


test3_genus_genus2<- test3_genus_genus  %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(test3_genus_genus2)

test3_genus_genus2=test3_genus_genus2%>%dplyr::rename("q2q3" = "value")
test3_genus_genus2=test3_genus_genus2%>%unite("Merged", c(rowname,colname), sep = "__")

detection=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_across_samples/detection.txt")
names(detection)

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Taxonomy_Genus_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "Genus")
names(Taxonomy_Genus_bins)

detection_genus=left_join(detection, Taxonomy_Genus_bins, by="bins")

detection_genus=detection_genus%>%unite("Merged", c(bins,Genus), sep = "-")

detection_genus=as.data.frame(detection_genus)

detection_genus <- data.frame(detection_genus[,-1], row.names=detection_genus[,1])


detection_genus2<- detection_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_genus2)

detection_genus3=detection_genus2
names(detection_genus3)
detection_genus3=detection_genus3%>%dplyr::rename("genome" = "rowname", "sample"="colname", "detection"="value")
detection_genus3=detection_genus3%>%unite("Merged", c(genome,sample), sep = "__")

test=left_join(test3_genus_genus2,detection_genus3, by="Merged")

test$q2q3_normalized=test$q2q3

test$q2q3_normalized[test$detection < 0.25] <- 0

test2=test%>%dplyr::select(Merged, q2q3_normalized)

test2=test2%>%separate(Merged, c('bins', 'sample'), sep='__')
test2=test2%>%separate(bins, c('genome', 'genus'), sep='-')

test3=test2%>%dplyr::group_by(genus,sample)%>%dplyr::summarise(sum=sum(q2q3_normalized))

###remove the genomes that aren't detected more than once 
#test3=test3%>%filter(genus !="Ia.1.I" & genus !="Ia.3.I" & genus !="Ia.3.IV")

test4=reshape2::dcast(test3, genus~sample)

test.rownames <- data.frame(test4[,-1], row.names=test4[,1])


freqs2 <- apply(test.rownames, 2, function(i) i/sum(i))

freqs3=freqs2*100
freqs3=as.data.frame(freqs3)
write.csv(freqs3, "output/SAR11_KByT_aloha_relative_read_recruit_corebygenera_normalized_oct.csv")


###remove the genera that aren't detected more than once 
row.names.remove <- c("Ia.1.I", "Ia.3.I","Ia.3.IV")

freqs_norare=freqs3[!(row.names(freqs3) %in% row.names.remove), ]
write.csv(freqs_norare, "output/SAR11_KByT_aloha_relative_read_recruit_corebygenera_normalized_oct_norare.csv")
 


t_freqs3 <- data.table::transpose(freqs3)

# get row and colnames in order
colnames(t_freqs3) <- rownames(freqs3)
rownames(t_freqs3) <- colnames(freqs3)

t_freqs3$Metagenome_ID=rownames(t_freqs3)


Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
Enviro2=Enviro%>%tidyr::drop_na(Metagenome_ID)%>%dplyr::select(Metagenome_ID, kmeansruns_cat)

MG_abundance=left_join(t_freqs3, Enviro2, by="Metagenome_ID")

write.csv(MG_abundance, "output/normalized_SAR11_genus_read_recruit_supptable_v2.csv")

```




####
Kmeans with environmental data 

```{r}
Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
Enviro2 =Enviro[-1]
Enviro2=Enviro2[,1:9]

iris_new <- Enviro2 %>% mutate_each_(list(~scale(.) %>% as.vector),
                                  vars = c("PRO.mL","SYN.mL", "PEUK.mL", "HBACT.mL", "chla.ug.per.L",  "Silicate", "pH", "SW.Temperature.at.site", "Salinity"))

df=iris_new


tunek <- fpc::kmeansruns(iris_new ,krange = 1:10,criterion = "ch",scaledata=TRUE, iter.max = 100, runs=100) 
tunek$cluster 
p=unlist(tunek)
write.csv(as.data.frame(p) ,"clusters.csv")
tunek$bestk
?kmeansruns()
library(factoextra)
fviz_nbclust(df, kmeans, method = "wss")

p=factoextra::fviz_nbclust(df, kmeans, method = "wss")
plot(p)



Enviro <- read.csv("MetadatawithKmeans_May2024.csv")


Enviro=Enviro%>%dplyr::select("Sample_ID", "Metagenome", "kmeansruns_cat", "kmeansruns","Project","PRO.mL","SYN.mL", "PEUK.mL", "HBACT.mL", "chla.ug.per.L",  "Silicate", "pH", "SW.Temperature.at.site", "Salinity")
#%>%subset(Metagenome!="None")

iris_new <- Enviro %>% mutate_each_(list(~scale(.) %>% as.vector),
                                  vars = c("PRO.mL","SYN.mL", "PEUK.mL", "HBACT.mL", "chla.ug.per.L",  "Silicate", "pH", "SW.Temperature.at.site", "Salinity"))


test=iris_new[,6:14]

mtcars.pca <- prcomp(iris_new[,6:14], center = TRUE,scale. = TRUE)

summary(mtcars.pca)
str(mtcars.pca)


library(ggbiplot)


Metagenome_type <- factor(iris_new$Metagenome, levels=c("None","KByT", "Aloha"))
iris_new$Metagenome=factor(iris_new$Metagenome, levels=c("None", "KByT", "Aloha"))

iris_new$Project=factor(iris_new$Project, levels=c("Aloha", "KByT"))


colorskmeans=c("Offshore"="#ff0000",
"Coastal"="black")


svg("output/newFCM_PCA_Average_kmeans_scaled_May2024.svg", width=10, height=7)

ggbiplot(mtcars.pca, color=as.factor(iris_new$kmeansruns))+
  ggtitle("")+geom_point(aes(color=as.factor(iris_new$kmeansruns),shape=as.factor(iris_new$Project)), size = 3)+ scale_shape_manual(name="Project",values=c(2,1))+ scale_color_manual(name="Clustering", values= c("#ff0000","black"))+
  theme(legend.position = "bottom")+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +theme(legend.position = "bottom") +theme(legend.position = "bottom") +xlim(c(-2,4))
dev.off()
ggsave("output/PCA_Average_kmeans_scaled_May2024.pdf", width=10, height=7)

```




#####

```{r}
#https://opendata.hawaii.gov/mk/dataset/coastline1
library(geojsonio)
require(sf)
#library(rgdal)


map <- read_sf("Coastline.geojson")

sites <- sf::st_as_sf(data.frame(longitude = c(-158.0), latitude = c(22.45)), coords = c("longitude", "latitude"), crs = 4326, 
agr = "constant")

#library(ggsn)


##full islands 
svg("Hawaii_Islands_Oahu_blue.svg", width=2.5, height=2.5 )
hawaii  <- ggplot(data = map) +
     geom_sf(fill = "grey") +
     geom_sf(data = sites, size = 4, shape = 24, fill = "#7d7d7d")+
     coord_sf(xlim = c(-154.5, -160.5), ylim = c(18.4, 22.74), expand = FALSE) +
     theme(panel.grid.major = element_blank(), panel.background = element_rect(fill = "#FFFFFF"), 
         panel.border = element_rect(fill = NA))+scale_x_discrete(position = "top")
hawaii
dev.off()


#svg("hawaii_zoom.svg", width=3, height=2)
##without big island 
hawaii_zoom  <- ggplot(data = map) +
     geom_sf(fill = "#FFFFFF") +
     geom_sf(data = sites, size = 2, shape = 24, fill = "#6495ED")+
     coord_sf(xlim = c(-156.0, -158.5), ylim = c(20.4, 22.54), expand = FALSE)+
     theme(panel.grid.major = element_blank(), panel.background = element_rect(fill = "#FFFFFF"), 
         panel.border = element_rect(fill = NA))+theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
hawaii_zoom
#dev.off()


svg("hawaii_zoom.svg", width=3, height=2)
hawaii_zoom  <- ggplot(data = map) +
     geom_sf(fill = "grey") +
     geom_sf(data = sites, size = 2, shape = 24, fill = "#6495ED")+
     coord_sf(xlim = c(-157.7, -157.9), ylim = c(21.4, 21.54), expand = FALSE)+
     theme(panel.grid.major = element_blank(), panel.background = element_rect(fill = "#FFFFFF"), 
         panel.border = element_rect(fill = NA))+theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
hawaii_zoom
dev.off()



```



##https://mareds.github.io/r_course/ex_case_study_1_solutions.html

```{r}

library(tidyverse)
library(lubridate)
library(reshape2)
library(MBA)
library(mgcv)
library(ggplot2)

ODV_colours <- c("#feb483", "#d31f2a", "#ffc000", "#27ab19", "#0db5e6", "#7139fe", "#d16cfa")

###the following can be repeated with all HPLC pigments

Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)


south=subset(Enviro,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1" | Site=="NTO1" | Site=="AR" | Site=="NR2" | Site=="CB" | Site=="AR" | Site=="NB" )
names(south)

south2=south%>%dplyr::select(latitude,longitude,PRO.mL)%>%drop_na()
south3=south2%>%reshape2::dcast(latitude~longitude)
names(south2)


summary(south$PRO.mL)
# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'latitude','longitude'), value.name = 'PRO.mL') 


# Finally we create our gridded result
PRO=ggplot(data = ctd_mba, aes(y = latitude, x =longitude)) +
  geom_raster(aes(fill = PRO.mL)) +
  scale_fill_gradientn(colours = rev(ODV_colours), limits=c(0,250000)) +
  geom_contour(aes(z = PRO.mL), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = PRO.mL),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = longitude,y =latitude),
     color = 'black', size = 0.2,alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "PRO.mL") #+
  #coord_cartesian(expand = 0)+ylim(21.40,21.55)
PRO


ggsave("PRO.mL_ODV_KByT_SB.pdf")




hawaii_zoom  <- ggplot() + geom_raster(data = ctd_mba, aes(fill = PRO.mL, y = latitude, x =longitude))+
  scale_fill_gradientn(colours = rev(ODV_colours), limits=c(0,290000))+
  geom_contour(data = ctd_mba, aes(z = PRO.mL, y = latitude, x =longitude), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(data = ctd_mba, aes(z = PRO.mL, y = latitude, x =longitude),  breaks=NULL, colour = "black", alpha = 0.2)  + coord_equal()+
 geom_point(data = south2, aes( x = longitude,y =latitude),
     color = 'black', size = 1.5,alpha = 0.4, shape = 21, fill='black')+
     geom_sf(data = map, fill = "grey") +
     geom_sf(data = sites, size = 2, shape = 21, fill = "#6495ED")+
     coord_sf(xlim = c(-157.72, -157.87), ylim = c(21.4, 21.54), expand = FALSE)+
     theme(panel.grid.major = element_blank(), panel.background = element_rect(fill = "#FFFFFF"), 
         panel.border = element_rect(fill = NA))+theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
hawaii_zoom
ggsave("color_raster_map_KByT_PRO_scaled.svg", height=5, width=4)


```


```{r}
Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)

dd=Enviro%>%subset(Metagenome!="None")
summary(dd$PRO.mL)
#%>%subset(Metagenome_ID!="KFE395NB")

dd$Metagenome_ID<- factor(dd$Metagenome_ID, level=c("KSe054HP",
"KSe055AR",
"KSe063NB",
"KMa157AR",
"KMa159R8",
"KOc293R8",
"KOc299NB",
"KFe387AR",
"KSe058SB",
"KMa156HP",
"KMa158CB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe386HP",
"KFe389R8",
"KFe395NB","KDe747HP", "KMy817HP",
"KSe062NR",
"KMa165NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KMa163NT",
"KMa162ST",
"KFe393NT",
"KFe416ST",
"KSe060ST",
"KSe061NT","KDe743ST","KMy813ST",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))

citation("MBA")
#colorskmeans=c("Offshore"="#4F0CBA","Coastal"="#77BA0C")

PRO<- ggplot(data=dd, aes(x=as.factor(Metagenome_ID), y=PRO.mL, color=PRO.mL, shape=Project))+geom_point(size=2) 
PRO=PRO + theme(legend.title = element_blank()) + ylab("") +xlab("") + theme(axis.text=element_text(size=6),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="none") + scale_color_gradientn(colours = rev(ODV_colours), limits=c(0,290000)) +scale_shape_manual(values=c(17,19))

PRO
```



##So now  we have the read recruitment for each genus per samples based on core read recruitment and standardized based on detection values

```{r}

freqs3=read.csv("output/SAR11_KByT_aloha_relative_read_recruit_corebygenera_normalized_oct_norare.csv", row.names = 1)

freqs4<- freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(freqs4)
samp.with.rownames3=freqs4




sample_order <- factor(samp.with.rownames3$rowname,level=c("Ib.1","Ia.3.III", "Ia.3.II","Ia.4.RS39","Ia.4.1483","Ia.4.II","Ia.3.V","Ia.3.VI"))


level_order <- factor(samp.with.rownames3$colname,level=c("KSe054HP",
"KSe055AR",
"KSe063NB",
"KMa157AR",
"KMa159R8",
"KOc293R8",
"KOc299NB",
"KFe387AR",
"KSe058SB",
"KMa156HP",
"KMa158CB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe386HP",
"KFe389R8",
"KFe395NB","KDe747HP", "KMy817HP",
"KSe062NR",
"KMa165NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KMa163NT",
"KMa162ST",
"KFe393NT",
"KFe416ST",
"KSe060ST",
"KSe061NT","KDe743ST","KMy813ST",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))

write.csv(samp.with.rownames3, "output/samp.with.rownames3.csv")
samp.with.rownames3[samp.with.rownames3 == 0] <- NA
samp.with.rownames3[samp.with.rownames3<1]<- NA
samp.with.rownames3[is.na(samp.with.rownames3)] <- 0

sapply(samp.with.rownames3, class)
names(samp.with.rownames3)

write.csv(samp.with.rownames3, "output/read_recruitment_proportion_Pelagibacteraceae.csv")



recruit=ggplot(samp.with.rownames3, aes(x = sample_order, y = level_order, fill = value)) +
    geom_tile(aes(fill = as.numeric(value))) + 
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) + scale_fill_continuous(type = "viridis")  +
  scale_fill_gradientn(colours = c("#EBEBEB","#808080","#545454","#2B2B2B", "black"), 
                         breaks=c(0,1,10,25,50, 75),
                         guide = "colorbar", limits=c(0.25,100), na.value = "white") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=5)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=6),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+ coord_flip()

recruit


ggsave("output/test.pdf")

pp <- list(PRO, recruit)
pp
library(ggpubr)

ggarrange(plotlist = pp, labels = c('a', 'b'), nrow = 2,align='v',heights = c(1,2))

ggsave("output/enviro_read_recruit_FMC_Cor_Oct2024_newcolors.svg", height=6, width=6)


```



```{r}
for_test=freqs4
for_test=for_test%>%dplyr::rename("Genus"="rowname", "Metagenome_ID"="colname")

Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
names(Enviro)
for_test=left_join(for_test, Enviro, by="Metagenome_ID")
####Check for each genus
Ia.3.VI=for_test%>%subset(Genus=="Ia.3.VI")
Ia.3.VI$kmeansruns_cat=factor(Ia.3.VI$kmeansruns_cat, levels=c("Coastal", "Offshore"))
table(Ia.3.VI$kmeansruns_cat)
library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.VI)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.3.II=for_test%>%subset(Genus=="Ia.3.II")
Ia.3.II$kmeansruns_cat=factor(Ia.3.II$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.II)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.3.III=for_test%>%subset(Genus=="Ia.3.III")
Ia.3.III$kmeansruns_cat=factor(Ia.3.III$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.III)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))

####Check for each genus
Ia.3.V=for_test%>%subset(Genus=="Ia.3.V")
Ia.3.V$kmeansruns_cat=factor(Ia.3.V$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.V)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))

####Check for each genus
Ia.4.1483=for_test%>%subset(Genus=="Ia.4.1483")
Ia.4.1483$kmeansruns_cat=factor(Ia.4.1483$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.1483)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.4.II=for_test%>%subset(Genus=="Ia.4.II")
Ia.4.II$kmeansruns_cat=factor(Ia.4.II$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.II)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.4.RS39=for_test%>%subset(Genus=="Ia.4.RS39")
Ia.4.RS39$kmeansruns_cat=factor(Ia.4.RS39$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.RS39)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ib.1=for_test%>%subset(Genus=="Ib.1")
Ib.1$kmeansruns_cat=factor(Ib.1$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ib.1)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))
```








```{r}
#### look at all genomes for detection levels
detection=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_across_samples/detection.txt")
names(detection)

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Taxonomy_Genus_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "Genus")
names(Taxonomy_Genus_bins)

detection_genus=left_join(detection, Taxonomy_Genus_bins, by="bins")

detection_genus=detection_genus%>%unite("Merged", c(bins,Genus))

detection_genus=as.data.frame(detection_genus)

detection_genus <- data.frame(detection_genus[,-1], row.names=detection_genus[,1])


detection_genus2<- detection_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_genus2)

detection_genus3=detection_genus2 %>% separate(rowname, c('bins', 'Genus'), sep='_')



detection_genus3=detection_genus3%>%dplyr::rename("Metagenome_ID"="colname")
Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
names(Enviro)
Enviro_MG=Enviro%>%filter(Metagenome=="KByT"| Metagenome=="Aloha")
detection_genus3=left_join(detection_genus3, Enviro_MG, by="Metagenome_ID")

names(detection_genus3)

detection_genus4=detection_genus3%>%filter(value >0.25)%>% group_by(Genus, bins, kmeansruns_cat) %>% tally(sort = TRUE)

detection_genus_dcast=reshape2::dcast(detection_genus4, Genus+bins~kmeansruns_cat)

detection_genus4=detection_genus3


####Check for each genus
Ia.3.VI=detection_genus4%>%subset(Genus=="Ia.3.VI")
Ia.3.VI$kmeansruns_cat=factor(Ia.3.VI$kmeansruns_cat, levels=c("Coastal", "Offshore"))
table(Ia.3.VI$kmeansruns_cat)
library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.VI)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.3.II=detection_genus4%>%subset(Genus=="Ia.3.II")
Ia.3.II$kmeansruns_cat=factor(Ia.3.II$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.II)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.3.III=detection_genus4%>%subset(Genus=="Ia.3.III")
Ia.3.III$kmeansruns_cat=factor(Ia.3.III$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.III)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))

####Check for each genus
Ia.3.V=detection_genus4%>%subset(Genus=="Ia.3.V")
Ia.3.V$kmeansruns_cat=factor(Ia.3.V$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.3.V)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))

####Check for each genus
Ia.4.1483=detection_genus4%>%subset(Genus=="Ia.4.1483")
Ia.4.1483$kmeansruns_cat=factor(Ia.4.1483$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.1483)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.4.II=detection_genus4%>%subset(Genus=="Ia.4.II")
Ia.4.II$kmeansruns_cat=factor(Ia.4.II$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.II)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ia.4.RS39=detection_genus4%>%subset(Genus=="Ia.4.RS39")
Ia.4.RS39$kmeansruns_cat=factor(Ia.4.RS39$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ia.4.RS39)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))



####Check for each genus
Ib.1=detection_genus4%>%subset(Genus=="Ib.1")
Ib.1$kmeansruns_cat=factor(Ib.1$kmeansruns_cat, levels=c("Coastal", "Offshore"))

library("multcomp")
amod <- aov(value~ kmeansruns_cat, data =Ib.1)
amod

g=glht(amod, linfct = mcp(kmeansruns_cat= "Tukey"))
summary(g, test = adjusted("holm"))

```



```{r}
#### look at all genomes for detection levels
detection=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_across_samples/detection.txt")
names(detection)


detection_genus=as.data.frame(detection_genus)

detection_genus <- data.frame(detection[,-1], row.names=detection[,1])


detection_genus2<- detection_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_genus2)


####this one 2 clusters
detection_genus2_level_order <- factor(detection_genus2$colname, level=c("KSe054HP",
"KSe055AR",
"KSe063NB",
"KMa157AR",
"KMa159R8",
"KOc293R8",
"KOc299NB",
"KFe387AR",
"KSe058SB",
"KMa156HP",
"KMa158CB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe386HP",
"KFe389R8",
"KFe395NB","KDe747HP", "KMy817HP",
"KSe062NR",
"KMa165NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KMa163NT",
"KMa162ST",
"KFe393NT",
"KFe416ST",
"KSe060ST",
"KSe061NT","KDe743ST","KMy813ST",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))




detection_genus2_sample_order <- factor(detection_genus2$rowname,level=c("RS40","HIMB2305","HTCC9565","NP1","HTCC1040","HTCC1016","HTCC1002","HTCC1062","HTCC1013","HIMB1527","HIMB1321","HIMB1549","HIMB1505","HIMB4","HIMB1412","HIMB1420","HIMB1427","HIMB1564","HIMB1520","HIMB5","HIMB2187","HIMB2226","HIMB1436","RS39","HIMB1483","HIMB2204","HIMB2215","HIMB2200","HIMB1402","HIMB1437","HIMB1863","HIMB2104","HTCC8051","HTCC9022","HTCC7214","HTCC7211","HTCC7217","FZCC0015","HIMB83","HIMB1456","HIMB2201","HIMB1444","HIMB2250","HIMB1409","HIMB1413","HIMB1709","HIMB1623","HIMB1765","HIMB1485","HIMB1715","HIMB1488","HIMB1506","HIMB2211","HIMB1782","HIMB1701","HIMB1509","HIMB1593","HIMB1685","HIMB1559","HIMB1597","HIMB1587","HIMB1495","HIMB1494","HIMB1490","HIMB1491","HIMB122","HIMB1631","HIMB1577","HIMB140","HIMB1611","HIMB1758","HIMB1430","HIMB1746","HIMB1542","HIMB1695","HIMB1573","HIMB1513","HIMB1521","HIMB1507","HIMB1518","HIMB1770","HIMB1493","HIMB1723","HIMB1662","HIMB1702","HIMB1556","HIMB1641","HIMB1552","HIMB1636","HIMB1526","HIMB1748","HIMB1710"))


svg('output/detection_KByT_Aloha_Oct2024_aboveaquarter_v2.svg', width=8.5, height=14)
detection_SAR11_plot=ggplot(detection_genus2, aes(y = detection_genus2_level_order, x = detection_genus2_sample_order)) + 
      geom_point(aes(size = value, fill = value), alpha = 0.75, shape = 21) + 
       scale_size_continuous(limits = c(0.25, 10), range = c(1,10), breaks = c(0.25,0.5,0.75,1)) +
    scale_fill_gradient(low="#FBFEF9",high="#A63446") + scale_x_discrete() + scale_y_discrete(name="Metagenome", position = "right") + labs(fill = "Detection") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, vjust = 1, 
hjust = -0.05, size=5)) + coord_flip() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 1))
detection_SAR11_plot
ggsave('output/detection_KByT_Aloha_Oct2024_aboveaquarter_v2.pdf', width=7, height=14)
dev.off()

library(patchwork)


kk+detection_SAR11_plot
ggsave('output/detection_KByT_Aloha_Oct2024_tree.svg',width=19, height=14)
```


###Look more closely at the distribution of Ia.3.V genomes 
```{r}
#### look at all genomes for detection levels
detection=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_across_samples/detection.txt")
names(detection)

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Taxonomy_Genus_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "Genus")
names(Taxonomy_Genus_bins)

detection_genus=left_join(detection, Taxonomy_Genus_bins, by="bins")

detection_genus=detection_genus%>%unite("Merged", c(bins,Genus))

detection_genus=as.data.frame(detection_genus)

detection_genus <- data.frame(detection_genus[,-1], row.names=detection_genus[,1])


detection_genus2<- detection_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_genus2)

detection_table=detection_genus2

Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
#Enviro<- read.csv("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_Isolates_Metagenomic_Analyses/working_in_R/MetadatawithKmeans_copy.csv", row.names = 1)

names(Enviro)

detection_genus2<- detection_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_genus2)

detection_genus3=detection_genus2 %>% separate(rowname, c('bins', 'Genus'), sep='_')
names(detection_genus3)
detection_enviro=detection_genus3%>%rename_all(recode,"colname"="Metagenome_ID")
names(Enviro)

detection_enviro$Metagenome_ID
Enviro$Metagenome_ID
detection_enviro=left_join(detection_enviro, Enviro, by="Metagenome_ID")
names(detection_enviro)
detection_enviro2=detection_enviro%>%dplyr::select("Metagenome_ID", "bins", "Genus", "kmeansruns_cat")


#detection_enviro_tally_sample=detection_enviro%>%filter(value <0.25)%>% group_by(Genus, bins) %>% tally(sort = TRUE)
detection_enviro_tally_sample=detection_enviro%>%filter(value >=0.25)%>% group_by(Genus, bins, kmeansruns_cat) %>% tally(sort = TRUE)

detection_enviro_tally_sample_v2<- reshape2::dcast(detection_enviro_tally_sample, bins+Genus~kmeansruns_cat)

prop_detection=detection_enviro_tally_sample_v2
names(prop_detection)
prop_detection$Coastal_prop=(prop_detection$Coastal/21)*100
prop_detection$Offshore_prop=(prop_detection$Offshore/27)*100
write.csv(prop_detection, "output/prop_detection.csv")


test=prop_detection%>%subset(Genus=="Ia.3.V")
test$more_coastal=""
test$more_coastal[test$bins=="HIMB83"]="Yes"
test$more_coastal[test$bins=="FZCC0015"]="Yes"
test$more_coastal[test$bins=="HIMB1456"]="Yes"
test$more_coastal[test$bins=="HIMB2250"]="Yes"


```


##lets look at read recruitment at the genome level

```{r}


#### look at all genomes for read recruiment levels
meanq2q3=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/SAR11_Oct2024-by-contig-SUMMARY/bins_across_samples/mean_coverage_Q2Q3.txt")
names(detection)

meanq2q3_genus=as.data.frame(meanq2q3)

meanq2q3_genus <- data.frame(meanq2q3_genus[,-1], row.names=meanq2q3_genus[,1])


meanq2q3_genus2<- meanq2q3_genus %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(meanq2q3_genus2)




meanq2q3_genus2=meanq2q3_genus2%>%dplyr::rename("q2q3" = "value")
detection_genus3=detection_genus2
names(detection_genus3)
detection_genus3=detection_genus3%>%dplyr::rename("genome_genus" = "rowname", "sample"="colname")

q2q3_detection=cbind(meanq2q3_genus2, detection_genus3)

q2q3_detection$q2q3_normalized=q2q3_detection$q2q3

q2q3_detection$q2q3_normalized[q2q3_detection$value < 0.25] <- 0

q2q3_detection2=q2q3_detection%>%dplyr::select(rowname, colname, q2q3_normalized)

test=reshape2::dcast(q2q3_detection2, rowname ~colname)

test.rownames <- data.frame(test[,-1], row.names=test[,1])

freqs2 <- apply(test.rownames, 2, function(i) i/sum(i))

freqs3=freqs2*100
freqs3=as.data.frame(freqs3)


read_recruit<-freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)


read_recruit=read_recruit%>%dplyr::rename("Metagenome_ID"="colname")
Enviro <- read.csv("MetadatawithKmeans_May2024.csv", row.names = 1)
names(Enviro)
Enviro_MG=Enviro%>%filter(Metagenome=="KByT"| Metagenome=="Aloha")
read_recruit=left_join(read_recruit, Enviro_MG, by="Metagenome_ID")

Taxonomy=read.delim("Genera_genome_info_layer_Oct2024.txt")
names(Taxonomy)

Taxonomy_Genus_bins=Taxonomy%>%rename_all(recode,"samples"="rowname")%>%dplyr::select("rowname", "Genus")
names(Taxonomy_Genus_bins)



read_recruit=left_join(read_recruit, Taxonomy_Genus_bins, by="rowname")

read_recruit_order <- factor(read_recruit$rowname,level=c("RS40","HIMB2305","HTCC9565","NP1","HTCC1040","HTCC1016","HTCC1002","HTCC1062","HTCC1013","HIMB1527","HIMB1321","HIMB1549","HIMB1505","HIMB4","HIMB1412","HIMB1420","HIMB1427","HIMB1564","HIMB1520","HIMB5","HIMB2187","HIMB2226","HIMB1436","RS39","HIMB1483","HIMB2204","HIMB2215","HIMB2200","HIMB1402","HIMB1437","HIMB1863","HIMB2104","HTCC8051","HTCC9022","HTCC7214","HTCC7211","HTCC7217","FZCC0015","HIMB83","HIMB1456","HIMB2201","HIMB1444","HIMB2250","HIMB1409","HIMB1413","HIMB1709","HIMB1623","HIMB1765","HIMB1485","HIMB1715","HIMB1488","HIMB1506","HIMB2211","HIMB1782","HIMB1701","HIMB1509","HIMB1593","HIMB1685","HIMB1559","HIMB1597","HIMB1587","HIMB1495","HIMB1494","HIMB1490","HIMB1491","HIMB122","HIMB1631","HIMB1577","HIMB140","HIMB1611","HIMB1758","HIMB1430","HIMB1746","HIMB1542","HIMB1695","HIMB1573","HIMB1513","HIMB1521","HIMB1507","HIMB1518","HIMB1770","HIMB1493","HIMB1723","HIMB1662","HIMB1702","HIMB1556","HIMB1641","HIMB1552","HIMB1636","HIMB1526","HIMB1748","HIMB1710"))



#read_recruit3=read_recruit2%>%filter(value <1)

read_recruit2=read_recruit
read_recruit2[read_recruit2 <0.25] <- NA
#halp4[halp4<1]<- NA


library("scales")

svg('q2q3_KByT_Aloha_Oct2024.svg', width=8.5, height=11)
geom.text.size = 3
recruit=ggplot(read_recruit2, aes(x = read_recruit_order, y = Metagenome_ID, fill = value)) +
    geom_tile(aes(fill = value)) + geom_text(aes(label = round(value, 0)), size=geom.text.size) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) + 
  scale_fill_gradientn(colours = c("yellow","orange","red"), 
                         values = rescale(c(0,5,50)),
                         guide = "colorbar", limits=c(0,50),  na.value = "white")  + scale_x_discrete(name="Genome") + scale_y_discrete(name="Metagenome", position = "left") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=5)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=9),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_text(size=9,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+ facet_wrap(~kmeansruns_cat,scales = "free_x")+coord_flip()


recruit
ggsave('q2q3_KByT_Aloha_Oct2024.pdf', width=7, height=10)
dev.off()

kk+recruit
ggsave('recruit_KByT_Aloha_Oct2024_tree.pdf',width=19, height=14)
ggsave('recruit_KByT_Aloha_Oct2024_tree.svg',width=19, height=14)

```


##Need some abundance stats per Environment

```{r}

read_recruit_stats=read_recruit
read_recruit_stats2=read_recruit_stats%>%group_by(rowname,Genus, kmeansruns_cat)%>%summarise(mean=round(mean(value),1), sd=round(sd(value),1))

coastaL_Ia.3.V=read_recruit_stats%>%subset(rowname=="HIMB83"| rowname=="HIMB2250" | rowname=="FZCC0015"| rowname=="HIMB1456")%>%group_by(Genus, kmeansruns_cat)%>%summarise(mean=round(mean(value),1), sd=round(sd(value),1))

HIMB1412=read_recruit_stats%>%group_by(rowname,kmeansruns_cat)%>%summarise(mean=round(mean(value),1), sd=round(sd(value),1))

```




###

##
To characterize ecologically-relevant genes that may support coastal and offshore habitat-preferences of Pelagibacteraceae in this system, we used an enrichment analysis to examine nutrient acquisition and cellular stress metabolic traits that were differentially distributed in genomes from either coastal or offshore genera. Using the output from the functional enrichment analysis, genes involved in central and alternative carbon metabolisms, amino acid, vitamin, and cofactor biosynthesis, and genes to cope with nutrient, osmotic, and oxidative stress were examined across the SAR11 isolate genomes.


A subset of genes identified as having distinct distributions with coastal and offshore Pelagibacteraceae. This subset is reported in Supplementary file 2f. Distribution of habitat-specific genes involved in nutrient acquisition and utilization and osmotic and oxidative stress across Pelagibacteraceae genera and serves as the basis of the pN/pS analysis in Figure 5.

```{r}
KOFam=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt")
names(KOFam)
COG20=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt")
names(COG20)

##Reorder so that the COG20 and KOFam have similar order
KOFam=KOFam%>%dplyr::select("KOfam",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="KOfam")

COG20=COG20%>%dplyr::select("COG20_FUNCTION",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="COG20_FUNCTION")


###bring in the list of desired genes to examine
set=read.delim("set_of_differentially_distributed_genes.txt")

names(set)

set_KOfam=set%>%subset(source_figure4=="KOfam")
set_COG20=set%>%subset(source_figure4=="COG20_FUNCTION")

set_KOfam2=left_join(set_KOfam, KOFam, by="FUNCTION")

set_COG202=left_join(set_COG20, COG20, by="FUNCTION")

tog_figure5=rbind(set_KOfam2, set_COG202)
names(tog_figure5)
identical(tog_figure5[['accession_check']],tog_figure5[['accession']])



tog_figure=tog_figure5%>%dplyr::arrange(Order)%>%dplyr::select("Category","Metabolism","Specific.function2","Specific.function","FUNCTION","Environment","source_figure4","Present_in_reps","accession","gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I", "Order")%>%rename("Lacunapelagibacter"="p_Ia.4.II", "Semelpelagibacter"="p_Ia.4.1483","Proprepelagibacter"="p_Ia.3.IV", "Atlantikopelagibacter"="p_Ia.3.I","Ampluspelagibacter"="p_Ia.3.VI", "Xanthipelagibacter"="p_Ia.3.V", "Coralipelagibacter"="p_Ia.3.II" , "Littoralipelagibacter"="p_Ia.3.III", "Pelagibacter"="p_Ia.1.I", "Superopelagibacter"="p_Ib.1", "Tantilluspelagibacter"="p_Ia.4.RS39", "Gene name"="Specific.function", "Specific Function Abbreviated"="Specific.function2")

write.csv(tog_figure, "output/Supplementary_file_2e.csv")

###Note that upon looking at the adhesion genes in the pangenome, we established that Ia.3.V did not contain an adhesion gene. Because a single genome shared a GC with other genomes and within those other genomes this GC was annotated as an adhesion gene the functional enrichment analysis also took that annotation. HOWEVER the genes within HIMB83 that shares this GC was not annotated, also the alignment wouldn't suggest a close relationship of this HIMB83 gene with genes from other genomes sharing that GC.

##furthermore,two GCs (GC_00002013 and GC_00001508) hit to EC 4.1.1.9 (malonyl-CoA decarboxylase) however GC_00001508 had equal near equal hit to K01578 and to K11690 and in many tables appears only as K11690. Update your tables to mkae sure these changes are reflected. mylcd is core in Lacunapelagibacter(Ia.4.II), Tantilluspelagibacter (Ia.4.RS39),	Superopelagibacter (Ib.1),	Atlantikopelagibacter (Ia.3.I), and present (0.2857) in Pelagibacter (Ia.1.I).    

#Finally, multiple genes may contribute to a given metabolic category defined here, so a version of this subset of genes collasped per metabolic pathway/complex is also presented in Figure 4. A single gene was chosen to represent the distribution of the pathway for reporting as here
```



The above dataset is then used below:

###Figure 4


```{r}
genotype <- read.csv("Figure4_Oct292024.csv")
names(genotype)
matchcog=genotype%>%dplyr::select("Specific.function", "source_figure4", "Metabolism", "Substrate", "Gene.description", "Order", "Ecological_pattern", "Specific.function2")%>%subset(source_figure4=="COG20_FUNCTION")
cog20_fe=read.delim("functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt")
cog20_fe=cog20_fe%>%dplyr::rename("Gene.description"="COG20_FUNCTION")

matchcog=left_join(matchcog, cog20_fe, by="Gene.description")


matchKOfam=genotype%>%dplyr::select("Specific.function", "source_figure4", "Metabolism", "Substrate", "Gene.description", "Order","Ecological_pattern","Specific.function2")%>%subset(source_figure4=="KOfam")

KOfam_fe=read.delim("functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt")
KOfam_fe=KOfam_fe%>%dplyr::rename("Gene.description"="KOfam")

matchKOfam=left_join(matchKOfam, KOfam_fe, by="Gene.description")

updated_Figure4=rbind(matchcog, matchKOfam)
updated_Figure4=updated_Figure4%>%arrange(Order)
names(updated_Figure4)
updated_Figure4=updated_Figure4%>%dplyr::select("Specific.function2",	"p_Ib.1",	"p_Ia.4.II",	"p_Ia.4.RS39","p_Ia.4.1483",	"p_Ia.3.V",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.III",	"Metabolism",	"Ecological_pattern",	"Substrate",	"Gene.description",	"accession",	"source_figure4",	"Specific.function",	"p_Ia.3.IV","p_Ia.1.I",	"p_Ia.3.I")
                                          
write.csv(updated_Figure4, "Figure4_Oct312024.csv")
###note that adhesion gene for Ia.3.V was reviewed manually and we determined that the adhesion gene was not present in any Ia.3.V genome. This call is an artifact of the enrichment analysis. 


```



```{r}
library(pheatmap)
library(ComplexHeatmap)

?pairwise.adonis2()
########

#genotype <- read.csv("Figure4_April52025_v2.csv", row.names=1)

#genotype <- read.csv("Figure4_June132024.csv", row.names=1)
genotype <- read.csv("Figure4_Oct312024.csv", row.names=1)
unique(genotype$Metabolism)
dim(genotype)

pattern_df = data.frame( "Metabolism" = genotype$Metabolism,"Ecological_pattern"=genotype$Ecological_pattern, "Substrate"=genotype$Substrate)
rownames(pattern_df) = rownames(genotype)

#pattern_df$Evolutionary_pattern <- factor(pattern_df$Evolutionary_pattern, levels = c("monophyletic", "none", "paraphyletic", "polyphyletic"))
pattern_df$Ecological_pattern <- factor(pattern_df$Ecological_pattern, levels = c("none", "Coastal", "Offshore"))
pattern_df$Metabolism <- factor(pattern_df$Metabolism, levels = c("Nitrogen","Phosphorus", "Oxidative","Carbon", "Metals","Osmotic", "Sulfur", "Adhesion"))
pattern_df$Substrate<- factor(pattern_df$Substrate, levels = c("Purines","Purine Nucleotides", "Allantoate","Amino acids", "C1-C3",">C4","Glycerophospholipids" ,"None"))


#my_colour = list(
  #  Metabolism=c("Carbon" = '#e6194b' , "Nitrogen"='#3cb44b', "Oxidative"='#ffe119', "Osmotic"='#4363d8', "Phosphorus"='#f58231', "Sulfur"='#911eb4', "Metals"='#46f0f0'),
  #      Ecological_pattern = c("none" = "#c8c4b7ff", "coastal" = "#f032e6", #"offshore"="#000075")
#)

my_colour = list(
    Metabolism=c("Carbon" = '#e6194b' , "Nitrogen"='#3cb44b', "Oxidative"='#ffe119', "Osmotic"='#4363d8', "Phosphorus"='#f58231', "Metals"='#46f0f0', "Sulfur"="red", "Adhesion"="orange"),
        Ecological_pattern = c("none" = "#c8c4b7ff", "Coastal" = "#f032e6", "Offshore"="#000075"), Substrate = c("Purines"="#228B22","Purine Nucleotides"="#3D550C", "Allantoate"="#21B6A8","Amino acids"="#94C973", "C1-C3"="#5D3FD3",">C4"="#770737", "None"="white", "Glycerophospholipids"="#CF9FFF")
)


svglite("output/Figure_4_draft_Dec202024.svg", height=11, width=8.5)
ComplexHeatmap::pheatmap(genotype[,1:8], cluster_cols = F, cluster_row = F, annotation_row = pattern_df, gaps_col=c(5,8), color = c("white", "#d7d7d7","#7a7a7a" ,"black"),cellwidth=20,cellheight=10,gaps_row=c(21,30, 33, 51),
         breaks = c(0,0.01, 0.15, 0.99, 1), annotation_colors = my_colour, fontsize_row=9, column_names_side = c("top"),row_names_side = c("left"),
                         angle_col = c("90"), border_color=TRUE)

dev.off()

library(svglite)


```


###Updating Supplemental Tables
Supplementary File 2d
Moco Enzymes
```{r}
KOFam=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt")
names(KOFam)
COG20=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt")
names(COG20)

##Reorder so that the COG20 and KOFam have similar order
KOFam=KOFam%>%dplyr::select("KOfam",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="KOfam")
names(KOFam)
COG20=COG20%>%dplyr::select("COG20_FUNCTION",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="COG20_FUNCTION")

names(COG20)

###bring in the list of desired genes to examine
set=read.delim("Moco_enzymes_to_look_for.txt")

set_KOfam=set%>%subset(Source=="KOfam")
set_COG20=set%>%subset(Source=="COG20_FUNCTION")

set_KOfam2=left_join(set_KOfam, KOFam, by="FUNCTION")

set_COG202=left_join(set_COG20, COG20, by="FUNCTION")

tog_figure5=rbind(set_KOfam2, set_COG202)
names(tog_figure5)
identical(tog_figure5[['accession_check']],tog_figure5[['accession']])

names(tog_figure5)

tog_figure=tog_figure5%>%dplyr::arrange(Order)%>%dplyr::select("Type","Category","Gene.name","FUNCTION","Source","Accession","COG","KOfam","EC","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I", "Order")%>%rename("Lacunapelagibacter"="p_Ia.4.II", "Semelpelagibacter"="p_Ia.4.1483","Proprepelagibacter"="p_Ia.3.IV", "Atlantikopelagibacter"="p_Ia.3.I","Ampluspelagibacter"="p_Ia.3.VI", "Xanthipelagibacter"="p_Ia.3.V", "Coralipelagibacter"="p_Ia.3.II" , "Littoralipelagibacter"="p_Ia.3.III", "Pelagibacter"="p_Ia.1.I", "Superopelagibacter"="p_Ib.1", "Tantilluspelagibacter"="p_Ia.4.RS39")

write.csv(tog_figure, "output/Supplemental_2c.csv")
```


###Updating Supplemental Tables
Supplementary File 2e
Purines Enzymes
```{r}
KOFam=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt")
names(KOFam)
COG20=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt")
names(COG20)

##Reorder so that the COG20 and KOFam have similar order
KOFam=KOFam%>%dplyr::select("KOfam",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="KOfam")
names(KOFam)
COG20=COG20%>%dplyr::select("COG20_FUNCTION",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="COG20_FUNCTION")

names(COG20)

###bring in the list of desired genes to examine
set=read.delim("Purine_enzymes_to_look_for.txt")

set_KOfam=set%>%subset(Source=="KOfam")
set_COG20=set%>%subset(Source=="COG20_FUNCTION")

set_KOfam2=left_join(set_KOfam, KOFam, by="FUNCTION")

set_COG202=left_join(set_COG20, COG20, by="FUNCTION")

tog_figure5=rbind(set_KOfam2, set_COG202)
names(tog_figure5)
identical(tog_figure5[['accession_check']],tog_figure5[['accession']])

names(tog_figure5)

tog_figure=tog_figure5%>%dplyr::arrange(Order)%>%dplyr::select("Category","FUNCTION","Source","Accession","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I", "Order")%>%rename("Lacunapelagibacter"="p_Ia.4.II", "Semelpelagibacter"="p_Ia.4.1483","Proprepelagibacter"="p_Ia.3.IV", "Atlantikopelagibacter"="p_Ia.3.I","Ampluspelagibacter"="p_Ia.3.VI", "Xanthipelagibacter"="p_Ia.3.V", "Coralipelagibacter"="p_Ia.3.II" , "Littoralipelagibacter"="p_Ia.3.III", "Pelagibacter"="p_Ia.1.I", "Superopelagibacter"="p_Ib.1", "Tantilluspelagibacter"="p_Ia.4.RS39")

write.csv(tog_figure, "output/Supplemental_2d_purines.csv")
```




###Updating Supplemental Tables
Supplementary File 2g
Distribution of genes involved in carbon metabolism, amino acid, vitamin, and cofactor synthesis, methionine salvage, nutrient acquisition and utilization, and adhesion, type IV pili, type II secretory genes, and sporulation.
```{r}
KOFam=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11-Oct2024-KOfam-functional-enrichment.txt")
names(KOFam)
COG20=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/functional_enrichment/SAR11_Oct2024-COG20_FUNCTION-functional-enrichment.txt")
names(COG20)

##Reorder so that the COG20 and KOFam have similar order
KOFam=KOFam%>%dplyr::select("KOfam",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="KOfam")
names(KOFam)
COG20=COG20%>%dplyr::select("COG20_FUNCTION",	"enrichment_score",	"unadjusted_p_value",	"adjusted_q_value",	"associated_groups",	"accession",	"gene_clusters_ids","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I")%>%rename("FUNCTION"="COG20_FUNCTION")

names(COG20)

###bring in the list of desired genes to examine
set=read.delim("all_other_fuctions_setup_2.txt")
names(set)
set_KOfam=set%>%subset(source=="KOfam")
set_COG20=set%>%subset(source=="COG20_FUNCTION")

set_KOfam2=left_join(set_KOfam, KOFam, by="FUNCTION")

set_COG202=left_join(set_COG20, COG20, by="FUNCTION")

tog_figure5=rbind(set_KOfam2, set_COG202)
names(tog_figure5)
identical(tog_figure5[['accession_check']],tog_figure5[['accession']])

names(tog_figure5)
test=tog_figure5%>% dplyr::filter(if_any(c("enrichment_score"), is.na))
#write.csv(test, "output/need_to_update_v2.csv")


tog_figure=tog_figure5%>%dplyr::arrange(Order)%>%dplyr::select("Category","Metabolism","Specific.function","FUNCTION","source","accession","p_Ia.4.II",	"p_Ia.3.III",	"p_Ia.3.VI",	"p_Ia.3.II",	"p_Ia.3.IV",	"p_Ia.3.V",	"p_Ia.4.1483",	"p_Ia.1.I",	"p_Ia.4.RS39",	"p_Ib.1",	"p_Ia.3.I", "Order")%>%rename("Lacunapelagibacter"="p_Ia.4.II", "Semelpelagibacter"="p_Ia.4.1483","Proprepelagibacter"="p_Ia.3.IV", "Atlantikopelagibacter"="p_Ia.3.I","Ampluspelagibacter"="p_Ia.3.VI", "Xanthipelagibacter"="p_Ia.3.V", "Coralipelagibacter"="p_Ia.3.II" , "Littoralipelagibacter"="p_Ia.3.III", "Pelagibacter"="p_Ia.1.I", "Superopelagibacter"="p_Ib.1", "Tantilluspelagibacter"="p_Ia.4.RS39", "Specific Function"="Specific.function")



write.csv(tog_figure, "output/Supplemental_2f_all.csv")
```
