---
title: "Anvio_Code_SAR11_Submission"
author: "Sarah Tucker"
date: "8/6/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{python}

source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate anvio-7

```

```{python}
anvi-gen-contigs-database -f final_new_references_SAR11_v8.fa -o /tank/tucker_data/New_Genomes_Depth/SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db -n "KByT Aloha BATS Depth New Genomes" --num-threads 30


anvi-run-hmms -c /tank/tucker_data/New_Genomes_Depth/SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db --num-threads 30

anvi-run-ncbi-cogs --cog-data-dir /tank/tucker_data/ANVIO_NCBI_COGS_2020 -c /tank/tucker_data/New_Genomes_Depth/SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db --num-threads 30

bowtie2-build final_new_references_SAR11_v8.fa SAR11-KbyTAlohaBats-New --threads 30


while IFS=$'\t' read -r sample f1 f2; do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 30 \
            -x SAR11-KbyTAlohaBats-New \
            -1 $f1 \
            -2 $f2 \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done < mendes_depth_KByT.txt


for sample in `awk '{print $1}' mendes_depth_KByT.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 25 \
                 -o $sample
done



anvi-merge */PROFILE.db -o SAR11-KbyT-AlohaBATS-newgenomes-MERGED -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db

```

```{python}


for split_name in `sqlite3 SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="_"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > SAR11-KbyT-AlohaBATS-newgenomes-GENOME-COLLECTION.txt


anvi-import-collection SAR11-KbyT-AlohaBATS-newgenomes-GENOME-COLLECTION.txt -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db -p SAR11-KbyT-AlohaBATS-newgenomes-MERGED/PROFILE.db -C Genomes

anvi-summarize -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db -p SAR11-KbyT-AlohaBATS-newgenomes-MERGED/PROFILE.db -C Genomes --init-gene-coverages -o SAR11-KbyT-Alohadepth-newgenomes-SUMMARY


```


```{python}
anvi-run-kegg-kofams -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db --kegg-data-dir /tank/tucker_data/KEGG


anvi-estimate-metabolism -c SAR11-KbyT-AlohaBATS-newgenomes-CONTIGS.db --kegg-data-dir /tank/tucker_data/KEGG -p SAR11-KbyT-AlohaBATS-newgenomes-MERGED/PROFILE.db -C Genomes --num-threads 25


```

```{python}
anvi-gen-genomes-storage -i just_Ia_revised_internal_genomes_v2.txt -o just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db

###between 10 and 2
anvi-pan-genome -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 6 --project-name just-Ia-revised-v2-Aloha_KByT-depth-PAN --num-threads 30

anvi-display-pan -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db

anvi-summarize -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db -C Ia_single_copy_v2 --init-gene-coverages -o just-Ia-revised-v2-Aloha_KByT-SUMMARY


anvi-compute-genome-similarity --internal-genomes just_Ia_revised_internal_genomes.txt --program fastANI --output-dir Ia-revised-ANI --num-threads 25 -p just-Ia-revised-Aloha_KByT-depth-PAN/just-Ia-revised-Aloha_KByT-depth-PAN-PAN.db

```


```{python}


anvi-compute-functional-enrichment-in-pan -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db -o Ia-revised-v2-Aloha_KByT-depth-enriched-KEGG_Module-functional-subclades.txt --category-variable Subclades --annotation-source KEGG_Module --functional-occurrence-table-output Ia-revised-v2-Aloha_KByT-depth-enriched-KEGG_Module-FUNC_OCCURRENCE.TXT

#add subclade ID
anvi-import-misc-data updated_Aloha_KByT_genome_info_layer.txt -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db --target-data-table layers



anvi-compute-functional-enrichment-in-pan -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db -o Ia-revised-v2-Aloha_KByT-depth-enriched-COG20-functional-subclades.txt --category-variable Subclades --annotation-source COG20_FUNCTION --functional-occurrence-table-output Ia-revised-v2-Aloha_KByT-depth-enriched-COG20-FUNC_OCCURRENCE.TXT



anvi-compute-functional-enrichment-in-pan -p just-Ia-revised-v2-Aloha_KByT-depth-PAN/just-Ia-revised-v2-Aloha_KByT-depth-PAN-PAN.db -g just-Ia-revised-v2-Aloha_KByT-depth-PAN-GENOMES.db -o Ia-revised-v2-Aloha_KByT-depth-enriched-KOFAMS-functional-subclades.txt --category-variable Subclades --annotation-source KOfam --functional-occurrence-table-output Ia-revised-v2-Aloha_KByT-depth-enriched-KOFAMS-FUNC_OCCURRENCE.TXT

```



```{python}

####just Ia.3.V

anvi-gen-genomes-storage -i just_Ia3V_internal_genomes.txt -o just-Ia3V-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3V-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name v2-just-Ia3V-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p v2-just-Ia3V-Aloha_KByT-depth-PAN/v2-just-Ia3V-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3V-Aloha_KByT-depth-PAN-GENOMES.db

anvi-summarize -p v2-just-Ia3V-Aloha_KByT-depth-PAN/v2-just-Ia3V-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3V-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3V_single_copy --init-gene-coverages -o just-Ia3V-Aloha_KByT-SUMMARY

anvi-display-functions -i just_Ia3V_internal_genomes.txt --annotation-source COG20_FUNCTION --profile-db Ia3V-COG20Function-PROFILE.db

```


```{python}

###just Ia.3.III

anvi-gen-genomes-storage -i just_Ia3III_internal_genomes.txt -o just-Ia3III-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3III-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia3III-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just-Ia3III-Aloha_KByT-depth-PAN/just-Ia3III-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3III-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia3III-Aloha_KByT-depth-PAN/just-Ia3III-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3III-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3III_single_copy --init-gene-coverages -o just-Ia3III-Aloha_KByT-SUMMARY
```

```{python}

####just Ia.3.II

anvi-gen-genomes-storage -i just_Ia3II_internal_genomes.txt -o just-Ia3II-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3II-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia3II-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just-Ia3II-Aloha_KByT-depth-PAN/just-Ia3II-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3II-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia3II-Aloha_KByT-depth-PAN/just-Ia3II-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3II-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3II_single_copy --init-gene-coverages -o just-Ia3II-Aloha_KByT-SUMMARY
```

```{python}


#### just Ia.3.VI

anvi-gen-genomes-storage -i just_Ia3VI_internal_genomes.txt -o just-Ia3VI-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3VI-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia3VI-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just-Ia3VI-Aloha_KByT-depth-PAN/just-Ia3VI-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3VI-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia3VI-Aloha_KByT-depth-PAN/just-Ia3VI-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3VI-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3VI_single_copy --init-gene-coverages -o just-Ia3VI-Aloha_KByT-SUMMARY
```


```{python}


#### just Ia.3.VI (no HIMB1602)

anvi-gen-genomes-storage -i just_Ia3VI-no1602_internal_genomes.txt -o  just-Ia3VI-no1602-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3VI-no1602-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just_Ia3VI-no1602-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just_Ia3VI-no1602-Aloha_KByT-depth-PAN/just_Ia3VI-no1602-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3VI-no1602-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just_Ia3VI-no1602-Aloha_KByT-depth-PAN/just_Ia3VI-no1602-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3VI-no1602-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3VI_single_copy --init-gene-coverages -o just-Ia3VI-no1602-Aloha_KByT-SUMMARY

anvi-display-functions -i just_Ia3VI-no1602_internal_genomes.txt --annotation-source KEGG_Module --profile-db KEGG_Module-PROFILE.db

anvi-migrate /tank/tucker_data/New_Genomes_Depth/SAR11-KbyT-AlohaBATS-newgenomes-MERGED/PROFILE.db --migrate-dbs-safely

anvi-display-functions -i just_Ia3VI-no1602_internal_genomes.txt --annotation-source COG20_FUNCTION --profile-db COG20Function-PROFILE.db

```



```{python}


##### just Ia.4.II

anvi-gen-genomes-storage -i just_Ia4II_internal_genomes.txt -o just-Ia4II-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia4II-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia4II-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just-Ia4II-Aloha_KByT-depth-PAN/just-Ia4II-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4II-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia4II-Aloha_KByT-depth-PAN/just-Ia4II-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4II-Aloha_KByT-depth-PAN-GENOMES.db -C Ia4II_single_copy --init-gene-coverages -o just-Ia4II-Aloha_KByT-SUMMARY

```

```{python}

#### just Ia.4.rs39

anvi-gen-genomes-storage -i just_Ia4rs39_internal_genomes.txt -o just-Ia4rs39-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia4rs39-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia4rs39-Aloha_KByT-depth-PAN --num-threads 20

anvi-display-pan -p just-Ia4rs39-Aloha_KByT-depth-PAN/just-Ia4rs39-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4rs39-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia4rs39-Aloha_KByT-depth-PAN/just-Ia4rs39-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4rs39-Aloha_KByT-depth-PAN-GENOMES.db -C Ia4rs39_single_copy --init-gene-coverages -o just-Ia4rs39-Aloha_KByT-SUMMARY
```


```{python}
#### just Ia1

anvi-gen-genomes-storage -i just_IaI_internal_genomes.txt -o just-IaI-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-IaI-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-IaI-Aloha_KByT-depth-PAN --num-threads 30

anvi-display-pan -p just-IaI-Aloha_KByT-depth-PAN/just-IaI-Aloha_KByT-depth-PAN-PAN.db -g just-IaI-Aloha_KByT-depth-PAN-GENOMES.db

anvi-summarize -p just-IaI-Aloha_KByT-depth-PAN/just-IaI-Aloha_KByT-depth-PAN-PAN.db -g just-IaI-Aloha_KByT-depth-PAN-GENOMES.db -C Ia1_single_copy --init-gene-coverages -o just-Ia1-Aloha_KByT-SUMMARY
```


```{python}

#### just Ia31

anvi-gen-genomes-storage -i just_Ia3I_internal_genomes.txt -o just-Ia3I-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3I-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia3I-Aloha_KByT-depth-PAN --num-threads 30

anvi-display-pan -p just-Ia3I-Aloha_KByT-depth-PAN/just-Ia3I-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3I-Aloha_KByT-depth-PAN-GENOMES.db

anvi-summarize -p just-Ia3I-Aloha_KByT-depth-PAN/just-Ia3I-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3I-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3I_single_copy --init-gene-coverages -o just-Ia3-Aloha_KByT-SUMMARY
```


```{python}
#### just Ia3IV

anvi-gen-genomes-storage -i just_Ia3IV_internal_genomes.txt -o just-Ia3IV-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia3IV-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia3IV-Aloha_KByT-depth-PAN --num-threads 30

anvi-display-pan -p just-Ia3IV-Aloha_KByT-depth-PAN/just-Ia3IV-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3IV-Aloha_KByT-depth-PAN-GENOMES.db

anvi-summarize -p just-Ia3IV-Aloha_KByT-depth-PAN/just-Ia3IV-Aloha_KByT-depth-PAN-PAN.db -g just-Ia3IV-Aloha_KByT-depth-PAN-GENOMES.db -C Ia3IV_single_copy --init-gene-coverages -o just-Ia3IV-Aloha_KByT-SUMMARY
```


```{python}



#### just Ia.4.II_Ia.4.1483

anvi-gen-genomes-storage -i just_Ia4II_Ia41483_internal_genomes.txt -o just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-GENOMES.db

anvi-pan-genome -g just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 10 --project-name just-Ia4II-Ia41483-Aloha_KByT-depth-PAN --num-threads 30

anvi-display-pan -p just-Ia4II-Ia41483-Aloha_KByT-depth-PAN/just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-GENOMES.db


anvi-summarize -p just-Ia4II-Ia41483-Aloha_KByT-depth-PAN/just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-PAN.db -g just-Ia4II-Ia41483-Aloha_KByT-depth-PAN-GENOMES.db -C Ia4II_Ia41483_single_copy --init-gene-coverages -o just-Ia4II-Ia41483-Aloha_KByT-SUMMARY

```

