---
title: "pNpS_May2024"
author: "Sarah Tucker"
date: "5/12/2024"
output: html_document
---



####new way of setting up these graphs

##HIMB5-Ia.3.III Coastal

```{r}
rm(list=ls())
enviro_specific=read.csv("figure_5_August42024.csv")
offshore_genes=enviro_specific%>%dplyr::filter(Environment=="Offshore")
Coastal_genes=enviro_specific%>%dplyr::filter(Environment=="Coastal")

```


```{r}


pnps_HIMB5=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/pNpS.txt")
functions_HIMB5=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/HIMB5-functions.txt")
functions_HIMB5=functions_HIMB5%>%filter(source=="COG20_FUNCTION" | source=="KOfam")
length(unique(functions_HIMB5$gene_callers_id))
#we have 1221 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
Coastal_genes_HIMB5=left_join(functions_HIMB5, Coastal_genes, by="accession")
dim(Coastal_genes_HIMB5)
length(unique(Coastal_genes_HIMB5$gene_callers_id))

names(Coastal_genes_HIMB5)
Coastal_genes_HIMB5=Coastal_genes_HIMB5%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Coastal")
dim(Coastal_genes_HIMB5)

Coastal_genes_HIMB5=distinct(Coastal_genes_HIMB5,gene_callers_id, .keep_all=TRUE)
write.csv(Coastal_genes_HIMB5, "Coastal_genes_HIMB5.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB5=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/HIMB5-gene_calls.txt")
length(unique(functions_HIMB5$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB5)
functions_HIMB5_Coastal_specific=left_join(functions_HIMB5,Coastal_genes_HIMB5)

```



##now we need to add in PN/PS
```{r}

functions_HIMB5_Coastal_specific_join=functions_HIMB5_Coastal_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")
HIMB5_pnps_function=left_join(pnps_HIMB5, functions_HIMB5_Coastal_specific_join, by="corresponding_gene_call")
length(unique(HIMB5_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB5_pnps_function$pNpS_gene_reference))
HIMB5_pnps_function[c('pNpS_gene_reference')][sapply(HIMB5_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB5_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB5_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_HIMB5_means.csv")

##
pNpS_within_sample_Quantiles=HIMB5_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_HIMB5_Quantiles.csv")

#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB5_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB5_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)


library(scales)

across_samples_HIMB5<- ggplot(data=HIMB5_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB5
across_samples_HIMB5=across_samples_HIMB5 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB5


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB5_pnps_function%>%dplyr::filter(sample_id=="KDe747HP" | sample_id=="KMy817HP") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB5_pnps_function=left_join(HIMB5_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")


HIMB5_pnps_function

```


```{r}
unique(HIMB5_pnps_function$sample_id)
HIMB5_pnps_function_reporting=HIMB5_pnps_function%>%filter(sample_id=="KMy817HP"| sample_id=="KDe747HP")
sum(is.na(HIMB5_pnps_function_reporting$pNpS_gene_reference))
HIMB5_pnps_function_reporting=HIMB5_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
HIMB5_pnps_function_reporting$log_pnps=log(HIMB5_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(HIMB5_pnps_function_reporting$log_pnps))
which(is.infinite(HIMB5_pnps_function_reporting$log_pnps))
HIMB5_pnps_function_reporting[c('log_pnps')][sapply(HIMB5_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(HIMB5_pnps_function_reporting$log_pnps))
HIMB5_pnps_function_reporting=HIMB5_pnps_function_reporting%>%drop_na(log_pnps)

hist(HIMB5_pnps_function_reporting$log_pnps)

```



```{r}

####let's break out pnps by samples 
backbone_HIMB5_pnps_function=distinct(HIMB5_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB5_pnps_function)
dim(functions_HIMB5_Coastal_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB5_pnps_function=backbone_HIMB5_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB5_pnps_function)


###subset and rename to have sample level information
unique(HIMB5_pnps_function$sample_id)
HIMB5_pnps_function_KDe747HP=HIMB5_pnps_function%>%filter(sample_id=="KDe747HP")
names(HIMB5_pnps_function_KDe747HP)
HIMB5_pnps_function_KDe747HP=HIMB5_pnps_function_KDe747HP%>%dplyr::rename("KDe747HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe747HP_pNpS_gene_reference")


HIMB5_pnps_function_KMy817HP=HIMB5_pnps_function%>%filter(sample_id=="KMy817HP")
HIMB5_pnps_function_KMy817HP=HIMB5_pnps_function_KMy817HP%>%dplyr::rename("KMy817HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy817HP_pNpS_gene_reference")



HIMB5_pnps_function_by_sample=left_join( backbone_HIMB5_pnps_function, HIMB5_pnps_function_KDe747HP,by="corresponding_gene_call")
HIMB5_pnps_function_by_sample=left_join( HIMB5_pnps_function_by_sample, HIMB5_pnps_function_KMy817HP,by="corresponding_gene_call")

dim(HIMB5_pnps_function_by_sample)

write.csv(HIMB5_pnps_function_by_sample, "HIMB5_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB5=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/HIMB5-gene_non_outlier_coverages.txt")
coverage_HIMB5=coverage_HIMB5%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB5_pnps_function_by_sample=left_join(HIMB5_pnps_function_by_sample, coverage_HIMB5, by= "corresponding_gene_call")

```



```{r}
###and bring in coverage info 
coverage_HIMB5=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/HIMB5-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB5$gene_callers_id))
##
a=length(unique(HIMB5_pnps_function_KMy817HP$corresponding_gene_call))-sum(is.na(HIMB5_pnps_function_KMy817HP$KMy817HP_pNpS_gene_reference))
a
##
b=length(unique(HIMB5_pnps_function_KDe747HP$corresponding_gene_call))-sum(is.na(HIMB5_pnps_function_KDe747HP$KDe747HP_pNpS_gene_reference))
b

percent=100*(b/total_length)
percent=100*(a/total_length)
###
sum(is.na(HIMB5_pnps_function_KDe747HP$corresponding_gene_call))
select_coverage_HIMB5_KDe747HP=coverage_HIMB5%>%dplyr::select("gene_callers_id","KDe747HP")
select_coverage_HIMB5_KDe747HP=select_coverage_HIMB5_KDe747HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB5_KDe747HP=left_join(select_coverage_HIMB5_KDe747HP, HIMB5_pnps_function_KDe747HP, by= "corresponding_gene_call")
select_coverage_HIMB5_KDe747HP$sample="KDe747HP"
select_coverage_HIMB5_KDe747HP=select_coverage_HIMB5_KDe747HP%>%dplyr::rename("pnps"="KDe747HP_pNpS_gene_reference", "coverage"="KDe747HP")
##
names(select_coverage_HIMB5_KDe747HP)

select_coverage_HIMB5_KMy817HP=coverage_HIMB5%>%dplyr::select("gene_callers_id","KMy817HP")
select_coverage_HIMB5_KMy817HP=select_coverage_HIMB5_KMy817HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB5_KMy817HP=left_join(select_coverage_HIMB5_KMy817HP, HIMB5_pnps_function_KMy817HP, by= "corresponding_gene_call")
select_coverage_HIMB5_KMy817HP$sample="KMy817HP"
select_coverage_HIMB5_KMy817HP=select_coverage_HIMB5_KMy817HP%>%dplyr::rename("pnps"="KMy817HP_pNpS_gene_reference", "coverage"="KMy817HP")



library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB5_KMy817HP, select_coverage_HIMB5_KDe747HP)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2

p_HIMB5_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.II, HIMB5")+scale_color_manual(values=c("orange","purple"))
p_HIMB5_cov



p_HIMB5_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+scale_color_manual(values=c("orange","purple"))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.II, HIMB5")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_HIMB5_cor



sum(is.na(coverage_bothsamp$pnps))

coverage_bothsamp$log_pnps=log(coverage_bothsamp$pnps)

sum(is.na(coverage_bothsamp$log_pnps))
which(is.infinite(coverage_bothsamp$log_pnps))
coverage_bothsamp[c('log_pnps')][sapply(coverage_bothsamp[c('log_pnps')], is.infinite)] <- NA
sum(is.na(coverage_bothsamp$log_pnps))
coverage_bothsamp=coverage_bothsamp%>%drop_na(log_pnps)




my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log_pnps ~ corresponding_gene_call + sample + coverage, data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "HIMB5_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)

```



###KDe747HP


```{r}
##ranking genes by pNpS per sample

HIMB5_pnps_function_ranks <- dplyr::arrange(HIMB5_pnps_function_by_sample, KDe747HP_pNpS_gene_reference)%>% dplyr::mutate(KDe747HP_rank = 1:length(KDe747HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe747HP_pNpS_gene_reference) is.na 
HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference[max(which(HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=HIMB5_pnps_function_ranks[which.min(abs(0.010-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.010$KDe747HP_rank
test_0.025=HIMB5_pnps_function_ranks[which.min(abs(0.025-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.025$KDe747HP_rank
test_0.050=HIMB5_pnps_function_ranks[which.min(abs(0.050-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.050$KDe747HP_rank
test_0.075=HIMB5_pnps_function_ranks[which.min(abs(0.075-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.075$KDe747HP_rank
test_0.10=HIMB5_pnps_function_ranks[which.min(abs(0.10-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.10$KDe747HP_rank
test_0.25=HIMB5_pnps_function_ranks[which.min(abs(0.25-HIMB5_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.25$KDe747HP_rank

order=as.numeric(test_0.010$KDe747HP_rank)
order=append(order, as.numeric(test_0.025$KDe747HP_rank))
order=append(order, as.numeric(test_0.050$KDe747HP_rank))
order=append(order, as.numeric(test_0.075$KDe747HP_rank))
order=append(order, as.numeric(test_0.10$KDe747HP_rank))
order=append(order, as.numeric(test_0.25$KDe747HP_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB5_pnps_function_ranks2=HIMB5_pnps_function_ranks%>%drop_na(KDe747HP_pNpS_gene_reference)
test=HIMB5_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""
test=test%>%
  arrange(KDe747HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))

HIMB5_pnps_function_ranks2$Environment[is.na(HIMB5_pnps_function_ranks2$Environment)] <- ""
library(seecolor)
#print_color(genes)
#names(HIMB5_pnps_function_ranks)
#HIMB5_pnps_function_ranks$genome="HIMB5"
#HIMB5_KDe747HP_map <- ggplot(data=HIMB5_pnps_function_ranks, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB5")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB5 KDe747HP")
#HIMB5_KDe747HP_map

##playing geom_line
HIMB5_pnps_function_ranks2$genome="HIMB5"
HIMB5_KDe747HP_map <- ggplot(data=HIMB5_pnps_function_ranks2, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB5")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB5 KDe747HP")  
  h=order
HIMB5_KDe747HP_map=HIMB5_KDe747HP_map+ geom_hline(yintercept = factor(HIMB5_pnps_function_ranks2$KDe747HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")+coord_flip()
HIMB5_KDe747HP_map

ggsave("HIMB5_KDe747HP.pdf", height=20, width=15)



```


###PER SAMPLE 1556 
###KMy817HP


```{r}
##ranking genes by pNpS per sample

HIMB5_pnps_function_ranks <- dplyr::arrange(HIMB5_pnps_function_by_sample, KMy817HP_pNpS_gene_reference)%>% dplyr::mutate(KMy817HP_rank = 1:length(KMy817HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy817HP_pNpS_gene_reference) is.na 
HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference[max(which(HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB5_pnps_function_ranks[which.min(abs(0.010-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.010$KMy817HP_rank
test_0.025=HIMB5_pnps_function_ranks[which.min(abs(0.025-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.025$KMy817HP_rank
test_0.050=HIMB5_pnps_function_ranks[which.min(abs(0.050-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.050$KMy817HP_rank
test_0.075=HIMB5_pnps_function_ranks[which.min(abs(0.075-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.075$KMy817HP_rank
test_0.10=HIMB5_pnps_function_ranks[which.min(abs(0.10-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.10$KMy817HP_rank
test_0.25=HIMB5_pnps_function_ranks[which.min(abs(0.25-HIMB5_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.25$KMy817HP_rank

order=as.numeric(test_0.010$KMy817HP_rank)
order=append(order, as.numeric(test_0.025$KMy817HP_rank))
order=append(order, as.numeric(test_0.050$KMy817HP_rank))
order=append(order, as.numeric(test_0.075$KMy817HP_rank))
order=append(order, as.numeric(test_0.10$KMy817HP_rank))
order=append(order, as.numeric(test_0.25$KMy817HP_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB5_pnps_function_ranks2=HIMB5_pnps_function_ranks%>%drop_na(KMy817HP_pNpS_gene_reference)
test=HIMB5_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

test=test%>%
  arrange(KMy817HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))

HIMB5_pnps_function_ranks2$Environment[is.na(HIMB5_pnps_function_ranks2$Environment)] <- ""


##playing geom_line
HIMB5_pnps_function_ranks2$genome="HIMB5"
HIMB5_KMy817HP_map <- ggplot(data=HIMB5_pnps_function_ranks2, aes(x = genome, y = factor(KMy817HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB5")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB5 KMy817HP")  
  h=order
HIMB5_KMy817HP_map=HIMB5_KMy817HP_map+ geom_hline(yintercept = factor(HIMB5_pnps_function_ranks2$KMy817HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")+coord_flip()
HIMB5_KMy817HP_map

ggsave("HIMB5_KMy817HP.pdf", height=20, width=15)


```



```{r}
pdf("HIMB5_twosamples.pdf", width=12, height=6.5)
plot_grid(HIMB5_KDe747HP_map,HIMB5_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svg("HIMB5_twosamples.svg", width=12, height=6.5)
plot_grid(HIMB5_KDe747HP_map,HIMB5_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```

###Mean HIMB5


```{r}
##ranking genes by pNpS per sample

names(HIMB5_pnps_function_by_sample)

HIMB5_pnps_function_ranks <- dplyr::arrange(HIMB5_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
HIMB5_pnps_function_ranks$pNpS.Mean[max(which(HIMB5_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB5_pnps_function_ranks[which.min(abs(0.010-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=HIMB5_pnps_function_ranks[which.min(abs(0.025-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=HIMB5_pnps_function_ranks[which.min(abs(0.050-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=HIMB5_pnps_function_ranks[which.min(abs(0.075-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=HIMB5_pnps_function_ranks[which.min(abs(0.10-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=HIMB5_pnps_function_ranks[which.min(abs(0.25-HIMB5_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank


order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB5_pnps_function_ranks2=HIMB5_pnps_function_ranks%>%drop_na(pNpS.Mean)
test=HIMB5_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))

HIMB5_pnps_function_ranks2$Environment[is.na(HIMB5_pnps_function_ranks2$Environment)] <- ""

##playing geom_line
HIMB5_pnps_function_ranks2$genome="HIMB5"
HIMB5_Mean_map <- ggplot(data=HIMB5_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB5")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.3.II")  
  h=order
HIMB5_Mean_map=HIMB5_Mean_map+ geom_hline(yintercept = factor(HIMB5_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position = "none")+coord_flip()+# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1.5, 
 size=10))
HIMB5_Mean_map

ggsave("HIMB5_Mean.pdf", height=20, width=15)




```

##############
####HIMB4 Ia.3.III
################

```{r}


pnps_HIMB4=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/pNpS.txt")
functions_HIMB4=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/HIMB4-functions.txt")
functions_HIMB4=functions_HIMB4%>%filter(source=="COG20_FUNCTION" | source=="KOfam")

length(unique(functions_HIMB4$gene_callers_id))
#we have 1221 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
Coastal_genes_HIMB4=left_join(functions_HIMB4, Coastal_genes, by="accession")
dim(Coastal_genes_HIMB4)
length(unique(Coastal_genes_HIMB4$gene_callers_id))

names(Coastal_genes_HIMB4)
Coastal_genes_HIMB4=Coastal_genes_HIMB4%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Coastal")
dim(Coastal_genes_HIMB4)

Coastal_genes_HIMB4=distinct(Coastal_genes_HIMB4,gene_callers_id, .keep_all=TRUE)
write.csv(Coastal_genes_HIMB4, "Coastal_genes_HIMB4.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB4=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/HIMB4-gene_calls.txt")
length(unique(functions_HIMB4$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB4)
functions_HIMB4_Coastal_specific=left_join(functions_HIMB4,Coastal_genes_HIMB4)

```



##now we need to add in PN/PS
```{r}

functions_HIMB4_Coastal_specific_join=functions_HIMB4_Coastal_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



HIMB4_pnps_function=left_join(pnps_HIMB4, functions_HIMB4_Coastal_specific_join, by="corresponding_gene_call")
length(unique(HIMB4_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB4_pnps_function$pNpS_gene_reference))
HIMB4_pnps_function[c('pNpS_gene_reference')][sapply(HIMB4_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB4_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB4_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_HIMB4_means.csv")
##
pNpS_within_sample_Quantiles=HIMB4_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_HIMB4_Quantiles.csv")

#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB4_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB4_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

across_samples_HIMB4<- ggplot(data=HIMB4_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB4
across_samples_HIMB4=across_samples_HIMB4 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB4


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB4_pnps_function%>%dplyr::filter(sample_id=="KDe747HP" | sample_id=="KMy817HP") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB4_pnps_function=left_join(HIMB4_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```



```{r}
unique(HIMB4_pnps_function$sample_id)
HIMB4_pnps_function_reporting=HIMB4_pnps_function%>%filter(sample_id=="KMy817HP"| sample_id=="KDe747HP")
sum(is.na(HIMB4_pnps_function_reporting$pNpS_gene_reference))
HIMB4_pnps_function_reporting=HIMB4_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
HIMB4_pnps_function_reporting$log_pnps=log(HIMB4_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(HIMB4_pnps_function_reporting$log_pnps))
which(is.infinite(HIMB4_pnps_function_reporting$log_pnps))
HIMB4_pnps_function_reporting[c('log_pnps')][sapply(HIMB4_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(HIMB4_pnps_function_reporting$log_pnps))
HIMB4_pnps_function_reporting=HIMB4_pnps_function_reporting%>%drop_na(log_pnps)

hist(HIMB4_pnps_function_reporting$log_pnps)

```






```{r}

####let's break out pnps by samples 
backbone_HIMB4_pnps_function=distinct(HIMB4_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB4_pnps_function)
dim(functions_HIMB4_Coastal_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB4_pnps_function=backbone_HIMB4_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB4_pnps_function)


###subset and rename to have sample level information
unique(HIMB4_pnps_function$sample_id)
HIMB4_pnps_function_KDe747HP=HIMB4_pnps_function%>%filter(sample_id=="KDe747HP")
names(HIMB4_pnps_function_KDe747HP)
HIMB4_pnps_function_KDe747HP=HIMB4_pnps_function_KDe747HP%>%dplyr::rename("KDe747HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe747HP_pNpS_gene_reference")


HIMB4_pnps_function_KMy817HP=HIMB4_pnps_function%>%filter(sample_id=="KMy817HP")
HIMB4_pnps_function_KMy817HP=HIMB4_pnps_function_KMy817HP%>%dplyr::rename("KMy817HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy817HP_pNpS_gene_reference")



HIMB4_pnps_function_by_sample=left_join( backbone_HIMB4_pnps_function, HIMB4_pnps_function_KDe747HP,by="corresponding_gene_call")
HIMB4_pnps_function_by_sample=left_join( HIMB4_pnps_function_by_sample, HIMB4_pnps_function_KMy817HP,by="corresponding_gene_call")
dim(HIMB4_pnps_function_by_sample)

write.csv(HIMB4_pnps_function_by_sample, "HIMB4_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB4=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/HIMB4-gene_non_outlier_coverages.txt")
coverage_HIMB4=coverage_HIMB4%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB4_pnps_function_by_sample=left_join(HIMB4_pnps_function_by_sample, coverage_HIMB4, by= "corresponding_gene_call")



p_KDe747HP<-ggplot(HIMB4_pnps_function_by_sample, aes(x=KDe747HP_pNpS_gene_reference, y=KDe747HP )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS HIMB4")+ylab("Non-outlier gene coverage KDe747HP")
p_KDe747HP


p_KMy817HP<-ggplot(HIMB4_pnps_function_by_sample, aes(x=KMy817HP_pNpS_gene_reference, y=KMy817HP )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage KMy817HP")
p_KMy817HP

```

```{r}
###and bring in coverage info 
coverage_HIMB4=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/HIMB4-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB4$gene_callers_id))
##
a=length(unique(HIMB4_pnps_function_KMy817HP$corresponding_gene_call))-sum(is.na(HIMB4_pnps_function_KMy817HP$KMy817HP_pNpS_gene_reference))
##
b=length(unique(HIMB4_pnps_function_KDe747HP$corresponding_gene_call))-sum(is.na(HIMB4_pnps_function_KDe747HP$KDe747HP_pNpS_gene_reference))
###
percenta=100*(a/total_length)
percentb=100*(b/total_length)

sum(is.na(HIMB4_pnps_function_KDe747HP$corresponding_gene_call))
select_coverage_HIMB4_KDe747HP=coverage_HIMB4%>%dplyr::select("gene_callers_id","KDe747HP")
select_coverage_HIMB4_KDe747HP=select_coverage_HIMB4_KDe747HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB4_KDe747HP=left_join(select_coverage_HIMB4_KDe747HP, HIMB4_pnps_function_KDe747HP, by= "corresponding_gene_call")
select_coverage_HIMB4_KDe747HP$sample="KDe747HP"
select_coverage_HIMB4_KDe747HP=select_coverage_HIMB4_KDe747HP%>%dplyr::rename("pnps"="KDe747HP_pNpS_gene_reference", "coverage"="KDe747HP")
##
names(select_coverage_HIMB4_KDe747HP)

select_coverage_HIMB4_KMy817HP=coverage_HIMB4%>%dplyr::select("gene_callers_id","KMy817HP")
select_coverage_HIMB4_KMy817HP=select_coverage_HIMB4_KMy817HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB4_KMy817HP=left_join(select_coverage_HIMB4_KMy817HP, HIMB4_pnps_function_KMy817HP, by= "corresponding_gene_call")
select_coverage_HIMB4_KMy817HP$sample="KMy817HP"
select_coverage_HIMB4_KMy817HP=select_coverage_HIMB4_KMy817HP%>%dplyr::rename("pnps"="KMy817HP_pNpS_gene_reference", "coverage"="KMy817HP")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB4_KMy817HP, select_coverage_HIMB4_KDe747HP)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2
p_HIMB4_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.III, HIMB4")+scale_color_manual(values=c("orange","purple"))
p_HIMB4_cov


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p

p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p_HIMB4_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=log(coverage), color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("log coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.III, HIMB4")+scale_color_manual(values=c("orange","purple"))
p_HIMB4_cov


p_HIMB4_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+scale_color_manual(values=c("orange","purple"))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.III, HIMB4")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_HIMB4_cor



my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log(pnps) ~ corresponding_gene_call + sample + log(coverage), data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "HIMB4_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)

```




###PER SAMPLE 
###KDe747HP


```{r}
##ranking genes by pNpS per sample

HIMB4_pnps_function_ranks <- dplyr::arrange(HIMB4_pnps_function_by_sample, KDe747HP_pNpS_gene_reference)%>% dplyr::mutate(KDe747HP_rank = 1:length(KDe747HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe747HP_pNpS_gene_reference) is.na 
HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference[max(which(HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=HIMB4_pnps_function_ranks[which.min(abs(0.010-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.010$KDe747HP_rank
test_0.025=HIMB4_pnps_function_ranks[which.min(abs(0.025-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.025$KDe747HP_rank
test_0.050=HIMB4_pnps_function_ranks[which.min(abs(0.050-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.050$KDe747HP_rank
test_0.075=HIMB4_pnps_function_ranks[which.min(abs(0.075-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.075$KDe747HP_rank
test_0.10=HIMB4_pnps_function_ranks[which.min(abs(0.10-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.10$KDe747HP_rank
test_0.25=HIMB4_pnps_function_ranks[which.min(abs(0.25-HIMB4_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.25$KDe747HP_rank


order=as.numeric(test_0.010$KDe747HP_rank)
order=append(order, as.numeric(test_0.025$KDe747HP_rank))
order=append(order, as.numeric(test_0.050$KDe747HP_rank))
order=append(order, as.numeric(test_0.075$KDe747HP_rank))
order=append(order, as.numeric(test_0.10$KDe747HP_rank))
order=append(order, as.numeric(test_0.25$KDe747HP_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB4_pnps_function_ranks2=HIMB4_pnps_function_ranks%>%drop_na(KDe747HP_pNpS_gene_reference)
test=HIMB4_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

test=test%>%arrange(KDe747HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))

HIMB4_pnps_function_ranks2$Environment[is.na(HIMB4_pnps_function_ranks2$Environment)] <- ""


library(seecolor)
#print_color(genes)
#names(HIMB4_pnps_function_ranks)
#HIMB4_pnps_function_ranks$genome="HIMB4"
#HIMB4_KDe747HP_map <- ggplot(data=HIMB4_pnps_function_ranks, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB4")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB4 KDe747HP")
#HIMB4_KDe747HP_map

##playing geom_line
HIMB4_pnps_function_ranks2$genome="HIMB4"
HIMB4_KDe747HP_map <- ggplot(data=HIMB4_pnps_function_ranks2, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB4")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB4 KDe747HP")  
  h=order
HIMB4_KDe747HP_map=HIMB4_KDe747HP_map+ geom_hline(yintercept = factor(HIMB4_pnps_function_ranks2$KDe747HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")
HIMB4_KDe747HP_map

ggsave("HIMB4_KDe747HP.pdf", height=20, width=15)



```


###PER SAMPLE 
###KMy817HP


```{r}
##ranking genes by pNpS per sample

HIMB4_pnps_function_ranks <- dplyr::arrange(HIMB4_pnps_function_by_sample, KMy817HP_pNpS_gene_reference)%>% dplyr::mutate(KMy817HP_rank = 1:length(KMy817HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy817HP_pNpS_gene_reference) is.na 
HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference[max(which(HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB4_pnps_function_ranks[which.min(abs(0.010-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.010$KMy817HP_rank
test_0.025=HIMB4_pnps_function_ranks[which.min(abs(0.025-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.025$KMy817HP_rank
test_0.050=HIMB4_pnps_function_ranks[which.min(abs(0.050-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.050$KMy817HP_rank
test_0.075=HIMB4_pnps_function_ranks[which.min(abs(0.075-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.075$KMy817HP_rank
test_0.10=HIMB4_pnps_function_ranks[which.min(abs(0.10-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.10$KMy817HP_rank
test_0.25=HIMB4_pnps_function_ranks[which.min(abs(0.25-HIMB4_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.25$KMy817HP_rank

order=as.numeric(test_0.010$KMy817HP_rank)
order=append(order, as.numeric(test_0.025$KMy817HP_rank))
order=append(order, as.numeric(test_0.050$KMy817HP_rank))
order=append(order, as.numeric(test_0.075$KMy817HP_rank))
order=append(order, as.numeric(test_0.10$KMy817HP_rank))
order=append(order, as.numeric(test_0.25$KMy817HP_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB4_pnps_function_ranks2=HIMB4_pnps_function_ranks%>%drop_na(KMy817HP_pNpS_gene_reference)
test=HIMB4_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB4_pnps_function_ranks2$Environment[is.na(HIMB4_pnps_function_ranks2$Environment)] <- ""


test=test%>%
  arrange(KMy817HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB4_pnps_function_ranks2$genome="HIMB4"
HIMB4_KMy817HP_map <- ggplot(data=HIMB4_pnps_function_ranks2, aes(x = genome, y = factor(KMy817HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB4")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB4 KMy817HP")  
  h=order
HIMB4_KMy817HP_map=HIMB4_KMy817HP_map+ geom_hline(yintercept = factor(HIMB4_pnps_function_ranks2$KMy817HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")
HIMB4_KMy817HP_map

ggsave("HIMB4_KMy817HP.pdf", height=20, width=15)



```


```{r}
pdf("HIMB4_twosamples.pdf", width=12, height=6.5)
plot_grid(HIMB4_KDe747HP_map,HIMB4_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("HIMB4_twosamples.svg", width=12, height=6.5)
plot_grid(HIMB4_KDe747HP_map,HIMB4_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```


###Mean HIMB4


```{r}
##ranking genes by pNpS per sample

names(HIMB4_pnps_function_by_sample)

HIMB4_pnps_function_ranks <- dplyr::arrange(HIMB4_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
HIMB4_pnps_function_ranks$pNpS.Mean[max(which(HIMB4_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB4_pnps_function_ranks[which.min(abs(0.010-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=HIMB4_pnps_function_ranks[which.min(abs(0.025-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=HIMB4_pnps_function_ranks[which.min(abs(0.050-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=HIMB4_pnps_function_ranks[which.min(abs(0.075-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=HIMB4_pnps_function_ranks[which.min(abs(0.10-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=HIMB4_pnps_function_ranks[which.min(abs(0.25-HIMB4_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank

order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB4_pnps_function_ranks2=HIMB4_pnps_function_ranks%>%drop_na(pNpS.Mean)

test=HIMB4_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB4_pnps_function_ranks2$Environment[is.na(HIMB4_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB4_pnps_function_ranks2$genome="HIMB4"
HIMB4_Mean_map <- ggplot(data=HIMB4_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB4")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.3.III")  
  h=order
HIMB4_Mean_map=HIMB4_Mean_map+ geom_hline(yintercept = factor(HIMB4_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position = "none")+coord_flip()+# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1.5,
 size=10))
HIMB4_Mean_map

ggsave("HIMB4_Mean.pdf", height=20, width=15)
ggsave("HIMB4_Mean.svg", height=20, width=15)

```


##############
####HIMB1556 Ia.3.VI
################

```{r}


pnps_HIMB1556=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/pNpS.txt")
functions_HIMB1556=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/HIMB1556-functions.txt")
functions_HIMB1556=functions_HIMB1556%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")
length(unique(functions_HIMB1556$gene_callers_id))
#we have 1221 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
Coastal_genes_HIMB1556=left_join(functions_HIMB1556, Coastal_genes, by="accession")
dim(Coastal_genes_HIMB1556)
length(unique(Coastal_genes_HIMB1556$gene_callers_id))

names(Coastal_genes_HIMB1556)
Coastal_genes_HIMB1556=Coastal_genes_HIMB1556%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Coastal")
dim(Coastal_genes_HIMB1556)

Coastal_genes_HIMB1556=distinct(Coastal_genes_HIMB1556,gene_callers_id, .keep_all=TRUE)
write.csv(Coastal_genes_HIMB1556, "Coastal_genes_HIMB1556.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB1556=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/HIMB1556-gene_calls.txt")
length(unique(functions_HIMB1556$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB1556)
functions_HIMB1556_Coastal_specific=left_join(functions_HIMB1556,Coastal_genes_HIMB1556)

```



##now we need to add in PN/PS
###Do I include the concatenated when I calculate the mean???
```{r}

functions_HIMB1556_Coastal_specific_join=functions_HIMB1556_Coastal_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



HIMB1556_pnps_function=left_join(pnps_HIMB1556, functions_HIMB1556_Coastal_specific_join, by="corresponding_gene_call")
length(unique(HIMB1556_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB1556_pnps_function$pNpS_gene_reference))
HIMB1556_pnps_function[c('pNpS_gene_reference')][sapply(HIMB1556_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB1556_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB1556_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_HIMB1556.csv")

##
pNpS_within_sample_Quantiles=HIMB1556_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_HIMB1556.csv")


#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB1556_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB1556_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p


p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p


p<-ggplot(HIMB1556_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1556 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(HIMB1556_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1556 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_HIMB1556<- ggplot(data=HIMB1556_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB1556
across_samples_HIMB1556=across_samples_HIMB1556 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB1556


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB1556_pnps_function%>%dplyr::filter(sample_id=="KDe747HP" | sample_id=="KMy817HP") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB1556_pnps_function=left_join(HIMB1556_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```


```{r}
unique(HIMB1556_pnps_function$sample_id)
HIMB1556_pnps_function_reporting=HIMB1556_pnps_function%>%filter(sample_id=="KMy817HP"| sample_id=="KDe747HP")

sum(is.na(HIMB1556_pnps_function_reporting$pNpS_gene_reference))
HIMB1556_pnps_function_reporting=HIMB1556_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
HIMB1556_pnps_function_reporting$log_pnps=log(HIMB1556_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(HIMB1556_pnps_function_reporting$log_pnps))
which(is.infinite(HIMB1556_pnps_function_reporting$log_pnps))
HIMB1556_pnps_function_reporting[c('log_pnps')][sapply(HIMB1556_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(HIMB1556_pnps_function_reporting$log_pnps))
HIMB1556_pnps_function_reporting=HIMB1556_pnps_function_reporting%>%drop_na(log_pnps)

hist(HIMB1556_pnps_function_reporting$log_pnps)



p<-ggplot(HIMB1556_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB1556_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(HIMB1556_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB1556_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p

```



```{r}

####let's break out pnps by samples 
backbone_HIMB1556_pnps_function=distinct(HIMB1556_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB1556_pnps_function)
dim(functions_HIMB1556_Coastal_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB1556_pnps_function=backbone_HIMB1556_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB1556_pnps_function)


###subset and rename to have sample level information
unique(HIMB1556_pnps_function$sample_id)
HIMB1556_pnps_function_KDe747HP=HIMB1556_pnps_function%>%filter(sample_id=="KDe747HP")
names(HIMB1556_pnps_function_KDe747HP)
HIMB1556_pnps_function_KDe747HP=HIMB1556_pnps_function_KDe747HP%>%dplyr::rename("KDe747HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe747HP_pNpS_gene_reference")


HIMB1556_pnps_function_KMy817HP=HIMB1556_pnps_function%>%filter(sample_id=="KMy817HP")
HIMB1556_pnps_function_KMy817HP=HIMB1556_pnps_function_KMy817HP%>%dplyr::rename("KMy817HP_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy817HP_pNpS_gene_reference")



HIMB1556_pnps_function_by_sample=left_join(backbone_HIMB1556_pnps_function, HIMB1556_pnps_function_KDe747HP,by="corresponding_gene_call")
HIMB1556_pnps_function_by_sample=left_join(HIMB1556_pnps_function_by_sample, HIMB1556_pnps_function_KMy817HP,by="corresponding_gene_call")
dim(HIMB1556_pnps_function_by_sample)

write.csv(HIMB1556_pnps_function_by_sample, "HIMB1556_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB1556=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/HIMB1556-gene_non_outlier_coverages.txt")
coverage_HIMB1556=coverage_HIMB1556%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB1556_pnps_function_by_sample=left_join(HIMB1556_pnps_function_by_sample, coverage_HIMB1556, by= "corresponding_gene_call")



p_KDe747HP<-ggplot(HIMB1556_pnps_function_by_sample, aes(x=KDe747HP_pNpS_gene_reference, y=KDe747HP )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS HIMB1556")+ylab("Non-outlier gene coverage KDe747HP")
p_KDe747HP


p_KMy817HP<-ggplot(HIMB1556_pnps_function_by_sample, aes(x=KMy817HP_pNpS_gene_reference, y=KMy817HP )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage KMy817HP")
p_KMy817HP



```

```{r}
###and bring in coverage info 
coverage_HIMB1556=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/HIMB1556-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB1556$gene_callers_id))
##
a=length(unique(HIMB1556_pnps_function_KMy817HP$corresponding_gene_call))-sum(is.na(HIMB1556_pnps_function_KMy817HP$KMy817HP_pNpS_gene_reference))
#
b=length(unique(HIMB1556_pnps_function_KDe747HP$corresponding_gene_call))-sum(is.na(HIMB1556_pnps_function_KDe747HP$KDe747HP_pNpS_gene_reference))
###

percenta=100*(a/total_length)
percentb=100*(b/total_length)

sum(is.na(HIMB1556_pnps_function_KDe747HP$corresponding_gene_call))
select_coverage_HIMB1556_KDe747HP=coverage_HIMB1556%>%dplyr::select("gene_callers_id","KDe747HP")
select_coverage_HIMB1556_KDe747HP=select_coverage_HIMB1556_KDe747HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1556_KDe747HP=left_join(select_coverage_HIMB1556_KDe747HP, HIMB1556_pnps_function_KDe747HP, by= "corresponding_gene_call")
select_coverage_HIMB1556_KDe747HP$sample="KDe747HP"
select_coverage_HIMB1556_KDe747HP=select_coverage_HIMB1556_KDe747HP%>%dplyr::rename("pnps"="KDe747HP_pNpS_gene_reference", "coverage"="KDe747HP")
##
names(select_coverage_HIMB1556_KDe747HP)

select_coverage_HIMB1556_KMy817HP=coverage_HIMB1556%>%dplyr::select("gene_callers_id","KMy817HP")
select_coverage_HIMB1556_KMy817HP=select_coverage_HIMB1556_KMy817HP%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1556_KMy817HP=left_join(select_coverage_HIMB1556_KMy817HP, HIMB1556_pnps_function_KMy817HP, by= "corresponding_gene_call")
select_coverage_HIMB1556_KMy817HP$sample="KMy817HP"
select_coverage_HIMB1556_KMy817HP=select_coverage_HIMB1556_KMy817HP%>%dplyr::rename("pnps"="KMy817HP_pNpS_gene_reference", "coverage"="KMy817HP")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB1556_KMy817HP, select_coverage_HIMB1556_KDe747HP)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2
p_HIMB1556_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.VI, HIMB1556")+scale_color_manual(values=c("orange","purple"))
p_HIMB1556_cov


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p_HIMB1556_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.VI, HIMB1556")+scale_color_manual(values=c("orange","purple"))
p_HIMB1556_cov



p_HIMB1556_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+scale_color_manual(values=c("orange","purple"))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.VI, HIMB1556")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_HIMB1556_cor



my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log(pnps) ~ corresponding_gene_call + sample + coverage, data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "HIMB1556_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)
```




###PER SAMPLE 
###KDe747HP


```{r}
##ranking genes by pNpS per sample

HIMB1556_pnps_function_ranks <- dplyr::arrange(HIMB1556_pnps_function_by_sample, KDe747HP_pNpS_gene_reference)%>% dplyr::mutate(KDe747HP_rank = 1:length(KDe747HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe747HP_pNpS_gene_reference) is.na 
HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference[max(which(HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=HIMB1556_pnps_function_ranks[which.min(abs(0.010-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.010$KDe747HP_rank
test_0.025=HIMB1556_pnps_function_ranks[which.min(abs(0.025-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.025$KDe747HP_rank
test_0.050=HIMB1556_pnps_function_ranks[which.min(abs(0.050-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.050$KDe747HP_rank
test_0.075=HIMB1556_pnps_function_ranks[which.min(abs(0.075-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.075$KDe747HP_rank
test_0.10=HIMB1556_pnps_function_ranks[which.min(abs(0.10-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.10$KDe747HP_rank
test_0.25=HIMB1556_pnps_function_ranks[which.min(abs(0.25-HIMB1556_pnps_function_ranks$KDe747HP_pNpS_gene_reference)),]
test_0.25$KDe747HP_rank


order=as.numeric(test_0.010$KDe747HP_rank)
order=append(order, as.numeric(test_0.025$KDe747HP_rank))
order=append(order, as.numeric(test_0.050$KDe747HP_rank))
order=append(order, as.numeric(test_0.075$KDe747HP_rank))
order=append(order, as.numeric(test_0.10$KDe747HP_rank))
order=append(order, as.numeric(test_0.25$KDe747HP_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB1556_pnps_function_ranks2=HIMB1556_pnps_function_ranks%>%drop_na(KDe747HP_pNpS_gene_reference)

test=HIMB1556_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1556_pnps_function_ranks2$Environment[is.na(HIMB1556_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KDe747HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))


library(seecolor)
#print_color(genes)
#names(HIMB1556_pnps_function_ranks)
#HIMB1556_pnps_function_ranks$genome="HIMB1556"
#HIMB1556_KDe747HP_map <- ggplot(data=HIMB1556_pnps_function_ranks, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1556")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1556 KDe747HP")
#HIMB1556_KDe747HP_map

##playing geom_line
HIMB1556_pnps_function_ranks2$genome="HIMB1556"
HIMB1556_KDe747HP_map <- ggplot(data=HIMB1556_pnps_function_ranks2, aes(x = genome, y = factor(KDe747HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1556")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1556 KDe747HP")  
  h=order
HIMB1556_KDe747HP_map=HIMB1556_KDe747HP_map+ geom_hline(yintercept = factor(HIMB1556_pnps_function_ranks2$KDe747HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")
HIMB1556_KDe747HP_map

ggsave("HIMB1556_KDe747HP.pdf", height=20, width=15)



```


###PER SAMPLE 
###KMy817HP


```{r}
##ranking genes by pNpS per sample

HIMB1556_pnps_function_ranks <- dplyr::arrange(HIMB1556_pnps_function_by_sample, KMy817HP_pNpS_gene_reference)%>% dplyr::mutate(KMy817HP_rank = 1:length(KMy817HP_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy817HP_pNpS_gene_reference) is.na 
HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference[max(which(HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB1556_pnps_function_ranks[which.min(abs(0.010-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.010$KMy817HP_rank
test_0.025=HIMB1556_pnps_function_ranks[which.min(abs(0.025-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.025$KMy817HP_rank
test_0.050=HIMB1556_pnps_function_ranks[which.min(abs(0.050-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.050$KMy817HP_rank
test_0.075=HIMB1556_pnps_function_ranks[which.min(abs(0.075-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.075$KMy817HP_rank
test_0.10=HIMB1556_pnps_function_ranks[which.min(abs(0.10-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.10$KMy817HP_rank
test_0.25=HIMB1556_pnps_function_ranks[which.min(abs(0.25-HIMB1556_pnps_function_ranks$KMy817HP_pNpS_gene_reference)),]
test_0.25$KMy817HP_rank

order=as.numeric(test_0.010$KMy817HP_rank)
order=append(order, as.numeric(test_0.025$KMy817HP_rank))
order=append(order, as.numeric(test_0.050$KMy817HP_rank))
order=append(order, as.numeric(test_0.075$KMy817HP_rank))
order=append(order, as.numeric(test_0.10$KMy817HP_rank))
order=append(order, as.numeric(test_0.25$KMy817HP_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB1556_pnps_function_ranks2=HIMB1556_pnps_function_ranks%>%drop_na(KMy817HP_pNpS_gene_reference)

test=HIMB1556_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1556_pnps_function_ranks2$Environment[is.na(HIMB1556_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KMy817HP_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB1556_pnps_function_ranks2$genome="HIMB1556"
HIMB1556_KMy817HP_map <- ggplot(data=HIMB1556_pnps_function_ranks2, aes(x = genome, y = factor(KMy817HP_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1556")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1556 KMy817HP")  
  h=order
HIMB1556_KMy817HP_map=HIMB1556_KMy817HP_map+ geom_hline(yintercept = factor(HIMB1556_pnps_function_ranks2$KMy817HP_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position="none")
HIMB1556_KMy817HP_map

ggsave("HIMB1556_KMy817HP.pdf", height=20, width=15)



```


```{r}
pdf("HIMB1556_twosamples.pdf", width=12, height=6.5)
plot_grid(HIMB1556_KDe747HP_map,HIMB1556_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("HIMB1556_twosamples.svg", width=12, height=6.5)
plot_grid(HIMB1556_KDe747HP_map,HIMB1556_KMy817HP_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```


###Mean HIMB1556


```{r}
##ranking genes by pNpS per sample

names(HIMB1556_pnps_function_by_sample)

HIMB1556_pnps_function_ranks <- dplyr::arrange(HIMB1556_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
HIMB1556_pnps_function_ranks$pNpS.Mean[max(which(HIMB1556_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB1556_pnps_function_ranks[which.min(abs(0.010-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=HIMB1556_pnps_function_ranks[which.min(abs(0.025-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=HIMB1556_pnps_function_ranks[which.min(abs(0.050-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=HIMB1556_pnps_function_ranks[which.min(abs(0.075-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=HIMB1556_pnps_function_ranks[which.min(abs(0.10-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=HIMB1556_pnps_function_ranks[which.min(abs(0.25-HIMB1556_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank

order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "Coastal_Specific"="yellow", "Break"="black")
HIMB1556_pnps_function_ranks2=HIMB1556_pnps_function_ranks%>%drop_na(pNpS.Mean)


test=HIMB1556_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1556_pnps_function_ranks2$Environment[is.na(HIMB1556_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB1556_pnps_function_ranks2$genome="HIMB1556"
HIMB1556_Mean_map <- ggplot(data=HIMB1556_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1556")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.3.VI")  
  h=order
HIMB1556_Mean_map=HIMB1556_Mean_map+ geom_hline(yintercept = factor(HIMB1556_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "black"))+ theme(legend.position = "none")+coord_flip()+ theme(axis.text.x = element_text(angle = 45, vjust = 1.5, size=10))
HIMB1556_Mean_map

ggsave("HIMB1556_Mean.pdf", height=20, width=15)
ggsave("HIMB1556_Mean.svg", height=20, width=15)


```
#####
Bring together all coastal
####

```{r}
pdf("Coastal_means_pnps.pdf", width=16, height=8.5)
plot_grid(HIMB1556_Mean_map,HIMB5_Mean_map, HIMB4_Mean_map, align = 'hv',label_size = 12, ncol = 1, nrow=3)
dev.off()

svglite::svglite("Coastal_means_pnps.svg", width=16, height=8.5)
plot_grid(HIMB1556_Mean_map,HIMB5_Mean_map, HIMB4_Mean_map, align = 'hv',label_size = 12, ncol = 3, nrow=1)
dev.off()

```





##############
####HIMB083 Ia.3.V
################

```{r}


pnps_HIMB083=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/pNpS.txt")
functions_HIMB083=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/HIMB83-functions.txt")
functions_HIMB083=functions_HIMB083%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")
length(unique(functions_HIMB083$gene_callers_id))
#we have 1263 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
offshore_genes_HIMB083=left_join(functions_HIMB083, offshore_genes, by="accession")
dim(offshore_genes_HIMB083)
length(unique(offshore_genes_HIMB083$gene_callers_id))

names(offshore_genes_HIMB083)
offshore_genes_HIMB083=offshore_genes_HIMB083%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Offshore")
dim(offshore_genes_HIMB083)


offshore_genes_HIMB083=distinct(offshore_genes_HIMB083,gene_callers_id, .keep_all=TRUE)
write.csv(offshore_genes_HIMB083, "offshore_genes_HIMB083.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB083=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/HIMB83-gene_calls.txt")
length(unique(functions_HIMB083$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB083)
functions_HIMB083_offshore_specific=left_join(functions_HIMB083,offshore_genes_HIMB083)

```



##now we need to add in PN/PS
###Do I include the concatenated when I calculate the mean???
```{r}

functions_HIMB083_offshore_specific_join=functions_HIMB083_offshore_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



HIMB083_pnps_function=left_join(pnps_HIMB083, functions_HIMB083_offshore_specific_join, by="corresponding_gene_call")
length(unique(HIMB083_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB083_pnps_function$pNpS_gene_reference))
HIMB083_pnps_function[c('pNpS_gene_reference')][sapply(HIMB083_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB083_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB083_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
##
write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_HIMB83.csv")
pNpS_within_sample_Quantiles=HIMB083_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_HIMB83.csv")

#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB083_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB083_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p


p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p


p<-ggplot(HIMB083_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB083 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(HIMB083_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB083 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_HIMB083<- ggplot(data=HIMB083_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB083
across_samples_HIMB083=across_samples_HIMB083 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB083


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB083_pnps_function%>%dplyr::filter(sample_id=="KDe743ST" | sample_id=="KMy813ST") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB083_pnps_function=left_join(HIMB083_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```


```{r}
HIMB083_pnps_function_reporting=HIMB083_pnps_function%>%filter(sample_id=="KMy813ST"| sample_id=="KDe743ST")


sum(is.na(HIMB083_pnps_function_reporting$pNpS_gene_reference))
HIMB083_pnps_function_reporting=HIMB083_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
HIMB083_pnps_function_reporting$log_pnps=log(HIMB083_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(HIMB083_pnps_function_reporting$log_pnps))
which(is.infinite(HIMB083_pnps_function_reporting$log_pnps))
HIMB083_pnps_function_reporting[c('log_pnps')][sapply(HIMB083_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(HIMB083_pnps_function_reporting$log_pnps))
HIMB083_pnps_function_reporting=HIMB083_pnps_function_reporting%>%drop_na(log_pnps)

hist(HIMB083_pnps_function_reporting$log_pnps)



p<-ggplot(HIMB083_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB083_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(HIMB083_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB083_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p

```


```{r}

####let's break out pnps by samples 
backbone_HIMB083_pnps_function=distinct(HIMB083_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB083_pnps_function)
dim(functions_HIMB083_offshore_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB083_pnps_function=backbone_HIMB083_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB083_pnps_function)


###subset and rename to have sample level information
unique(HIMB083_pnps_function$sample_id)
HIMB083_pnps_function_KDe743ST=HIMB083_pnps_function%>%filter(sample_id=="KDe743ST")
names(HIMB083_pnps_function_KDe743ST)
HIMB083_pnps_function_KDe743ST=HIMB083_pnps_function_KDe743ST%>%dplyr::rename("KDe743ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe743ST_pNpS_gene_reference")


HIMB083_pnps_function_KMy813ST=HIMB083_pnps_function%>%filter(sample_id=="KMy813ST")
HIMB083_pnps_function_KMy813ST=HIMB083_pnps_function_KMy813ST%>%dplyr::rename("KMy813ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy813ST_pNpS_gene_reference")



HIMB083_pnps_function_by_sample=left_join( backbone_HIMB083_pnps_function, HIMB083_pnps_function_KDe743ST,by="corresponding_gene_call")
HIMB083_pnps_function_by_sample=left_join( HIMB083_pnps_function_by_sample, HIMB083_pnps_function_KMy813ST,by="corresponding_gene_call")

write.csv(HIMB083_pnps_function_by_sample, "HIMB083_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB083=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/HIMB83-gene_non_outlier_coverages.txt")
coverage_HIMB083=coverage_HIMB083%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB083_pnps_function_by_sample=left_join(HIMB083_pnps_function_by_sample, coverage_HIMB083, by= "corresponding_gene_call")



p_KDe743ST<-ggplot(HIMB083_pnps_function_by_sample, aes(x=KDe743ST_pNpS_gene_reference, y=KDe743ST )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS HIMB083")+ylab("Non-outlier gene coverage KDe743ST")
p_KDe743ST


p_KMy813ST<-ggplot(HIMB083_pnps_function_by_sample, aes(x=KMy813ST_pNpS_gene_reference, y=KMy813ST )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage KMy813ST")
p_KMy813ST


```


```{r}
###and bring in coverage info 
coverage_HIMB083=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/HIMB83-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB083$gene_callers_id))
##
a=length(unique(HIMB083_pnps_function_KMy813ST$corresponding_gene_call))-sum(is.na(HIMB083_pnps_function_KMy813ST$KMy813ST_pNpS_gene_reference))
##
b=length(unique(HIMB083_pnps_function_KDe743ST$corresponding_gene_call))-sum(is.na(HIMB083_pnps_function_KDe743ST$KDe743ST_pNpS_gene_reference))

percenta=100*(a/total_length)
percentb=100*(b/total_length)
###
sum(is.na(HIMB083_pnps_function_KDe743ST$corresponding_gene_call))
select_coverage_HIMB083_KDe743ST=coverage_HIMB083%>%dplyr::select("gene_callers_id","KDe743ST")
select_coverage_HIMB083_KDe743ST=select_coverage_HIMB083_KDe743ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB083_KDe743ST=left_join(select_coverage_HIMB083_KDe743ST, HIMB083_pnps_function_KDe743ST, by= "corresponding_gene_call")
select_coverage_HIMB083_KDe743ST$sample="KDe743ST"
select_coverage_HIMB083_KDe743ST=select_coverage_HIMB083_KDe743ST%>%dplyr::rename("pnps"="KDe743ST_pNpS_gene_reference", "coverage"="KDe743ST")
##
names(select_coverage_HIMB083_KDe743ST)

select_coverage_HIMB083_KMy813ST=coverage_HIMB083%>%dplyr::select("gene_callers_id","KMy813ST")
select_coverage_HIMB083_KMy813ST=select_coverage_HIMB083_KMy813ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB083_KMy813ST=left_join(select_coverage_HIMB083_KMy813ST, HIMB083_pnps_function_KMy813ST, by= "corresponding_gene_call")
select_coverage_HIMB083_KMy813ST$sample="KMy813ST"
select_coverage_HIMB083_KMy813ST=select_coverage_HIMB083_KMy813ST%>%dplyr::rename("pnps"="KMy813ST_pNpS_gene_reference", "coverage"="KMy813ST")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB083_KMy813ST, select_coverage_HIMB083_KDe743ST)

p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p



coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2
p_HIMB083_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.V, HIMB83")
p_HIMB083_cov




p_HIMB083_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.V, HIMB083")
p_HIMB083_cov


p_HIMB083_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.3.V, HIMB83")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_HIMB083_cor



sum(is.na(coverage_bothsamp$pnps))

coverage_bothsamp$log_pnps=log(coverage_bothsamp$pnps)

sum(is.na(coverage_bothsamp$log_pnps))
which(is.infinite(coverage_bothsamp$log_pnps))
coverage_bothsamp[c('log_pnps')][sapply(coverage_bothsamp[c('log_pnps')], is.infinite)] <- NA
sum(is.na(coverage_bothsamp$log_pnps))
coverage_bothsamp=coverage_bothsamp%>%drop_na(log_pnps)


my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log_pnps ~ corresponding_gene_call + sample + coverage, data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "HIMB083_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)

```


###PER SAMPLE 
###KDe743ST


```{r}
##ranking genes by pNpS per sample

HIMB083_pnps_function_ranks <- dplyr::arrange(HIMB083_pnps_function_by_sample, KDe743ST_pNpS_gene_reference)%>% dplyr::mutate(KDe743ST_rank = 1:length(KDe743ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe743ST_pNpS_gene_reference) is.na 
HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference[max(which(HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=HIMB083_pnps_function_ranks[which.min(abs(0.010-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.010$KDe743ST_rank
test_0.025=HIMB083_pnps_function_ranks[which.min(abs(0.025-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.025$KDe743ST_rank
test_0.050=HIMB083_pnps_function_ranks[which.min(abs(0.050-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.050$KDe743ST_rank
test_0.075=HIMB083_pnps_function_ranks[which.min(abs(0.075-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.075$KDe743ST_rank
test_0.10=HIMB083_pnps_function_ranks[which.min(abs(0.10-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.10$KDe743ST_rank
test_0.25=HIMB083_pnps_function_ranks[which.min(abs(0.25-HIMB083_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.25$KDe743ST_rank

order=as.numeric(test_0.010$KDe743ST_rank)
order=append(order, as.numeric(test_0.025$KDe743ST_rank))
order=append(order, as.numeric(test_0.050$KDe743ST_rank))
order=append(order, as.numeric(test_0.075$KDe743ST_rank))
order=append(order, as.numeric(test_0.10$KDe743ST_rank))
order=append(order, as.numeric(test_0.25$KDe743ST_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB083_pnps_function_ranks2=HIMB083_pnps_function_ranks%>%drop_na(KDe743ST_pNpS_gene_reference)
test=HIMB083_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB083_pnps_function_ranks2$Environment[is.na(HIMB083_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KDe743ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


library(seecolor)
#print_color(genes)
#names(HIMB083_pnps_function_ranks)
#HIMB083_pnps_function_ranks$genome="HIMB083"
#HIMB083_KDe743ST_map <- ggplot(data=HIMB083_pnps_function_ranks, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB083")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB083 KDe743ST")
#HIMB083_KDe743ST_map

##playing geom_line
HIMB083_pnps_function_ranks2$genome="HIMB083"
HIMB083_KDe743ST_map <- ggplot(data=HIMB083_pnps_function_ranks2, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB083")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB083 KDe743ST")  
  h=order
HIMB083_KDe743ST_map=HIMB083_KDe743ST_map+ geom_hline(yintercept = factor(HIMB083_pnps_function_ranks2$KDe743ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
HIMB083_KDe743ST_map

ggsave("HIMB083_KDe743ST.pdf", height=20, width=15)

```


###PER SAMPLE 
###KMy813ST


```{r}
##ranking genes by pNpS per sample

HIMB083_pnps_function_ranks <- dplyr::arrange(HIMB083_pnps_function_by_sample, KMy813ST_pNpS_gene_reference)%>% dplyr::mutate(KMy813ST_rank = 1:length(KMy813ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy813ST_pNpS_gene_reference) is.na 
HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference[max(which(HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB083_pnps_function_ranks[which.min(abs(0.010-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.010$KMy813ST_rank
test_0.025=HIMB083_pnps_function_ranks[which.min(abs(0.025-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.025$KMy813ST_rank
test_0.050=HIMB083_pnps_function_ranks[which.min(abs(0.050-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.050$KMy813ST_rank
test_0.075=HIMB083_pnps_function_ranks[which.min(abs(0.075-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.075$KMy813ST_rank
test_0.10=HIMB083_pnps_function_ranks[which.min(abs(0.10-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.10$KMy813ST_rank
test_0.25=HIMB083_pnps_function_ranks[which.min(abs(0.25-HIMB083_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.25$KMy813ST_rank

order=as.numeric(test_0.010$KMy813ST_rank)
order=append(order, as.numeric(test_0.025$KMy813ST_rank))
order=append(order, as.numeric(test_0.050$KMy813ST_rank))
order=append(order, as.numeric(test_0.075$KMy813ST_rank))
order=append(order, as.numeric(test_0.10$KMy813ST_rank))
order=append(order, as.numeric(test_0.25$KMy813ST_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB083_pnps_function_ranks2=HIMB083_pnps_function_ranks%>%drop_na(KMy813ST_pNpS_gene_reference)

test=HIMB083_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB083_pnps_function_ranks2$Environment[is.na(HIMB083_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KMy813ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB083_pnps_function_ranks2$genome="HIMB083"
HIMB083_KMy813ST_map <- ggplot(data=HIMB083_pnps_function_ranks2, aes(x = genome, y = factor(KMy813ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB083")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB083 KMy813ST")  
  h=order
HIMB083_KMy813ST_map=HIMB083_KMy813ST_map+ geom_hline(yintercept = factor(HIMB083_pnps_function_ranks2$KMy813ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
HIMB083_KMy813ST_map

ggsave("HIMB083_KMy813ST.pdf", height=20, width=15)
```





```{r}
pdf("HIMB083_twosamples.pdf", width=12, height=6.5)
plot_grid(HIMB083_KDe743ST_map,HIMB083_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("HIMB083_twosamples.svg", width=12, height=6.5)
plot_grid(HIMB083_KDe743ST_map,HIMB083_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```


###Mean HIMB083


```{r}
##ranking genes by pNpS per sample

names(HIMB083_pnps_function_by_sample)

HIMB083_pnps_function_ranks <- dplyr::arrange(HIMB083_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
HIMB083_pnps_function_ranks$pNpS.Mean[max(which(HIMB083_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB083_pnps_function_ranks[which.min(abs(0.010-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=HIMB083_pnps_function_ranks[which.min(abs(0.025-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=HIMB083_pnps_function_ranks[which.min(abs(0.050-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=HIMB083_pnps_function_ranks[which.min(abs(0.075-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=HIMB083_pnps_function_ranks[which.min(abs(0.10-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=HIMB083_pnps_function_ranks[which.min(abs(0.25-HIMB083_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank

order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB083_pnps_function_ranks2=HIMB083_pnps_function_ranks%>%drop_na(pNpS.Mean)

test=HIMB083_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB083_pnps_function_ranks2$Environment[is.na(HIMB083_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB083_pnps_function_ranks2$genome="HIMB083"
HIMB083_Mean_map <- ggplot(data=HIMB083_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB083")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.3.V")  
  h=order
HIMB083_Mean_map=HIMB083_Mean_map+ geom_hline(yintercept = factor(HIMB083_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+ theme(legend.position = "none")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")+ theme(axis.text.x = element_text(angle = 45, vjust = 1.5, size=10))+coord_flip()
HIMB083_Mean_map

ggsave("HIMB083_Mean.pdf", height=20, width=15)
ggsave("HIMB083_Mean.svg", height=20, width=15)
```



####
The coverage was too low for HIMB1483 to proceed with mapping of the pN/pS values for functional genes of interest

##############
####HIMB1483 Ia.4.1483
################

```{r}


pnps_HIMB1483=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/pNpS.txt")
functions_HIMB1483=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/HIMB1483-functions.txt")
functions_HIMB1483=functions_HIMB1483%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")
length(unique(functions_HIMB1483$gene_callers_id))
#we have 1263 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
offshore_genes_HIMB1483=left_join(functions_HIMB1483, offshore_genes, by="accession")
dim(offshore_genes_HIMB1483)
length(unique(offshore_genes_HIMB1483$gene_callers_id))

names(offshore_genes_HIMB1483)
offshore_genes_HIMB1483=offshore_genes_HIMB1483%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Offshore")
dim(offshore_genes_HIMB1483)


offshore_genes_HIMB1483=distinct(offshore_genes_HIMB1483,gene_callers_id, .keep_all=TRUE)
write.csv(offshore_genes_HIMB1483,"offshore_genes_HIMB1483.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB1483=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/HIMB1483-gene_calls.txt")
length(unique(functions_HIMB1483$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB1483)
functions_HIMB1483_offshore_specific=left_join(functions_HIMB1483,offshore_genes_HIMB1483)

```



##now we need to add in PN/PS


```{r}

functions_HIMB1483_offshore_specific_join=functions_HIMB1483_offshore_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")
HIMB1483_pnps_function=left_join(pnps_HIMB1483, functions_HIMB1483_offshore_specific_join, by="corresponding_gene_call")
length(unique(HIMB1483_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB1483_pnps_function$pNpS_gene_reference))
HIMB1483_pnps_function[c('pNpS_gene_reference')][sapply(HIMB1483_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB1483_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB1483_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_HIMB1483.csv")
##
pNpS_within_sample_Quantiles=HIMB1483_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_HIMB1483.csv")


#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB1483_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB1483_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p


p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p


p<-ggplot(HIMB1483_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1483 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(HIMB1483_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1483 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_HIMB1483<- ggplot(data=HIMB1483_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB1483
across_samples_HIMB1483=across_samples_HIMB1483 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB1483


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB1483_pnps_function%>%dplyr::filter(sample_id=="KDe743ST" | sample_id=="KMy813ST") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB1483_pnps_function=left_join(HIMB1483_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```

```{r}

####let's break out pnps by samples 
backbone_HIMB1483_pnps_function=distinct(HIMB1483_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB1483_pnps_function)
dim(functions_HIMB1483_offshore_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB1483_pnps_function=backbone_HIMB1483_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB1483_pnps_function)


###subset and rename to have sample level information
unique(HIMB1483_pnps_function$sample_id)
HIMB1483_pnps_function_KDe743ST=HIMB1483_pnps_function%>%filter(sample_id=="KDe743ST")
names(HIMB1483_pnps_function_KDe743ST)
HIMB1483_pnps_function_KDe743ST=HIMB1483_pnps_function_KDe743ST%>%dplyr::rename("KDe743ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe743ST_pNpS_gene_reference")


HIMB1483_pnps_function_KMy813ST=HIMB1483_pnps_function%>%filter(sample_id=="KMy813ST")
HIMB1483_pnps_function_KMy813ST=HIMB1483_pnps_function_KMy813ST%>%dplyr::rename("KMy813ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy813ST_pNpS_gene_reference")


HIMB1483_pnps_function_by_sample=left_join( backbone_HIMB1483_pnps_function, HIMB1483_pnps_function_KDe743ST,by="corresponding_gene_call")
HIMB1483_pnps_function_by_sample=left_join( HIMB1483_pnps_function_by_sample, HIMB1483_pnps_function_KMy813ST,by="corresponding_gene_call")
dim(HIMB1483_pnps_function_by_sample)

write.csv(HIMB1483_pnps_function_by_sample, "HIMB1483_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB1483=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/HIMB1483-gene_non_outlier_coverages.txt")
coverage_HIMB1483=coverage_HIMB1483%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB1483_pnps_function_by_sample=left_join(HIMB1483_pnps_function_by_sample, coverage_HIMB1483, by= "corresponding_gene_call")



p_KDe743ST<-ggplot(HIMB1483_pnps_function_by_sample, aes(x=KDe743ST_pNpS_gene_reference, y=KDe743ST )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS HIMB1483")+ylab("Non-outlier gene coverage KDe743ST")
p_KDe743ST


p_KMy813ST<-ggplot(HIMB1483_pnps_function_by_sample, aes(x=KMy813ST_pNpS_gene_reference, y=KMy813ST )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage KMy813ST")
p_KMy813ST



```


```{r}
###and bring in coverage info 
coverage_HIMB1483=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/HIMB1483-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB1483$gene_callers_id))
##
a=length(unique(HIMB1483_pnps_function_KMy813ST$corresponding_gene_call))-sum(is.na(HIMB1483_pnps_function_KMy813ST$KMy813ST_pNpS_gene_reference))
##
b=length(unique(HIMB1483_pnps_function_KDe743ST$corresponding_gene_call))-sum(is.na(HIMB1483_pnps_function_KDe743ST$KDe743ST_pNpS_gene_reference))
###

percenta=100*(a/total_length)
percentb=100*(b/total_length)
##

sum(is.na(HIMB1483_pnps_function_KDe743ST$corresponding_gene_call))
select_coverage_HIMB1483_KDe743ST=coverage_HIMB1483%>%dplyr::select("gene_callers_id","KDe743ST")
select_coverage_HIMB1483_KDe743ST=select_coverage_HIMB1483_KDe743ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1483_KDe743ST=left_join(select_coverage_HIMB1483_KDe743ST, HIMB1483_pnps_function_KDe743ST, by= "corresponding_gene_call")
select_coverage_HIMB1483_KDe743ST$sample="KDe743ST"
select_coverage_HIMB1483_KDe743ST=select_coverage_HIMB1483_KDe743ST%>%dplyr::rename("pnps"="KDe743ST_pNpS_gene_reference", "coverage"="KDe743ST")
##
names(select_coverage_HIMB1483_KDe743ST)

select_coverage_HIMB1483_KMy813ST=coverage_HIMB1483%>%dplyr::select("gene_callers_id","KMy813ST")
select_coverage_HIMB1483_KMy813ST=select_coverage_HIMB1483_KMy813ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1483_KMy813ST=left_join(select_coverage_HIMB1483_KMy813ST, HIMB1483_pnps_function_KMy813ST, by= "corresponding_gene_call")
select_coverage_HIMB1483_KMy813ST$sample="KMy813ST"
select_coverage_HIMB1483_KMy813ST=select_coverage_HIMB1483_KMy813ST%>%dplyr::rename("pnps"="KMy813ST_pNpS_gene_reference", "coverage"="KMy813ST")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB1483_KMy813ST, select_coverage_HIMB1483_KDe743ST)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2
p_HIMB1483_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.1483, HIMB1483")
p_HIMB1483_cov


```





##############
####HIMB1437 
################

```{r}


pnps_HIMB1437=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/pNpS.txt")
functions_HIMB1437=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/HIMB1437-functions.txt")
functions_HIMB1437=functions_HIMB1437%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")

length(unique(functions_HIMB1437$gene_callers_id))
#we have 1263 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
offshore_genes_HIMB1437=left_join(functions_HIMB1437, offshore_genes, by="accession")
dim(offshore_genes_HIMB1437)
length(unique(offshore_genes_HIMB1437$gene_callers_id))



names(offshore_genes_HIMB1437)
offshore_genes_HIMB1437=offshore_genes_HIMB1437%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Offshore")
dim(offshore_genes_HIMB1437)


offshore_genes_HIMB1437=distinct(offshore_genes_HIMB1437,gene_callers_id, .keep_all=TRUE)
write.csv(offshore_genes_HIMB1437, "offshore_genes_HIMB1437.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_HIMB1437=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/HIMB1437-gene_calls.txt")
length(unique(functions_HIMB1437$gene_callers_id))
#we have 1436 genes 
names(functions_HIMB1437)
functions_HIMB1437_offshore_specific=left_join(functions_HIMB1437,offshore_genes_HIMB1437)

```



##now we need to add in PN/PS
###Do I include the concatenated when I calculate the mean???
```{r}

functions_HIMB1437_offshore_specific_join=functions_HIMB1437_offshore_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



HIMB1437_pnps_function=left_join(pnps_HIMB1437, functions_HIMB1437_offshore_specific_join, by="corresponding_gene_call")
length(unique(HIMB1437_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(HIMB1437_pnps_function$pNpS_gene_reference))
HIMB1437_pnps_function[c('pNpS_gene_reference')][sapply(HIMB1437_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(HIMB1437_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=HIMB1437_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means

write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_HIMB1437.csv")
##
pNpS_within_sample_Quantiles=HIMB1437_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))

write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_HIMB1437.csv")
#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=HIMB1437_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(HIMB1437_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p
ggsave("between_samples.png")

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p


p<-ggplot(HIMB1437_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1437 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(HIMB1437_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of HIMB1437 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_HIMB1437<- ggplot(data=HIMB1437_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_HIMB1437
across_samples_HIMB1437=across_samples_HIMB1437 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_HIMB1437


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=HIMB1437_pnps_function%>%dplyr::filter(sample_id=="KDe743ST" | sample_id=="KMy813ST") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


HIMB1437_pnps_function=left_join(HIMB1437_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```

```{r}
HIMB1437_pnps_function_reporting=HIMB1437_pnps_function%>%filter(sample_id=="KMy813ST"| sample_id=="KDe743ST")

sum(is.na(HIMB1437_pnps_function_reporting$pNpS_gene_reference))
HIMB1437_pnps_function_reporting=HIMB1437_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
HIMB1437_pnps_function_reporting$log_pnps=log(HIMB1437_pnps_function_reporting$pNpS_gene_reference)

which(is.infinite(HIMB1437_pnps_function_reporting$log_pnps))
sum(is.na(HIMB1437_pnps_function_reporting$log_pnps))
#HIMB1437_pnps_function_reporting[c('log_pnps')][sapply(HIMB1437_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(HIMB1437_pnps_function_reporting$log_pnps))
HIMB1437_pnps_function_reporting=HIMB1437_pnps_function_reporting%>%drop_na(log_pnps)

hist(HIMB1437_pnps_function_reporting$log_pnps)

p<-ggplot(HIMB1437_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB1437_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(HIMB1437_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(HIMB1437_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


```


```{r}

####let's break out pnps by samples 
backbone_HIMB1437_pnps_function=distinct(HIMB1437_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_HIMB1437_pnps_function)
dim(functions_HIMB1437_offshore_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_HIMB1437_pnps_function=backbone_HIMB1437_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_HIMB1437_pnps_function)


###subset and rename to have sample level information
unique(HIMB1437_pnps_function$sample_id)
HIMB1437_pnps_function_KDe743ST=HIMB1437_pnps_function%>%filter(sample_id=="KDe743ST")
names(HIMB1437_pnps_function_KDe743ST)
HIMB1437_pnps_function_KDe743ST=HIMB1437_pnps_function_KDe743ST%>%dplyr::rename("KDe743ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe743ST_pNpS_gene_reference")


HIMB1437_pnps_function_KMy813ST=HIMB1437_pnps_function%>%filter(sample_id=="KMy813ST")
HIMB1437_pnps_function_KMy813ST=HIMB1437_pnps_function_KMy813ST%>%dplyr::rename("KMy813ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy813ST_pNpS_gene_reference")



HIMB1437_pnps_function_by_sample=left_join( backbone_HIMB1437_pnps_function, HIMB1437_pnps_function_KDe743ST,by="corresponding_gene_call")
HIMB1437_pnps_function_by_sample=left_join( HIMB1437_pnps_function_by_sample, HIMB1437_pnps_function_KMy813ST,by="corresponding_gene_call")

write.csv(HIMB1437_pnps_function_by_sample, "HIMB1437_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_HIMB1437=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/HIMB1437-gene_non_outlier_coverages.txt")
coverage_HIMB1437=coverage_HIMB1437%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
HIMB1437_pnps_function_by_sample=left_join(HIMB1437_pnps_function_by_sample, coverage_HIMB1437, by= "corresponding_gene_call")


p_KDe743ST<-ggplot(HIMB1437_pnps_function_by_sample, aes(x=KDe743ST_pNpS_gene_reference, y=KDe743ST )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS HIMB1437")+ylab("Non-outlier gene coverage")
p_KDe743ST


p_KMy813ST<-ggplot(HIMB1437_pnps_function_by_sample, aes(x=KMy813ST_pNpS_gene_reference, y=KMy813ST )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage")
p_KMy813ST

```




```{r}
###and bring in coverage info 
coverage_HIMB1437=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/HIMB1437-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_HIMB1437$gene_callers_id))
##
a=length(unique(HIMB1437_pnps_function_KMy813ST$corresponding_gene_call))-sum(is.na(HIMB1437_pnps_function_KMy813ST$KMy813ST_pNpS_gene_reference))
##
b=length(unique(HIMB1437_pnps_function_KDe743ST$corresponding_gene_call))-sum(is.na(HIMB1437_pnps_function_KDe743ST$KDe743ST_pNpS_gene_reference))

percenta=100*(a/total_length)
percentb=100*(b/total_length)

###
sum(is.na(HIMB1437_pnps_function_KDe743ST$corresponding_gene_call))
select_coverage_HIMB1437_KDe743ST=coverage_HIMB1437%>%dplyr::select("gene_callers_id","KDe743ST")
select_coverage_HIMB1437_KDe743ST=select_coverage_HIMB1437_KDe743ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1437_KDe743ST=left_join(select_coverage_HIMB1437_KDe743ST, HIMB1437_pnps_function_KDe743ST, by= "corresponding_gene_call")
select_coverage_HIMB1437_KDe743ST$sample="KDe743ST"
select_coverage_HIMB1437_KDe743ST=select_coverage_HIMB1437_KDe743ST%>%dplyr::rename("pnps"="KDe743ST_pNpS_gene_reference", "coverage"="KDe743ST")
##
names(select_coverage_HIMB1437_KDe743ST)

select_coverage_HIMB1437_KMy813ST=coverage_HIMB1437%>%dplyr::select("gene_callers_id","KMy813ST")
select_coverage_HIMB1437_KMy813ST=select_coverage_HIMB1437_KMy813ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_HIMB1437_KMy813ST=left_join(select_coverage_HIMB1437_KMy813ST, HIMB1437_pnps_function_KMy813ST, by= "corresponding_gene_call")
select_coverage_HIMB1437_KMy813ST$sample="KMy813ST"
select_coverage_HIMB1437_KMy813ST=select_coverage_HIMB1437_KMy813ST%>%dplyr::rename("pnps"="KMy813ST_pNpS_gene_reference", "coverage"="KMy813ST")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_HIMB1437_KMy813ST, select_coverage_HIMB1437_KDe743ST)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2
p_HIMB1437_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.II, HIMB1437")
p_HIMB1437_cov


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p


p_HIMB1437_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.II, HIMB1437")
p_HIMB1437_cov



p_HIMB1437_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.II, HIMB1437")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_HIMB1437_cor



sum(is.na(coverage_bothsamp$pnps))

coverage_bothsamp$log_pnps=log(coverage_bothsamp$pnps)

sum(is.na(coverage_bothsamp$log_pnps))
which(is.infinite(coverage_bothsamp$log_pnps))
coverage_bothsamp[c('log_pnps')][sapply(coverage_bothsamp[c('log_pnps')], is.infinite)] <- NA
sum(is.na(coverage_bothsamp$log_pnps))
coverage_bothsamp=coverage_bothsamp%>%drop_na(log_pnps)


my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log_pnps ~ corresponding_gene_call + sample + coverage, data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "HIMB1437_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)
```


###PER SAMPLE 
###KDe743ST


```{r}
##ranking genes by pNpS per sample

HIMB1437_pnps_function_ranks <- dplyr::arrange(HIMB1437_pnps_function_by_sample, KDe743ST_pNpS_gene_reference)%>% dplyr::mutate(KDe743ST_rank = 1:length(KDe743ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe743ST_pNpS_gene_reference) is.na 
HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference[max(which(HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=HIMB1437_pnps_function_ranks[which.min(abs(0.010-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.010$KDe743ST_rank
test_0.025=HIMB1437_pnps_function_ranks[which.min(abs(0.025-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.025$KDe743ST_rank
test_0.050=HIMB1437_pnps_function_ranks[which.min(abs(0.050-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.050$KDe743ST_rank
test_0.075=HIMB1437_pnps_function_ranks[which.min(abs(0.075-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.075$KDe743ST_rank
test_0.10=HIMB1437_pnps_function_ranks[which.min(abs(0.10-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.10$KDe743ST_rank
test_0.25=HIMB1437_pnps_function_ranks[which.min(abs(0.25-HIMB1437_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.25$KDe743ST_rank


order=as.numeric(test_0.010$KDe743ST_rank)
order=append(order, as.numeric(test_0.025$KDe743ST_rank))
order=append(order, as.numeric(test_0.050$KDe743ST_rank))
order=append(order, as.numeric(test_0.075$KDe743ST_rank))
order=append(order, as.numeric(test_0.10$KDe743ST_rank))
order=append(order, as.numeric(test_0.25$KDe743ST_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB1437_pnps_function_ranks2=HIMB1437_pnps_function_ranks%>%drop_na(KDe743ST_pNpS_gene_reference)

test=HIMB1437_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1437_pnps_function_ranks2$Environment[is.na(HIMB1437_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KDe743ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


library(seecolor)
#print_color(genes)
#names(HIMB1437_pnps_function_ranks)
#HIMB1437_pnps_function_ranks$genome="HIMB1437"
#HIMB1437_KDe743ST_map <- ggplot(data=HIMB1437_pnps_function_ranks, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1437")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1437 KDe743ST")
#HIMB1437_KDe743ST_map

##playing geom_line
HIMB1437_pnps_function_ranks2$genome="HIMB1437"
HIMB1437_KDe743ST_map <- ggplot(data=HIMB1437_pnps_function_ranks2, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1437")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1437 KDe743ST")  
  h=order
HIMB1437_KDe743ST_map=HIMB1437_KDe743ST_map+ geom_hline(yintercept = factor(HIMB1437_pnps_function_ranks2$KDe743ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
HIMB1437_KDe743ST_map

ggsave("HIMB1437_KDe743ST.pdf", height=20, width=15)

```




###PER SAMPLE 
###KMy813ST


```{r}
##ranking genes by pNpS per sample

HIMB1437_pnps_function_ranks <- dplyr::arrange(HIMB1437_pnps_function_by_sample, KMy813ST_pNpS_gene_reference)%>% dplyr::mutate(KMy813ST_rank = 1:length(KMy813ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy813ST_pNpS_gene_reference) is.na 
HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference[max(which(HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB1437_pnps_function_ranks[which.min(abs(0.010-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.010$KMy813ST_rank
test_0.025=HIMB1437_pnps_function_ranks[which.min(abs(0.025-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.025$KMy813ST_rank
test_0.050=HIMB1437_pnps_function_ranks[which.min(abs(0.050-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.050$KMy813ST_rank
test_0.075=HIMB1437_pnps_function_ranks[which.min(abs(0.075-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.075$KMy813ST_rank
test_0.10=HIMB1437_pnps_function_ranks[which.min(abs(0.10-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.10$KMy813ST_rank
test_0.25=HIMB1437_pnps_function_ranks[which.min(abs(0.25-HIMB1437_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.25$KMy813ST_rank


order=as.numeric(test_0.010$KMy813ST_rank)
order=append(order, as.numeric(test_0.025$KMy813ST_rank))
order=append(order, as.numeric(test_0.050$KMy813ST_rank))
order=append(order, as.numeric(test_0.075$KMy813ST_rank))
order=append(order, as.numeric(test_0.10$KMy813ST_rank))
order=append(order, as.numeric(test_0.25$KMy813ST_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB1437_pnps_function_ranks2=HIMB1437_pnps_function_ranks%>%drop_na(KMy813ST_pNpS_gene_reference)
test=HIMB1437_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1437_pnps_function_ranks2$Environment[is.na(HIMB1437_pnps_function_ranks2$Environment)] <- ""


test=test%>%
  arrange(KMy813ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB1437_pnps_function_ranks2$genome="HIMB1437"
HIMB1437_KMy813ST_map <- ggplot(data=HIMB1437_pnps_function_ranks2, aes(x = genome, y = factor(KMy813ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1437")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("HIMB1437 KMy813ST")  
  h=order
HIMB1437_KMy813ST_map=HIMB1437_KMy813ST_map+ geom_hline(yintercept = factor(HIMB1437_pnps_function_ranks2$KMy813ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
HIMB1437_KMy813ST_map

ggsave("HIMB1437_KMy813ST.pdf", height=20, width=15)
```


```{r}
pdf("HIMB1437_twosamples.pdf", width=12, height=6.5)
plot_grid(HIMB1437_KDe743ST_map,HIMB1437_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("HIMB1437_twosamples.svg", width=12, height=6.5)
plot_grid(HIMB1437_KDe743ST_map,HIMB1437_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```





###Mean HIMB1437


```{r}
##ranking genes by pNpS per sample

names(HIMB1437_pnps_function_by_sample)

HIMB1437_pnps_function_ranks <- dplyr::arrange(HIMB1437_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
HIMB1437_pnps_function_ranks$pNpS.Mean[max(which(HIMB1437_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=HIMB1437_pnps_function_ranks[which.min(abs(0.010-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=HIMB1437_pnps_function_ranks[which.min(abs(0.025-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=HIMB1437_pnps_function_ranks[which.min(abs(0.050-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=HIMB1437_pnps_function_ranks[which.min(abs(0.075-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=HIMB1437_pnps_function_ranks[which.min(abs(0.10-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=HIMB1437_pnps_function_ranks[which.min(abs(0.25-HIMB1437_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank

order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
HIMB1437_pnps_function_ranks2=HIMB1437_pnps_function_ranks%>%drop_na(pNpS.Mean)
test=HIMB1437_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

HIMB1437_pnps_function_ranks2$Environment[is.na(HIMB1437_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
HIMB1437_pnps_function_ranks2$genome="HIMB1437"
HIMB1437_Mean_map <- ggplot(data=HIMB1437_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("HIMB1437")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.4.II")  
  h=order
HIMB1437_Mean_map=HIMB1437_Mean_map+ geom_hline(yintercept = factor(HIMB1437_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+ theme(legend.position = "none")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")+theme(axis.text.x = element_text(angle = 45, vjust = 1.5, size=10))+coord_flip()
HIMB1437_Mean_map

ggsave("HIMB1437_Mean.pdf", height=20, width=15)
ggsave("HIMB1437_Mean.svg", height=20, width=15)
```





##############
####RS39 Ia.4.rs39
################

```{r}


pnps_RS39=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/pNpS.txt")
functions_RS39=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/RS39-functions.txt")
functions_RS39=functions_RS39%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")
length(unique(functions_RS39$gene_callers_id))
#we have 1263 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
offshore_genes_RS39=left_join(functions_RS39, offshore_genes, by="accession")
dim(offshore_genes_RS39)
length(unique(offshore_genes_RS39$gene_callers_id))

names(offshore_genes_RS39)
offshore_genes_RS39=offshore_genes_RS39%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Offshore")
dim(offshore_genes_RS39)


offshore_genes_RS39=distinct(offshore_genes_RS39,gene_callers_id, .keep_all=TRUE)
write.csv(offshore_genes_RS39, "offshore_genes_RS39.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_RS39=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/RS39-gene_calls.txt")
length(unique(functions_RS39$gene_callers_id))
#we have 1436 genes 
names(functions_RS39)
functions_RS39_offshore_specific=left_join(functions_RS39,offshore_genes_RS39)

```



##now we need to add in PN/PS
```{r}

functions_RS39_offshore_specific_join=functions_RS39_offshore_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



RS39_pnps_function=left_join(pnps_RS39, functions_RS39_offshore_specific_join, by="corresponding_gene_call")
length(unique(RS39_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(RS39_pnps_function$pNpS_gene_reference))
RS39_pnps_function[c('pNpS_gene_reference')][sapply(RS39_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(RS39_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=RS39_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_RS39.csv")
##
pNpS_within_sample_Quantiles=RS39_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_RS39.csv")

#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=RS39_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(RS39_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p
ggsave("Minimal_pnps_diff_RS39.png")

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p
ggsave("Minimal_pnps_diff_RS39_0.25.png")

p<-ggplot(RS39_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of RS39 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(RS39_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of RS39 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_RS39<- ggplot(data=RS39_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_RS39
across_samples_RS39=across_samples_RS39 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_RS39


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=RS39_pnps_function%>%dplyr::filter(sample_id=="KDe743ST" | sample_id=="KMy813ST") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


RS39_pnps_function=left_join(RS39_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```



```{r}

RS39_pnps_function_reporting=RS39_pnps_function%>%filter(sample_id=="KMy813ST"| sample_id=="KDe743ST")

sum(is.na(RS39_pnps_function_reporting$pNpS_gene_reference))
RS39_pnps_function_reporting=RS39_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
RS39_pnps_function_reporting$log_pnps=log(RS39_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(RS39_pnps_function_reporting$log_pnps))
which(is.infinite(RS39_pnps_function_reporting$log_pnps))
RS39_pnps_function_reporting[c('log_pnps')][sapply(RS39_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(RS39_pnps_function_reporting$log_pnps))
RS39_pnps_function_reporting=RS39_pnps_function_reporting%>%drop_na(log_pnps)

hist(RS39_pnps_function_reporting$log_pnps)



p<-ggplot(RS39_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(RS39_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(RS39_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(RS39_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p




```



```{r}

####let's break out pnps by samples 
backbone_RS39_pnps_function=distinct(RS39_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_RS39_pnps_function)
dim(functions_RS39_offshore_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_RS39_pnps_function=backbone_RS39_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_RS39_pnps_function)


###subset and rename to have sample level information
unique(RS39_pnps_function$sample_id)
RS39_pnps_function_KDe743ST=RS39_pnps_function%>%filter(sample_id=="KDe743ST")
names(RS39_pnps_function_KDe743ST)
RS39_pnps_function_KDe743ST=RS39_pnps_function_KDe743ST%>%dplyr::rename("KDe743ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe743ST_pNpS_gene_reference")


RS39_pnps_function_KMy813ST=RS39_pnps_function%>%filter(sample_id=="KMy813ST")
RS39_pnps_function_KMy813ST=RS39_pnps_function_KMy813ST%>%dplyr::rename("KMy813ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy813ST_pNpS_gene_reference")



RS39_pnps_function_by_sample=left_join( backbone_RS39_pnps_function, RS39_pnps_function_KDe743ST,by="corresponding_gene_call")
RS39_pnps_function_by_sample=left_join( RS39_pnps_function_by_sample, RS39_pnps_function_KMy813ST,by="corresponding_gene_call")

write.csv(RS39_pnps_function_by_sample, "RS39_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_RS39=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/RS39-gene_non_outlier_coverages.txt")
coverage_RS39=coverage_RS39%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
RS39_pnps_function_by_sample=left_join(RS39_pnps_function_by_sample, coverage_RS39, by= "corresponding_gene_call")



p_KDe743ST<-ggplot(RS39_pnps_function_by_sample, aes(x=KDe743ST_pNpS_gene_reference, y=KDe743ST )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS RS39")+ylab("Non-outlier gene coverage KDe743ST")
p_KDe743ST


p_KMy813ST<-ggplot(RS39_pnps_function_by_sample, aes(x=KMy813ST_pNpS_gene_reference, y=KMy813ST )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage KMy813ST")
p_KMy813ST


```



```{r}
###and bring in coverage info 
coverage_RS39=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/RS39-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_RS39$gene_callers_id))
##
a=length(unique(RS39_pnps_function_KMy813ST$corresponding_gene_call))-sum(is.na(RS39_pnps_function_KMy813ST$KMy813ST_pNpS_gene_reference))
##
b=length(unique(RS39_pnps_function_KDe743ST$corresponding_gene_call))-sum(is.na(RS39_pnps_function_KDe743ST$KDe743ST_pNpS_gene_reference))
#
percenta=100*(a/total_length)
percentb=100*(b/total_length)

##
sum(is.na(RS39_pnps_function_KDe743ST$corresponding_gene_call))
select_coverage_RS39_KDe743ST=coverage_RS39%>%dplyr::select("gene_callers_id","KDe743ST")
select_coverage_RS39_KDe743ST=select_coverage_RS39_KDe743ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_RS39_KDe743ST=left_join(select_coverage_RS39_KDe743ST, RS39_pnps_function_KDe743ST, by= "corresponding_gene_call")
select_coverage_RS39_KDe743ST$sample="KDe743ST"
select_coverage_RS39_KDe743ST=select_coverage_RS39_KDe743ST%>%dplyr::rename("pnps"="KDe743ST_pNpS_gene_reference", "coverage"="KDe743ST")
##
names(select_coverage_RS39_KDe743ST)

select_coverage_RS39_KMy813ST=coverage_RS39%>%dplyr::select("gene_callers_id","KMy813ST")
select_coverage_RS39_KMy813ST=select_coverage_RS39_KMy813ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_RS39_KMy813ST=left_join(select_coverage_RS39_KMy813ST, RS39_pnps_function_KMy813ST, by= "corresponding_gene_call")
select_coverage_RS39_KMy813ST$sample="KMy813ST"
select_coverage_RS39_KMy813ST=select_coverage_RS39_KMy813ST%>%dplyr::rename("pnps"="KMy813ST_pNpS_gene_reference", "coverage"="KMy813ST")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_RS39_KMy813ST, select_coverage_RS39_KDe743ST)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2


p_RS39_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.RS39, RS39")
p_RS39_cov


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p


p_RS39_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.RS39, RS39")
p_RS39_cov



p_RS39_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=coverage, color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ia.4.RS39, RS39")+ylab("log pN/pS per gene")+xlab("coverage per gene")
p_RS39_cor




sum(is.na(coverage_bothsamp$pnps))

coverage_bothsamp$log_pnps=log(coverage_bothsamp$pnps)

sum(is.na(coverage_bothsamp$log_pnps))
which(is.infinite(coverage_bothsamp$log_pnps))
coverage_bothsamp[c('log_pnps')][sapply(coverage_bothsamp[c('log_pnps')], is.infinite)] <- NA
sum(is.na(coverage_bothsamp$log_pnps))
coverage_bothsamp=coverage_bothsamp%>%drop_na(log_pnps)




my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log_pnps ~ corresponding_gene_call + sample + coverage, data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "RS39_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)
```

###PER SAMPLE 
###KDe743ST


```{r}
##ranking genes by pNpS per sample

RS39_pnps_function_ranks <- dplyr::arrange(RS39_pnps_function_by_sample, KDe743ST_pNpS_gene_reference)%>% dplyr::mutate(KDe743ST_rank = 1:length(KDe743ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe743ST_pNpS_gene_reference) is.na 
RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference[max(which(RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=RS39_pnps_function_ranks[which.min(abs(0.010-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.010$KDe743ST_rank
test_0.025=RS39_pnps_function_ranks[which.min(abs(0.025-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.025$KDe743ST_rank
test_0.050=RS39_pnps_function_ranks[which.min(abs(0.050-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.050$KDe743ST_rank
test_0.075=RS39_pnps_function_ranks[which.min(abs(0.075-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.075$KDe743ST_rank
test_0.10=RS39_pnps_function_ranks[which.min(abs(0.10-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.10$KDe743ST_rank
test_0.25=RS39_pnps_function_ranks[which.min(abs(0.25-RS39_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.25$KDe743ST_rank

order=as.numeric(test_0.010$KDe743ST_rank)
order=append(order, as.numeric(test_0.025$KDe743ST_rank))
order=append(order, as.numeric(test_0.050$KDe743ST_rank))
order=append(order, as.numeric(test_0.075$KDe743ST_rank))
order=append(order, as.numeric(test_0.10$KDe743ST_rank))
order=append(order, as.numeric(test_0.25$KDe743ST_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS39_pnps_function_ranks2=RS39_pnps_function_ranks%>%drop_na(KDe743ST_pNpS_gene_reference)
test=RS39_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

RS39_pnps_function_ranks2$Environment[is.na(RS39_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KDe743ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


library(seecolor)
#print_color(genes)
#names(RS39_pnps_function_ranks)
#RS39_pnps_function_ranks$genome="RS39"
#RS39_KDe743ST_map <- ggplot(data=RS39_pnps_function_ranks, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS39")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("RS39 KDe743ST")
#RS39_KDe743ST_map

##playing geom_line
RS39_pnps_function_ranks2$genome="RS39"
RS39_KDe743ST_map <- ggplot(data=RS39_pnps_function_ranks2, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS39")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("RS39 KDe743ST")  
  h=order
RS39_KDe743ST_map=RS39_KDe743ST_map+ geom_hline(yintercept = factor(RS39_pnps_function_ranks2$KDe743ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
RS39_KDe743ST_map

ggsave("RS39_KDe743ST.pdf", height=20, width=15)

```




RS39_KDe743ST_map
###PER SAMPLE 
###KMy813ST


```{r}
##ranking genes by pNpS per sample

RS39_pnps_function_ranks <- dplyr::arrange(RS39_pnps_function_by_sample, KMy813ST_pNpS_gene_reference)%>% dplyr::mutate(KMy813ST_rank = 1:length(KMy813ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy813ST_pNpS_gene_reference) is.na 
RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference[max(which(RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=RS39_pnps_function_ranks[which.min(abs(0.010-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.010$KMy813ST_rank
test_0.025=RS39_pnps_function_ranks[which.min(abs(0.025-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.025$KMy813ST_rank
test_0.050=RS39_pnps_function_ranks[which.min(abs(0.050-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.050$KMy813ST_rank
test_0.075=RS39_pnps_function_ranks[which.min(abs(0.075-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.075$KMy813ST_rank
test_0.10=RS39_pnps_function_ranks[which.min(abs(0.10-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.10$KMy813ST_rank
test_0.25=RS39_pnps_function_ranks[which.min(abs(0.25-RS39_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.25$KMy813ST_rank

order=as.numeric(test_0.010$KMy813ST_rank)
order=append(order, as.numeric(test_0.025$KMy813ST_rank))
order=append(order, as.numeric(test_0.050$KMy813ST_rank))
order=append(order, as.numeric(test_0.075$KMy813ST_rank))
order=append(order, as.numeric(test_0.10$KMy813ST_rank))
order=append(order, as.numeric(test_0.25$KMy813ST_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS39_pnps_function_ranks2=RS39_pnps_function_ranks%>%drop_na(KMy813ST_pNpS_gene_reference)

test=RS39_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

RS39_pnps_function_ranks2$Environment[is.na(RS39_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(KMy813ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
RS39_pnps_function_ranks2$genome="RS39"
RS39_KMy813ST_map <- ggplot(data=RS39_pnps_function_ranks2, aes(x = genome, y = factor(KMy813ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS39")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("RS39 KMy813ST")  
  h=order
RS39_KMy813ST_map=RS39_KMy813ST_map+ geom_hline(yintercept = factor(RS39_pnps_function_ranks2$KMy813ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
RS39_KMy813ST_map

ggsave("RS39_KMy813ST.pdf", height=20, width=15)
```


```{r}
pdf("RS39_twosamples.pdf", width=12, height=6.5)
plot_grid(RS39_KDe743ST_map,RS39_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("RS39_twosamples.svg", width=12, height=6.5)
plot_grid(RS39_KDe743ST_map,RS39_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```



###Mean RS39


```{r}
##ranking genes by pNpS per sample

names(RS39_pnps_function_by_sample)

RS39_pnps_function_ranks <- dplyr::arrange(RS39_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
RS39_pnps_function_ranks$pNpS.Mean[max(which(RS39_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=RS39_pnps_function_ranks[which.min(abs(0.010-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=RS39_pnps_function_ranks[which.min(abs(0.025-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=RS39_pnps_function_ranks[which.min(abs(0.050-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=RS39_pnps_function_ranks[which.min(abs(0.075-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=RS39_pnps_function_ranks[which.min(abs(0.10-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=RS39_pnps_function_ranks[which.min(abs(0.25-RS39_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank

order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))
order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS39_pnps_function_ranks2=RS39_pnps_function_ranks%>%drop_na(pNpS.Mean)

test=RS39_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""

RS39_pnps_function_ranks2$Environment[is.na(RS39_pnps_function_ranks2$Environment)] <- ""

test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
RS39_pnps_function_ranks2$genome="RS39"
RS39_Mean_map <- ggplot(data=RS39_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS39")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ia.4.RS39")  
  h=order
RS39_Mean_map=RS39_Mean_map+ geom_hline(yintercept = factor(RS39_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+ theme(legend.position = "none")+scale_fill_manual(values=c("#d3d3d3", "#ff0000"))+theme(axis.text.x = element_text(angle = 45, vjust = 1.5, size=10))+coord_flip()
RS39_Mean_map

ggsave("RS39_Mean.pdf", height=20, width=15)
ggsave("RS39_Mean.svg", height=20, width=15)
```




##############
####RS40 Ib
################

```{r}


pnps_RS40=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/pNpS.txt")
functions_RS40=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/RS40-functions.txt")
functions_RS40=functions_RS40%>%dplyr::filter(source=="COG20_FUNCTION"| source=="KOfam")
length(unique(functions_RS40$gene_callers_id))
#we have 1263 genes with functional annotations 


####get gene_caller_ids for all environment specific genes 
offshore_genes_RS40=left_join(functions_RS40, offshore_genes, by="accession")
dim(offshore_genes_RS40)
length(unique(offshore_genes_RS40$gene_callers_id))

names(offshore_genes_RS40)
offshore_genes_RS40=offshore_genes_RS40%>%dplyr::select(gene_callers_id, Category, accession, Metabolism, Specific.function, Gene.description,Environment, accession,	Environment,	source_figure4,	FUNCTION,	enrichment_score,	unadjusted_p_value,	adjusted_q_value,	associated_groups,	gene_clusters_ids,	p_Ia.4.II,	p_Ia.3.III,	p_Ia.3.VI,	p_Ia.3.II,	p_Ia.3.IV,	p_Ia.3.V,	p_Ia.4.1483,	p_Ia.1.I,	p_Ia.4.RS39,	p_Ib.1,	p_Ia.3.I,	Notes )%>%filter(Environment=="Offshore")
dim(offshore_genes_RS40)


offshore_genes_RS40=distinct(offshore_genes_RS40,gene_callers_id, .keep_all=TRUE)
write.csv(offshore_genes_RS40, "offshore_genes_RS40.csv")
###lets use this gene calls matrix which has all the functional annotation per gene per row 
functions_RS40=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/RS40-gene_calls.txt")
length(unique(functions_RS40$gene_callers_id))
#we have 1436 genes 
names(functions_RS40)
functions_RS40_offshore_specific=left_join(functions_RS40,offshore_genes_RS40)

```





##now we need to add in PN/PS
###Do I include the concatenated when I calculate the mean???
```{r}

functions_RS40_offshore_specific_join=functions_RS40_offshore_specific%>%rename_all(recode,"gene_callers_id"="corresponding_gene_call")



RS40_pnps_function=left_join(pnps_RS40, functions_RS40_offshore_specific_join, by="corresponding_gene_call")
length(unique(RS40_pnps_function$corresponding_gene_call))
##1309 genes have met the coverage criteria

##pNpS values were Inf instead of just NA, will replace them with NA 
which(is.infinite(RS40_pnps_function$pNpS_gene_reference))
RS40_pnps_function[c('pNpS_gene_reference')][sapply(RS40_pnps_function[c('pNpS_gene_reference')], is.infinite)] <- NA

##lets look at mean and sd
names(RS40_pnps_function)
#within samples pn/ps varation
pNpS_within_sample_means=RS40_pnps_function%>% dplyr::group_by(sample_id)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE), pNpS.median=median(pNpS_gene_reference, na.rm = TRUE),pNpS.min=min(pNpS_gene_reference, na.rm = TRUE), pNpS.max=max(pNpS_gene_reference, na.rm = TRUE) )
pNpS_within_sample_means
write.csv(pNpS_within_sample_means, "pNpS_within_sample_means_RS40.csv")



##
pNpS_within_sample_Quantiles=RS40_pnps_function %>% 
  dplyr::group_by(sample_id) %>% 
  dplyr::group_modify(~as.data.frame(t(quantile(.$pNpS_gene_reference, na.rm = TRUE))))
write.csv(pNpS_within_sample_Quantiles, "pNpS_within_sample_Quantiles_RS40.csv")

#across samples pn/ps mean and variation (per gene)
pNpS_across_samples=RS40_pnps_function%>% dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_samples

unique(RS40_pnps_function$sample_id)

hist(pNpS_across_samples$pNpS.sd)

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")
p
ggsave("Minimal_pnps_diff_RS40.png")

p<-ggplot(pNpS_across_samples, aes(x=pNpS.sd)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Minimal between sample variation in pNpS across metagenomic samples ")+xlim(0,0.025)
p
ggsave("Minimal_pnps_diff_RS40_0.25.png")

p<-ggplot(RS40_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of RS40 genes ")+facet_wrap(~sample_id)
p


p<-ggplot(RS40_pnps_function, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("Strong purifying selection for the vast majority of RS40 genes ")+facet_wrap(~sample_id)+xlim(0,0.1)
p


library(scales)

across_samples_RS40<- ggplot(data=RS40_pnps_function, aes(x=sample_id, y=pNpS_gene_reference, color=sample_id))+geom_violin() 
across_samples_RS40
across_samples_RS40=across_samples_RS40 + theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_samples_RS40


###lets add mean and sd pNpS to the larger dataset 
#remove diel sample and concentated sample, get mean and join that to the main table 
pNpS_across_select_samples=RS40_pnps_function%>%dplyr::filter(sample_id=="KDe743ST" | sample_id=="KMy813ST") %>%dplyr::group_by(corresponding_gene_call)%>%dplyr:: summarise(pNpS.mean.n = n(), pNpS.Mean=mean(pNpS_gene_reference, na.rm = TRUE), pNpS.sd=sd(pNpS_gene_reference, na.rm = TRUE))
pNpS_across_select_samples


RS40_pnps_function=left_join(RS40_pnps_function, pNpS_across_select_samples, by="corresponding_gene_call")

```


```{r}

unique(RS40_pnps_function$Environment)
RS40_pnps_function_reporting=RS40_pnps_function%>%filter(sample_id=="KMy813ST"| sample_id=="KDe743ST")
sum(is.na(RS40_pnps_function_reporting$pNpS_gene_reference))
RS40_pnps_function_reporting=RS40_pnps_function_reporting%>%drop_na(pNpS_gene_reference)
RS40_pnps_function_reporting$log_pnps=log(RS40_pnps_function_reporting$pNpS_gene_reference)

sum(is.na(RS40_pnps_function_reporting$log_pnps))
which(is.infinite(RS40_pnps_function_reporting$log_pnps))
RS40_pnps_function_reporting[c('log_pnps')][sapply(RS40_pnps_function_reporting[c('log_pnps')], is.infinite)] <- NA
sum(is.na(RS40_pnps_function_reporting$log_pnps))
RS40_pnps_function_reporting=RS40_pnps_function_reporting%>%drop_na(log_pnps)

hist(RS40_pnps_function_reporting$log_pnps)



p<-ggplot(RS40_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(RS40_pnps_function_reporting, aes(x=pNpS_gene_reference)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(RS40_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample_id)
p

p<-ggplot(RS40_pnps_function_reporting, aes(x=log_pnps)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p




```



```{r}

####let's break out pnps by samples 
backbone_RS40_pnps_function=distinct(RS40_pnps_function, corresponding_gene_call, .keep_all = TRUE)

dim(backbone_RS40_pnps_function)
dim(functions_RS40_offshore_specific)
##the difference comes from genes that didnÊ»t meet the coverage requirment  
backbone_RS40_pnps_function=backbone_RS40_pnps_function%>%dplyr::select(!("sample_id"))%>%dplyr::select(!("pNpS_gene_reference"))%>%dplyr::select(!("X"))
dim(backbone_RS40_pnps_function)


###subset and rename to have sample level information
unique(RS40_pnps_function$sample_id)
RS40_pnps_function_KDe743ST=RS40_pnps_function%>%filter(sample_id=="KDe743ST")
names(RS40_pnps_function_KDe743ST)
RS40_pnps_function_KDe743ST=RS40_pnps_function_KDe743ST%>%dplyr::rename("KDe743ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KDe743ST_pNpS_gene_reference")


RS40_pnps_function_KMy813ST=RS40_pnps_function%>%filter(sample_id=="KMy813ST")
RS40_pnps_function_KMy813ST=RS40_pnps_function_KMy813ST%>%dplyr::rename("KMy813ST_pNpS_gene_reference"="pNpS_gene_reference")%>%dplyr::select("corresponding_gene_call", "KMy813ST_pNpS_gene_reference")



RS40_pnps_function_by_sample=left_join( backbone_RS40_pnps_function, RS40_pnps_function_KDe743ST,by="corresponding_gene_call")
RS40_pnps_function_by_sample=left_join( RS40_pnps_function_by_sample, RS40_pnps_function_KMy813ST,by="corresponding_gene_call")

write.csv(RS40_pnps_function_by_sample, "RS40_pnps_function_by_sample.csv")



###and bring in coverage info 
coverage_RS40=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/RS40-gene_non_outlier_coverages.txt")
coverage_RS40=coverage_RS40%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
RS40_pnps_function_by_sample=left_join(RS40_pnps_function_by_sample, coverage_RS40, by= "corresponding_gene_call")



p_KDe743ST<-ggplot(RS40_pnps_function_by_sample, aes(x=KDe743ST_pNpS_gene_reference, y=KDe743ST )) + 
  geom_point(color="black", fill="white")+ggtitle("Coverage vs pNpS RS40")+ylab("Non-outlier gene coverage")
p_KDe743ST


p_KMy813ST<-ggplot(RS40_pnps_function_by_sample, aes(x=KMy813ST_pNpS_gene_reference, y=KMy813ST )) + 
  geom_point(color="black", fill="white")+ggtitle("")+ylab("Non-outlier gene coverage")+ theme(legend.title = element_blank()) + ylab("Coverage") +xlab("pNpS") +geom_jitter(size=0.05)+ theme(axis.text=element_text(size=10),axis.title=element_text(size=10,face="bold"), legend.text=element_text(size=10), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
p_KMy813ST
ggsave("coverage.svg", width=2.5, height=2.5)

```



```{r}
###and bring in coverage info 
coverage_RS40=read.delim("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/RS40-gene_non_outlier_coverages.txt")
total_length=length(unique(coverage_RS40$gene_callers_id))
##
a=length(unique(RS40_pnps_function_KMy813ST$corresponding_gene_call))-sum(is.na(RS40_pnps_function_KMy813ST$KMy813ST_pNpS_gene_reference))
##
b=length(unique(RS40_pnps_function_KDe743ST$corresponding_gene_call))-sum(is.na(RS40_pnps_function_KDe743ST$KDe743ST_pNpS_gene_reference))

percenta=100*(a/total_length)
percentb=100*(b/total_length)

###
sum(is.na(RS40_pnps_function_KDe743ST$corresponding_gene_call))
select_coverage_RS40_KDe743ST=coverage_RS40%>%dplyr::select("gene_callers_id","KDe743ST")
select_coverage_RS40_KDe743ST=select_coverage_RS40_KDe743ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_RS40_KDe743ST=left_join(select_coverage_RS40_KDe743ST, RS40_pnps_function_KDe743ST, by= "corresponding_gene_call")
select_coverage_RS40_KDe743ST$sample="KDe743ST"
select_coverage_RS40_KDe743ST=select_coverage_RS40_KDe743ST%>%dplyr::rename("pnps"="KDe743ST_pNpS_gene_reference", "coverage"="KDe743ST")
##
names(select_coverage_RS40_KDe743ST)

select_coverage_RS40_KMy813ST=coverage_RS40%>%dplyr::select("gene_callers_id","KMy813ST")
select_coverage_RS40_KMy813ST=select_coverage_RS40_KMy813ST%>%dplyr::rename("corresponding_gene_call"="gene_callers_id")
select_coverage_RS40_KMy813ST=left_join(select_coverage_RS40_KMy813ST, RS40_pnps_function_KMy813ST, by= "corresponding_gene_call")
select_coverage_RS40_KMy813ST$sample="KMy813ST"
select_coverage_RS40_KMy813ST=select_coverage_RS40_KMy813ST%>%dplyr::rename("pnps"="KMy813ST_pNpS_gene_reference", "coverage"="KMy813ST")

library(ggpmisc)

coverage_bothsamp=rbind(select_coverage_RS40_KMy813ST, select_coverage_RS40_KDe743ST)
coverage_bothsamp=coverage_bothsamp%>%drop_na(pnps)%>%drop_na(coverage)
max_coverage=max(coverage_bothsamp$coverage)
max_pnps=max(coverage_bothsamp$pnps)
max_pnps=2

p_RS40_cov<-ggplot(coverage_bothsamp, aes(y=pnps, x=coverage, color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("pN/pS per gene")+xlab("coverage per gene")+xlim(0,max_coverage)+ylim(0,max_pnps)+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ib.1, RS40")
p_RS40_cov




p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=coverage)) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p

p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")
p


p<-ggplot(coverage_bothsamp, aes(x=log(coverage))) + 
  geom_histogram(color="black", fill="white")+ggtitle("")+facet_wrap(~sample)
p



p_RS40_cov<-ggplot(coverage_bothsamp, aes(y=log(pnps), x=log(coverage), color=sample )) + 
  geom_point(alpha=0.3)+ggtitle("")+ylab("log pN/pS per gene")+xlab("log coverage per gene")+
  stat_poly_line() +
  stat_poly_eq(aes(label =  paste(after_stat(adj.rr.label),after_stat(f.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*"))) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ib.1, RS40")
p_RS40_cov




p_RS40_cor=ggplot(coverage_bothsamp, aes(y=log(pnps), x=log(coverage), color=sample )) +
  geom_point() +
  stat_correlation(aes(label =  paste(after_stat(r.label),after_stat(t.value.label),after_stat(p.value.label), 
                                  sep = "*\", \"*")))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+ggtitle("Ib.1, RS40")+ylab("log pN/pS per gene")+xlab("log coverage per gene")
p_RS40_cor



sum(is.na(coverage_bothsamp$pnps))

coverage_bothsamp$log_pnps=log(coverage_bothsamp$pnps)

sum(is.na(coverage_bothsamp$log_pnps))
which(is.infinite(coverage_bothsamp$log_pnps))
coverage_bothsamp[c('log_pnps')][sapply(coverage_bothsamp[c('log_pnps')], is.infinite)] <- NA
sum(is.na(coverage_bothsamp$log_pnps))
coverage_bothsamp=coverage_bothsamp%>%drop_na(log_pnps)


my_anova <- coverage_bothsamp%>% mutate(corresponding_gene_call = as.factor(corresponding_gene_call)) %>% lm(log_pnps ~ corresponding_gene_call + sample + log(coverage), data = .) %>% anova()
gene_explained <- my_anova$`Sum Sq`[1] / sum(my_anova$`Sum Sq`) * 100
sample_explained <- my_anova$`Sum Sq`[2] / sum(my_anova$`Sum Sq`) * 100
coverage_explained <- my_anova$`Sum Sq`[3] / sum(my_anova$`Sum Sq`) * 100

print(my_anova)
write.csv(my_anova, "RS40_my_anova_coverage.csv")
print(gene_explained)
print(sample_explained)
print(coverage_explained)
```

###Bring together all the coverage information

```{r}
pdf("coverage_all_genome_reps_pvalue_log.pdf", width=10, height=13)
plot_grid(p_HIMB5_cov,p_HIMB4_cov,p_HIMB1556_cov,p_HIMB083_cov,p_HIMB1437_cov,p_RS39_cov,p_RS40_cov, align = 'hv',label_size = 12, ncol = 2, nrow=4)
dev.off()

svglite::svglite("coverage_all_genome_reps_pvalue_log.svg", width=10, height=12)
plot_grid(p_HIMB5_cov,p_HIMB4_cov,p_HIMB1556_cov,p_HIMB083_cov,p_HIMB1437_cov,p_RS39_cov,p_RS40_cov, align = 'hv',label_size = 12, ncol = 2, nrow=4)
dev.off()

library(cowplot)
pdf("correlation_coverage_all_genome_reps_pvalue_log.pdf", width=10, height=13)
plot_grid(p_HIMB5_cor,p_HIMB4_cor,p_HIMB1556_cor,p_HIMB083_cor,p_HIMB1437_cor,p_RS39_cor,p_RS40_cor, align = 'hv',label_size = 12, ncol = 2, nrow=4)
dev.off()

svglite::svglite("correlation_coverage_genome_reps_pvalue_log.svg", width=10, height=12)
plot_grid(p_HIMB5_cor,p_HIMB4_cor,p_HIMB1556_cor,p_HIMB083_cor,p_HIMB1437_cor,p_RS39_cor,p_RS40_cor, align = 'hv',label_size = 12, ncol = 2, nrow=4)
dev.off()
```



###PER SAMPLE 
###KDe743ST


```{r}
##ranking genes by pNpS per sample

RS40_pnps_function_ranks <- dplyr::arrange(RS40_pnps_function_by_sample, KDe743ST_pNpS_gene_reference)%>% dplyr::mutate(KDe743ST_rank = 1:length(KDe743ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KDe743ST_pNpS_gene_reference) is.na 
RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference[max(which(RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference > 0), na.rm = TRUE)]

###identify scale 
test_0.010=RS40_pnps_function_ranks[which.min(abs(0.010-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.010$KDe743ST_rank
test_0.025=RS40_pnps_function_ranks[which.min(abs(0.025-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.025$KDe743ST_rank
test_0.050=RS40_pnps_function_ranks[which.min(abs(0.050-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.050$KDe743ST_rank
test_0.075=RS40_pnps_function_ranks[which.min(abs(0.075-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.075$KDe743ST_rank
test_0.10=RS40_pnps_function_ranks[which.min(abs(0.10-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.10$KDe743ST_rank
test_0.25=RS40_pnps_function_ranks[which.min(abs(0.25-RS40_pnps_function_ranks$KDe743ST_pNpS_gene_reference)),]
test_0.25$KDe743ST_rank


order=as.numeric(test_0.010$KDe743ST_rank)
order=append(order, as.numeric(test_0.025$KDe743ST_rank))
order=append(order, as.numeric(test_0.050$KDe743ST_rank))
order=append(order, as.numeric(test_0.075$KDe743ST_rank))
order=append(order, as.numeric(test_0.10$KDe743ST_rank))
order=append(order, as.numeric(test_0.25$KDe743ST_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")

##need to check that environment is not defined in these calles and then 

#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS40_pnps_function_ranks2=RS40_pnps_function_ranks%>%drop_na(KDe743ST_pNpS_gene_reference)
test=RS40_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""
RS40_pnps_function_ranks2$Environment[is.na(RS40_pnps_function_ranks2$Environment)] <- ""

test=test%>%arrange(KDe743ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


library(seecolor)
#print_color(genes)
#names(RS40_pnps_function_ranks)
#RS40_pnps_function_ranks$genome="RS40"
#RS40_KDe743ST_map <- ggplot(data=RS40_pnps_function_ranks, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS40")+  scale_y_discrete(labels=test$order_in_group)+ggtitle("RS40 KDe743ST")
#RS40_KDe743ST_map

##playing geom_line
RS40_pnps_function_ranks2$genome="RS40"
RS40_KDe743ST_map <- ggplot(data=RS40_pnps_function_ranks2, aes(x = genome, y = factor(KDe743ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS40")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("RS40 KDe743ST")  
  h=order
RS40_KDe743ST_map=RS40_KDe743ST_map+ geom_hline(yintercept = factor(RS40_pnps_function_ranks2$KDe743ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
RS40_KDe743ST_map

ggsave("RS40_KDe743ST.pdf", height=20, width=15)

```



###PER SAMPLE 
###KMy813ST


```{r}
##ranking genes by pNpS per sample

RS40_pnps_function_ranks <- dplyr::arrange(RS40_pnps_function_by_sample, KMy813ST_pNpS_gene_reference)%>% dplyr::mutate(KMy813ST_rank = 1:length(KMy813ST_pNpS_gene_reference))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (KMy813ST_pNpS_gene_reference) is.na 
RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference[max(which(RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference > 0), na.rm = TRUE)]


###identify scale 
test_0.010=RS40_pnps_function_ranks[which.min(abs(0.010-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.010$KMy813ST_rank
test_0.025=RS40_pnps_function_ranks[which.min(abs(0.025-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.025$KMy813ST_rank
test_0.050=RS40_pnps_function_ranks[which.min(abs(0.050-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.050$KMy813ST_rank
test_0.075=RS40_pnps_function_ranks[which.min(abs(0.075-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.075$KMy813ST_rank
test_0.10=RS40_pnps_function_ranks[which.min(abs(0.10-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.10$KMy813ST_rank
test_0.25=RS40_pnps_function_ranks[which.min(abs(0.25-RS40_pnps_function_ranks$KMy813ST_pNpS_gene_reference)),]
test_0.25$KMy813ST_rank

order=as.numeric(test_0.010$KMy813ST_rank)
order=append(order, as.numeric(test_0.025$KMy813ST_rank))
order=append(order, as.numeric(test_0.050$KMy813ST_rank))
order=append(order, as.numeric(test_0.075$KMy813ST_rank))
order=append(order, as.numeric(test_0.10$KMy813ST_rank))
order=append(order, as.numeric(test_0.25$KMy813ST_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25" )


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS40_pnps_function_ranks2=RS40_pnps_function_ranks%>%drop_na(KMy813ST_pNpS_gene_reference)
test=RS40_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""
RS40_pnps_function_ranks2$Environment[is.na(RS40_pnps_function_ranks2$Environment)] <- ""

test=test%>%arrange(KMy813ST_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
RS40_pnps_function_ranks2$genome="RS40"
RS40_KMy813ST_map <- ggplot(data=RS40_pnps_function_ranks2, aes(x = genome, y = factor(KMy813ST_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS40")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("RS40 KMy813ST")  
  h=order
RS40_KMy813ST_map=RS40_KMy813ST_map+ geom_hline(yintercept = factor(RS40_pnps_function_ranks2$KMy813ST_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")
RS40_KMy813ST_map

ggsave("RS40_KMy813ST.pdf", height=20, width=15)
```




```{r}
pdf("RS40_twosamples.pdf", width=12, height=6.5)
plot_grid(RS40_KDe743ST_map,RS40_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()

svglite::svglite("RS40_twosamples.svg", width=12, height=6.5)
plot_grid(RS40_KDe743ST_map,RS40_KMy813ST_map, align = 'hv',label_size = 12, ncol = 2, nrow=1)
dev.off()
```


###Mean RS40


```{r}
##ranking genes by pNpS per sample

names(RS40_pnps_function_by_sample)

RS40_pnps_function_ranks <- dplyr::arrange(RS40_pnps_function_by_sample, pNpS.Mean)%>% dplyr::mutate(Mean_rank = 1:length(pNpS.Mean))


##note NAs are put last and ranked in this way. So need to consider removing these ranks when plotting by ranks, using a filter of the sample pNps (pNpS.Mean) is.na 
RS40_pnps_function_ranks$pNpS.Mean[max(which(RS40_pnps_function_ranks$pNpS.Mean > 0), na.rm = TRUE)]


###identify scale 
test_0.010=RS40_pnps_function_ranks[which.min(abs(0.010-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.010$Mean_rank
test_0.025=RS40_pnps_function_ranks[which.min(abs(0.025-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.025$Mean_rank
test_0.050=RS40_pnps_function_ranks[which.min(abs(0.050-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.050$Mean_rank
test_0.075=RS40_pnps_function_ranks[which.min(abs(0.075-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.075$Mean_rank
test_0.10=RS40_pnps_function_ranks[which.min(abs(0.10-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.10$Mean_rank
test_0.25=RS40_pnps_function_ranks[which.min(abs(0.25-RS40_pnps_function_ranks$pNpS.Mean)),]
test_0.25$Mean_rank


order=as.numeric(test_0.010$Mean_rank)
order=append(order, as.numeric(test_0.025$Mean_rank))
order=append(order, as.numeric(test_0.050$Mean_rank))
order=append(order, as.numeric(test_0.075$Mean_rank))
order=append(order, as.numeric(test_0.10$Mean_rank))
order=append(order, as.numeric(test_0.25$Mean_rank))

order
breaks=c("0.01", "0.025", "0.050", "0.075", "0.10", "0.25")


#genes=c("none"="darkgray", "Ribosomal"="orchid", "broader_Pelagibacteraceae"="lightblue", "offshore_Specific"="yellow", "Break"="black")
RS40_pnps_function_ranks2=RS40_pnps_function_ranks%>%drop_na(pNpS.Mean)
test=RS40_pnps_function_ranks2
test$Specific.function[is.na(test$Specific.function)] <- ""
RS40_pnps_function_ranks2$Environment[is.na(RS40_pnps_function_ranks2$Environment)] <- ""
test=test%>%
  arrange(Mean_rank)%>%mutate(order_in_group=as.factor(Specific.function))


##playing geom_line
RS40_pnps_function_ranks2$genome="RS40"
RS40_Mean_map <- ggplot(data=RS40_pnps_function_ranks2, aes(x = genome, y = factor(Mean_rank)))+ geom_tile(aes( fill=Environment))+ggtitle("RS40")+ scale_y_discrete(labels=test$order_in_group)+ggtitle("Ib.1")  
  h=order
RS40_Mean_map=RS40_Mean_map+ geom_hline(yintercept = factor(RS40_pnps_function_ranks2$Mean_rank[order]), 
              color = "#6497bf", 
             lwd = 0.25) +  annotate("text", x= 1, y=h, label=breaks, size=3, color="#6497bf")+ylab("")+xlab("")+ theme(legend.position = "none")+ scale_fill_manual(values=c("#d3d3d3", "#ff0000")) + theme(legend.position = "none")+theme(axis.text.x = element_text(angle = 45, vjust = 1.5, size=10))+coord_flip()
RS40_Mean_map

ggsave("RS40_Mean.pdf", height=20, width=15)
ggsave("RS40_Mean.svg", height=20, width=15)
```


#####
Bring together offshore
####

```{r}
pdf("offshore_means_pnps.pdf", width=16, height=16)
plot_grid(HIMB083_Mean_map, HIMB1437_Mean_map, RS39_Mean_map, RS40_Mean_map,align = 'hv',label_size = 12, ncol = 3, nrow=2)
dev.off()

svglite::svglite("offshore_means_pnps.svg", width=16, height=16)
plot_grid(HIMB083_Mean_map, HIMB1437_Mean_map, RS39_Mean_map, RS40_Mean_map,align = 'hv',label_size = 12, ncol = 3, nrow=2)
dev.off()

```

```{r}
view(RS40_pnps_function_ranks2)
names(RS40_pnps_function_ranks2)
to_save_RS40=RS40_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_RS39=RS39_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_HIMB1437=HIMB1437_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_HIMB083=HIMB083_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_HIMB5=HIMB5_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_HIMB4=HIMB4_pnps_function_ranks2%>%drop_na(Specific.function)
to_save_HIMB1556=HIMB1556_pnps_function_ranks2%>%drop_na(Specific.function)
dim(to_save_RS40)
dim(to_save_RS39)

to_save_pnps_offshore=rbind(to_save_RS40,to_save_RS39, to_save_HIMB1437,to_save_HIMB083)
write.csv(to_save_pnps_offshore, "to_save_pnps_offshore.csv")
to_save_pnps_coastal=rbind(to_save_HIMB5,to_save_HIMB4, to_save_HIMB1556)
write.csv(to_save_pnps_coastal, "to_save_pnps_coastal.csv")
```





#####
Bring together all
####

```{r}

pdf("all_means_pnps_4_shortest_5.5.pdf", width=5.5, height=10)
plot_grid(HIMB1556_Mean_map,HIMB083_Mean_map,HIMB1437_Mean_map,RS39_Mean_map, HIMB5_Mean_map,HIMB4_Mean_map,RS40_Mean_map,align = 'hv',label_size = 12, ncol = 1, nrow=7)
dev.off()

svglite::svglite("all_means_pnps_4_shortest_5.5.svg", width=5.5, height=10)
plot_grid(HIMB1556_Mean_map,HIMB083_Mean_map,HIMB1437_Mean_map,RS39_Mean_map, HIMB5_Mean_map,HIMB4_Mean_map,RS40_Mean_map, align = 'hv',label_size = 12, ncol = 1, nrow=7)
dev.off()
```






#####
Bring together all coastal
####

```{r}
pdf("Coastal_means_pnps.pdf", width=16, height=8.5)
plot_grid(HIMB1556_Mean_map,HIMB5_Mean_map, HIMB4_Mean_map, align = 'hv',label_size = 12, ncol = 3, nrow=1)
dev.off()

svglite::svglite("Coastal_means_pnps.svg", width=16, height=8.5)
plot_grid(HIMB1556_Mean_map,HIMB5_Mean_map, HIMB4_Mean_map, align = 'hv',label_size = 12, ncol = 3, nrow=1)
dev.off()

```





```{r}
names(RS40_pnps_function_by_sample)
RS40_pnps_tog=RS40_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd", "Specific.function")%>%dplyr::rename("RS40_pNpS.Mean"="pNpS.Mean","RS40_pNpS.sd"="pNpS.sd")
RS40_pnps_tog


RS39_pnps_tog=RS39_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("RS39_pNpS.Mean"="pNpS.Mean","RS39_pNpS.sd"="pNpS.sd")

HIMB083_pnps_tog=HIMB083_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("HIMB083_pNpS.Mean"="pNpS.Mean","HIMB083_pNpS.sd"="pNpS.sd")


HIMB1437_pnps_tog=HIMB1437_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("HIMB1437_pNpS.Mean"="pNpS.Mean","HIMB1437_pNpS.sd"="pNpS.sd")



HIMB1556_pnps_tog=HIMB1556_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("HIMB1556_pNpS.Mean"="pNpS.Mean","HIMB1556_pNpS.sd"="pNpS.sd")


HIMB5_pnps_tog=HIMB5_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("HIMB5_pNpS.Mean"="pNpS.Mean","HIMB5_pNpS.sd"="pNpS.sd")


HIMB4_pnps_tog=HIMB4_pnps_function_by_sample%>%drop_na("Specific.function")%>%dplyr::select("accession", "pNpS.Mean","pNpS.sd")%>%dplyr::rename("HIMB4_pNpS.Mean"="pNpS.Mean","HIMB4_pNpS.sd"="pNpS.sd")

#enviro_specific=read.csv("figure_5_may212024.csv")
enviro_specific_pnps=enviro_specific
enviro_specific_pnps=full_join(enviro_specific_pnps,RS40_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,RS39_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,HIMB083_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,HIMB1556_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,HIMB5_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,HIMB1437_pnps_tog, by="accession")
enviro_specific_pnps=full_join(enviro_specific_pnps,HIMB5_pnps_tog, by="accession")
write.csv(enviro_specific_pnps, "enviro_specific_pnps.csv")
```



####Table 2 
###lets look at pNpS values for genes that might important to the Sar11 core


```{r}
names(HIMB5_pnps_function_ranks)

RS40_pnps_function_ranks_top100=RS40_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe743ST_pNpS_gene_reference", "KMy813ST_pNpS_gene_reference")
RS40_pnps_function_ranks_top100$genome="RS40"
range(RS40_pnps_function_ranks_top100$pNpS.Mean)
RS39_pnps_function_ranks_top100=RS39_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe743ST_pNpS_gene_reference", "KMy813ST_pNpS_gene_reference")
RS39_pnps_function_ranks_top100$genome="RS39"
range(RS39_pnps_function_ranks_top100$pNpS.Mean)
HIMB083_pnps_function_ranks_top100=HIMB083_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe743ST_pNpS_gene_reference", "KMy813ST_pNpS_gene_reference")
HIMB083_pnps_function_ranks_top100$genome="HIMB083"
range(HIMB083_pnps_function_ranks_top100$pNpS.Mean)
HIMB1437_pnps_function_ranks_top100=HIMB1437_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe743ST_pNpS_gene_reference", "KMy813ST_pNpS_gene_reference")
HIMB1437_pnps_function_ranks_top100$genome="HIMB1437"
range(HIMB1437_pnps_function_ranks_top100$pNpS.Mean)
HIMB5_pnps_function_ranks_top100=HIMB5_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe747HP_pNpS_gene_reference","KMy817HP_pNpS_gene_reference" )
HIMB5_pnps_function_ranks_top100$genome="HIMB5"
range(HIMB5_pnps_function_ranks_top100$pNpS.Mean)
HIMB4_pnps_function_ranks_top100=HIMB4_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe747HP_pNpS_gene_reference","KMy817HP_pNpS_gene_reference")
HIMB4_pnps_function_ranks_top100$genome="HIMB4"
range(HIMB4_pnps_function_ranks_top100$pNpS.Mean)
HIMB1556_pnps_function_ranks_top100=HIMB1556_pnps_function_ranks%>%dplyr::select("corresponding_gene_call", "pNpS.Mean", "pNpS.sd", "Mean_rank","Specific.function", "COG20_FUNCTION","COG20_FUNCTION..ACCESSION.", "KOfam", "KOfam..ACCESSION.", "KDe747HP_pNpS_gene_reference","KMy817HP_pNpS_gene_reference")
HIMB1556_pnps_function_ranks_top100$genome="HIMB1556"
range(HIMB1556_pnps_function_ranks_top100$pNpS.Mean)
offshore_pnps_function_ranks_top100=rbind(RS40_pnps_function_ranks_top100,RS39_pnps_function_ranks_top100,HIMB083_pnps_function_ranks_top100,HIMB1437_pnps_function_ranks_top100)
offshore_pnps_function_ranks_top100=offshore_pnps_function_ranks_top100%>%dplyr::rename("Dec_samp"="KDe743ST_pNpS_gene_reference", "May_samp"="KMy813ST_pNpS_gene_reference")

coastal_pnps_function_ranks_top100=rbind(HIMB5_pnps_function_ranks_top100, HIMB4_pnps_function_ranks_top100, HIMB1556_pnps_function_ranks_top100)
coastal_pnps_function_ranks_top100=coastal_pnps_function_ranks_top100%>%dplyr::rename("Dec_samp"="KDe747HP_pNpS_gene_reference", "May_samp"="KMy817HP_pNpS_gene_reference")


tog_pnps_function_ranks_top100=rbind(coastal_pnps_function_ranks_top100, offshore_pnps_function_ranks_top100)



across_genomes<- ggplot(data=tog_pnps_function_ranks_top100, aes(x=genome, y=pNpS.Mean, color=genome))+geom_violin() 
across_genomes
across_genomes=across_genomes+ theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
across_genomes
ggsave("pNpS_accross_genomes.pdf")

across_genomes<- ggplot(data=tog_pnps_function_ranks_top100, aes(x=genome, y=pNpS.Mean, color=genome))+geom_violin() 
across_genomes
across_genomes=across_genomes+ theme(legend.title = element_blank()) + ylab("pNpS") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) +  theme(legend.position="none")+ylim(0,1)
across_genomes

library(ggridges)

colors_genomes=c("HIMB5"  = "#87ceeb",  "HIMB4"="#adff2f"  , "HIMB083"="#ba55d3","HIMB1556"="#0000cd","HIMB1437"="#ff8c00" , "RS39"="#ffd700", "RS40"="#dda0dd")

tog_pnps_function_ranks_top100$genome <- factor(tog_pnps_function_ranks_top100$genome, levels=c("HIMB1556","HIMB083", "HIMB1437", "RS39", "HIMB5", "HIMB4", "RS40"))

tog_pnps_function_ranks_top100$genome <- factor(tog_pnps_function_ranks_top100$genome, levels=c("RS40","HIMB4", "HIMB5",  "RS39", "HIMB1437", "HIMB083","HIMB1556"))
svg("Density_plots_SAR11_upto1.svg", width=3.2, height=1.9)
ggplot(tog_pnps_function_ranks_top100, aes(x = pNpS.Mean, y = genome, fill=genome)) + geom_density_ridges()+xlim(0,0.3)+ theme_ridges()+scale_fill_manual(values=(colors_genomes))+  theme(legend.position="none")

ggsave("Density_plots_SAR11_upto1.pdf", width=3.2, height=1.9)
dev.off()

###



```


```{r}

length(unique(tog_pnps_function_ranks_top100$COG20_FUNCTION..ACCESSION.))
length(unique(tog_pnps_function_ranks_top100$KOfam..ACCESSION.))


all_pnps_function_ranks_top100_test=reshape2::dcast(tog_pnps_function_ranks_top100, KOfam..ACCESSION.~ genome)

##okay lets turn this into a binary matrix 
samp2 <- all_pnps_function_ranks_top100_test[,-1]
rownames(samp2) <-all_pnps_function_ranks_top100_test[,1]

samp2[samp2>0]=1
sum=rowSums(samp2)
summary(sum)
hist(sum)

samp2$sum=rowSums(samp2)


function_all=tog_pnps_function_ranks_top100%>%dplyr::select( "KOfam","KOfam..ACCESSION.")
function_all=distinct(function_all,KOfam..ACCESSION., .keep_all=TRUE)

samp2$KOfam..ACCESSION.=rownames(samp2)
test=left_join(samp2,function_all, by= "KOfam..ACCESSION.")
test2=test%>%filter(sum==7)
test2=test2[test2$KOfam..ACCESSION. != "",]
listy=test2$KOfam..ACCESSION.
listy
write.csv(test, "Shared_KOfams_pnps.csv")
```



```{r}


function_all=tog_pnps_function_ranks_top100%>%dplyr::select("genome", "KOfam..ACCESSION.", "Dec_samp", "May_samp", "KOfam")
function_all=function_all[function_all$KOfam..ACCESSION. != "",]
function_all_melt=melt(function_all)
function_all_melt=function_all_melt%>%dplyr::rename("pnps"="value")

function_all_melt_2 <-function_all_melt[function_all_melt$KOfam..ACCESSION. %in% listy,]


function_all_test=reshape2::dcast(function_all_melt_2, KOfam..ACCESSION.~genome+variable, fun.aggregate = mean, drop=FALSE)
dim(function_all_test)
function_all_test$mean=rowMeans(function_all_test[,-1,], na.rm=TRUE)


library(matrixStats)
function_all_test=function_all_test%>%mutate(sd = rowSds(as.matrix(.[,-1]), na.rm=TRUE)) 


function_all=tog_pnps_function_ranks_top100%>%dplyr::select( "KOfam","KOfam..ACCESSION.")
function_all=distinct(function_all,KOfam..ACCESSION., .keep_all=TRUE)

function_all_test=left_join(function_all_test,function_all, by= "KOfam..ACCESSION.")


##lets look at what is has the highest purifying selection among our genome representatives, so we can compare it to what it found core among the top 25 of the Core Pelagibacteraceae (Considering what is core among all 92 genomes)
core_pnps_low=function_all_test%>%filter(mean <0.021)
core_pnps_low2=core_pnps_low%>%dplyr::select("KOfam..ACCESSION.", "KOfam", "mean", "sd")
write.csv(core_pnps_low2, "core_pnps_low_wiggle_6.csv")
```


```{r}
library(dplyr)
library(tidyverse)
ggplot(function_all_test,aes(x=mean, y=sd))+geom_point()


Core_Func_Pelagi=read.csv("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/October_2024_Code_Submission/Func_distribution_SAR11_Oct2024.csv")
names(Core_Func_Pelagi)
Core_Func_Pelagi=Core_Func_Pelagi%>%dplyr::rename("num_genomes_Func_Cluster_has_hits"="summed")
core_function_ranks=left_join(function_all_test,Core_Func_Pelagi, by="KOfam") 
names(core_function_ranks)

core_function_rank2 <-core_function_ranks%>%filter(num_genomes_Func_Cluster_has_hits>91)
core_function_ranks_ordered <-core_function_rank2%>%dplyr::arrange(mean)%>%dplyr::mutate(mean_rank = 1:length(mean))
###Table2
table2=core_function_ranks_ordered%>%dplyr::slice(1:25)
write.csv(table2, "table2_SAR11_October2024.csv")

```



