---
title: "pnps_SAR11_October2024"
author: "Sarah Tucker"
date: "2024-11-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```





QC samples

```{r}
iu-gen-configs sample_additional.txt -o QC

clusterize "iu-filter-quality-minoche QC/KDe743ST.ini" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

clusterize "iu-filter-quality-minoche QC/KDe747HP.ini" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

clusterize "iu-filter-quality-minoche QC/KMy813ST.ini" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

clusterize "iu-filter-quality-minoche QC/KMy817HP.ini" --job-name my_job --num-nodes 1 --num-tasks-per-node 10
```

Build contig db

```{r}
clusterize -n 120 "anvi-gen-contigs-database -f /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556.fa -o /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-CONTIGS.db -n "HIMB1556"" --job-name my_job

clusterize -n 20 "anvi-run-hmms -c /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-CONTIGS.db"

clusterize -n 120 "anvi-run-ncbi-cogs -c /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-CONTIGS.db"

clusterize -n 120 "anvi-run-kegg-kofams -c /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-CONTIGS.db"
```

Map reads and get SNVs
#Note repeat for all genomes, using the samples_QCed_Offshore.txt for offshore genera and the samples_QCed_Coastal.txt for coastal genera 
```{bash}

clusterize "bowtie2-build HIMB1412.fa HIMB1412" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1412\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1412-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1412-MERGED -c HIMB1412-CONTIGS.db" -n 120 -j merge




for split_name in `sqlite3 HIMB1412-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1412-GENOME-COLLECTION.txt


anvi-import-collection HIMB1412-GENOME-COLLECTION.txt -c HIMB1412-CONTIGS.db -p HIMB1412-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1412-CONTIGS.db -p HIMB1412-MERGED/PROFILE.db -C Default -b HIMB1412

clusterize "anvi-gen-variability-profile -c HIMB1412-CONTIGS.db -p HIMB1412-MERGED/PROFILE.db -C Default -b HIMB1412 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1412_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

clusterize "anvi-get-pn-ps-ratio -V HIMB1412_SCVs_KByT.txt -c HIMB1412-CONTIGS.db -m 20 -o PNPSratio_HIMB1412_KByT" --job-name pnps --num-nodes 1 --num-tasks-per-node 120

anvi-export-functions -c HIMB1412-CONTIGS.db -o HIMB1412-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1412_PNPS/HIMB1412-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1412_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1412_PNPS/HIMB1412_SCVs_KByTST01.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1412_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1412_PNPS/PNPSratio_HIMB1412_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1412_PNPS/

clusterize "anvi-summarize -c HIMB1412-CONTIGS.db -p HIMB1412-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1412-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1412_PNPS/HIMB1412-KByT-SUMMARY/bin_by_bin/HIMB1412/HIMB1412-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1412_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1412_PNPS/HIMB1412-KByT-SUMMARY/bin_by_bin/HIMB1412/HIMB1412-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1412_PNPS/

```


```{r}
clusterize "bowtie2-build HIMB1420.fa HIMB1420" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1420\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1420-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



anvi-merge */PROFILE.db -o HIMB1420-MERGED -c HIMB1420-CONTIGS.db




for split_name in `sqlite3 HIMB1420-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1420-GENOME-COLLECTION.txt


anvi-import-collection HIMB1420-GENOME-COLLECTION.txt -c HIMB1420-CONTIGS.db -p HIMB1420-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1420-CONTIGS.db -p HIMB1420-MERGED/PROFILE.db -C Default -b HIMB1420

clusterize "anvi-gen-variability-profile -c HIMB1420-CONTIGS.db -p HIMB1420-MERGED/PROFILE.db -C Default -b HIMB1420 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1420_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

clusterize "anvi-get-pn-ps-ratio -V HIMB1420_SCVs_KByT.txt -c HIMB1420-CONTIGS.db -m 20 -o PNPSratio_HIMB1420_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

anvi-export-functions -c HIMB1420-CONTIGS.db -o HIMB1420-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1420_PNPS/HIMB1420-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1420_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1420_PNPS/HIMB1420_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1420_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1420_PNPS/PNPSratio_HIMB1420_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1420_PNPS/


clusterize "anvi-summarize -c HIMB1420-CONTIGS.db -p HIMB1420-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1420-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1420_PNPS/HIMB1420-KByT-SUMMARY/bin_by_bin/HIMB1420/HIMB1420-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1420_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1420_PNPS/HIMB1420-KByT-SUMMARY/bin_by_bin/HIMB1420/HIMB1420-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1420_PNPS/
```


```{bash}
clusterize "bowtie2-build HIMB1427.fa HIMB1427" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1427\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1427-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1427-MERGED -c HIMB1427-CONTIGS.db" -n 120 -o profile.txt -j profile




for split_name in `sqlite3 HIMB1427-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1427-GENOME-COLLECTION.txt


anvi-import-collection HIMB1427-GENOME-COLLECTION.txt -c HIMB1427-CONTIGS.db -p HIMB1427-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1427-CONTIGS.db -p HIMB1427-MERGED/PROFILE.db -C Default -b HIMB1427

clusterize "anvi-gen-variability-profile -c HIMB1427-CONTIGS.db -p HIMB1427-MERGED/PROFILE.db -C Default -b HIMB1427 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1427_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

clusterize "anvi-get-pn-ps-ratio -V HIMB1427_SCVs_KByT.txt -c HIMB1427-CONTIGS.db -m 20 -o PNPSratio_HIMB1427_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

anvi-export-functions -c HIMB1427-CONTIGS.db -o HIMB1427-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1427_PNPS/HIMB1427-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1427_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1427_PNPS/HIMB1427_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1427_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1427_PNPS/PNPSratio_HIMB1427_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1427_PNPS/



clusterize "anvi-summarize -c HIMB1427-CONTIGS.db -p HIMB1427-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1427-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1427_PNPS/HIMB1427-KByT-SUMMARY/bin_by_bin/HIMB1427/HIMB1427-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1427_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1427_PNPS/HIMB1427-KByT-SUMMARY/bin_by_bin/HIMB1427/HIMB1427-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1427_PNPS/
```


```{bash}
clusterize "bowtie2-build HIMB1520.fa HIMB1520" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1520\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1520-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1520-MERGED -c HIMB1520-CONTIGS.db" -n 40 -o profile2.txt -j profile2




for split_name in `sqlite3 HIMB1520-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1520-GENOME-COLLECTION.txt


anvi-import-collection HIMB1520-GENOME-COLLECTION.txt -c HIMB1520-CONTIGS.db -p HIMB1520-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1520-CONTIGS.db -p HIMB1520-MERGED/PROFILE.db -C Default -b HIMB1520

clusterize "anvi-gen-variability-profile -c HIMB1520-CONTIGS.db -p HIMB1520-MERGED/PROFILE.db -C Default -b HIMB1520 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1520_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

clusterize "anvi-get-pn-ps-ratio -V HIMB1520_SCVs_KByT.txt -c HIMB1520-CONTIGS.db -m 20 -o PNPSratio_HIMB1520_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

anvi-export-functions -c HIMB1520-CONTIGS.db -o HIMB1520-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1520_PNPS/HIMB1520-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1520_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1520_PNPS/HIMB1520_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1520_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1520_PNPS/PNPSratio_HIMB1520_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1520_PNPS/



clusterize "anvi-summarize -c HIMB1520-CONTIGS.db -p HIMB1520-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1520-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1520_PNPS/HIMB1520-KByT-SUMMARY/bin_by_bin/HIMB1520/HIMB1520-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1520_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1520_PNPS/HIMB1520-KByT-SUMMARY/bin_by_bin/HIMB1520/HIMB1520-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1520_PNPS/
```


```{bash}
clusterize "bowtie2-build HIMB1564.fa HIMB1564" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1564\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1564-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1564-MERGED -c HIMB1564-CONTIGS.db" -n 120 -o profile.txt -j profile




for split_name in `sqlite3 HIMB1564-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1564-GENOME-COLLECTION.txt


anvi-import-collection HIMB1564-GENOME-COLLECTION.txt -c HIMB1564-CONTIGS.db -p HIMB1564-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1564-CONTIGS.db -p HIMB1564-MERGED/PROFILE.db -C Default -b HIMB1564

clusterize "anvi-gen-variability-profile -c HIMB1564-CONTIGS.db -p HIMB1564-MERGED/PROFILE.db -C Default -b HIMB1564 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1564_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

clusterize "anvi-get-pn-ps-ratio -V HIMB1564_SCVs_KByT.txt -c HIMB1564-CONTIGS.db -m 20 -o PNPSratio_HIMB1564_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 120

anvi-export-functions -c HIMB1564-CONTIGS.db -o HIMB1564-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1564_PNPS/HIMB1564-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1564_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1564_PNPS/HIMB1564_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1564_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1564_PNPS/PNPSratio_HIMB1564_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1564_PNPS/


clusterize "anvi-summarize -c HIMB1564-CONTIGS.db -p HIMB1564-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1564-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1564_PNPS/HIMB1564-KByT-SUMMARY/bin_by_bin/HIMB1564/HIMB1564-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1564_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1564_PNPS/HIMB1564-KByT-SUMMARY/bin_by_bin/HIMB1564/HIMB1564-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1564_PNPS/
```



```{bash}
clusterize "bowtie2-build HIMB4.fa HIMB4" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB4\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB4-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB4-MERGED -c HIMB4-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 HIMB4-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB4-GENOME-COLLECTION.txt


anvi-import-collection HIMB4-GENOME-COLLECTION.txt -c HIMB4-CONTIGS.db -p HIMB4-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB4-CONTIGS.db -p HIMB4-MERGED/PROFILE.db -C Default -b HIMB4

clusterize "anvi-gen-variability-profile -c HIMB4-CONTIGS.db -p HIMB4-MERGED/PROFILE.db -C Default -b HIMB4 --engine CDN --include-site-pnps --kiefl-mode -o HIMB4_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB4_SCVs_KByT.txt -c HIMB4-CONTIGS.db -m 20 -o PNPSratio_HIMB4_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB4-CONTIGS.db -o HIMB4-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB4_PNPS/HIMB4-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB4_PNPS/HIMB4_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB4_PNPS/PNPSratio_HIMB4_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/

clusterize "anvi-summarize -c HIMB4-CONTIGS.db -p HIMB4-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB4-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB4_PNPS/HIMB4-KByT-SUMMARY/bin_by_bin/HIMB4/HIMB4-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB4_PNPS/HIMB4-KByT-SUMMARY/bin_by_bin/HIMB4/HIMB4-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB4_PNPS/
```


```{bash}
clusterize "bowtie2-build HIMB5.fa HIMB5" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB5\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB5-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB5-MERGED -c HIMB5-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 HIMB5-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB5-GENOME-COLLECTION.txt


anvi-import-collection HIMB5-GENOME-COLLECTION.txt -c HIMB5-CONTIGS.db -p HIMB5-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB5-CONTIGS.db -p HIMB5-MERGED/PROFILE.db -C Default -b HIMB5

clusterize "anvi-gen-variability-profile -c HIMB5-CONTIGS.db -p HIMB5-MERGED/PROFILE.db -C Default -b HIMB5 --engine CDN --include-site-pnps --kiefl-mode -o HIMB5_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB5_SCVs_KByT.txt -c HIMB5-CONTIGS.db -m 20 -o PNPSratio_HIMB5_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB5-CONTIGS.db -o HIMB5-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB5_PNPS/HIMB5-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB5_PNPS/HIMB5_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB5_PNPS/PNPSratio_HIMB5_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/

clusterize "anvi-summarize -c HIMB5-CONTIGS.db -p HIMB5-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB5-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB5_PNPS/HIMB5-KByT-SUMMARY/bin_by_bin/HIMB5/HIMB5-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB5_PNPS/HIMB5-KByT-SUMMARY/bin_by_bin/HIMB5/HIMB5-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB5_PNPS/
```



```{r}
clusterize "bowtie2-build HIMB83.fa HIMB83" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB83\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB83-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB83-MERGED -c HIMB083-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 HIMB83-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB83-GENOME-COLLECTION.txt


anvi-import-collection HIMB083-GENOME-COLLECTION.txt -c HIMB083-CONTIGS.db -p HIMB83-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB083-CONTIGS.db -p HIMB83-MERGED/PROFILE.db -C Default -b HIMB083

clusterize "anvi-gen-variability-profile -c HIMB083-CONTIGS.db -p HIMB83-MERGED/PROFILE.db -C Default -b HIMB083 --engine CDN --include-site-pnps --kiefl-mode -o HIMB83_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB83_SCVs_KByT.txt -c HIMB083-CONTIGS.db -m 20 -o PNPSratio_HIMB83_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB083-CONTIGS.db -o HIMB83-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB83_PNPS/HIMB83-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB83_PNPS/HIMB83_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB83_PNPS/PNPSratio_HIMB83_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/

clusterize "anvi-summarize -c HIMB083-CONTIGS.db -p HIMB83-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB83-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB83_PNPS/HIMB83-KByT-SUMMARY/bin_by_bin/HIMB083/HIMB083-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB83_PNPS/HIMB83-KByT-SUMMARY/bin_by_bin/HIMB083/HIMB083-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB83_PNPS/
```






```{r}
clusterize "bowtie2-build HIMB1556.fa HIMB1556" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/samples_QCed_Coastal2.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 105 \
            -x HIMB1556\
            -1  /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 105 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/samples_QCed_Coastal2.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1556-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 104 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 104 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1556-MERGED -c HIMB1556-CONTIGS.db" -n 104 -o profile.txt -j profile




for split_name in `sqlite3 HIMB1556-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1556-GENOME-COLLECTION.txt


anvi-import-collection HIMB1556-GENOME-COLLECTION.txt -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default -b HIMB1556

clusterize "anvi-gen-variability-profile -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default -b HIMB1556 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1556_SCVs_AdditionalST01.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 10

anvi-get-pn-ps-ratio -V HIMB1556_SCVs_AdditionalST01.txt -c HIMB1556-CONTIGS.db -m 20 -o PNPSratio_HIMB1556_Additional_ST01

anvi-export-functions -c HIMB1556-CONTIGS.db -o HIMB1556-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556_SCVs_AdditionalST01.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/PNPSratio_HIMB1556_Additional_ST01/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

clusterize "anvi-summarize -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1556-Additional-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-Additional-SUMMARY/bin_by_bin/HIMB1556/HIMB1556-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-Additional-SUMMARY/bin_by_bin/HIMB1556/HIMB1556-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

```


```{bash}
clusterize "bowtie2-build RS40.fa RS40" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x RS40\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c RS40-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o RS40-MERGED -c RS40-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 RS40-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > RS40-GENOME-COLLECTION.txt


anvi-import-collection RS40-GENOME-COLLECTION.txt -c RS40-CONTIGS.db -p RS40-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c RS40-CONTIGS.db -p RS40-MERGED/PROFILE.db -C Default -b RS40

clusterize "anvi-gen-variability-profile -c RS40-CONTIGS.db -p RS40-MERGED/PROFILE.db -C Default -b RS40 --engine CDN --include-site-pnps --kiefl-mode -o RS40_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V RS40_SCVs_KByT.txt -c RS40-CONTIGS.db -m 20 -o PNPSratio_RS40_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c RS40-CONTIGS.db -o RS40-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS40_PNPS/RS40-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS40_PNPS/RS40_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS40_PNPS/PNPSratio_RS40_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/

clusterize "anvi-summarize -c RS40-CONTIGS.db -p RS40-MERGED/PROFILE.db -C Default --init-gene-coverages -o RS40-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS40_PNPS/RS40-KByT-SUMMARY/bin_by_bin/RS40/RS40-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS40_PNPS/RS40-KByT-SUMMARY/bin_by_bin/RS40/RS40-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS40_PNPS/
```


```{bash}
clusterize "bowtie2-build RS39.fa RS39" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x RS39\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c RS39-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o RS39-MERGED -c RS39-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 RS39-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > RS39-GENOME-COLLECTION.txt


anvi-import-collection RS39-GENOME-COLLECTION.txt -c RS39-CONTIGS.db -p RS39-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c RS39-CONTIGS.db -p RS39-MERGED/PROFILE.db -C Default -b RS39

clusterize "anvi-gen-variability-profile -c RS39-CONTIGS.db -p RS39-MERGED/PROFILE.db -C Default -b RS39 --engine CDN --include-site-pnps --kiefl-mode -o RS39_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V RS39_SCVs_KByT.txt -c RS39-CONTIGS.db -m 20 -o PNPSratio_RS39_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c RS39-CONTIGS.db -o RS39-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS39_PNPS/RS39-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS39_PNPS/RS39_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS39_PNPS/PNPSratio_RS39_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/

clusterize "anvi-summarize -c RS39-CONTIGS.db -p RS39-MERGED/PROFILE.db -C Default --init-gene-coverages -o RS39-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS39_PNPS/RS39-KByT-SUMMARY/bin_by_bin/RS39/RS39-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/RS39_PNPS/RS39-KByT-SUMMARY/bin_by_bin/RS39/RS39-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/RS39_PNPS/
```

```{r}
clusterize "bowtie2-build HIMB1437.fa HIMB1437" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1437\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1437-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1437-MERGED -c HIMB1437-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 HIMB1437-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1437-GENOME-COLLECTION.txt


anvi-import-collection HIMB1437-GENOME-COLLECTION.txt -c HIMB1437-CONTIGS.db -p HIMB1437-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1437-CONTIGS.db -p HIMB1437-MERGED/PROFILE.db -C Default -b HIMB1437

clusterize "anvi-gen-variability-profile -c HIMB1437-CONTIGS.db -p HIMB1437-MERGED/PROFILE.db -C Default -b HIMB1437 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1437_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB1437_SCVs_KByT.txt -c HIMB1437-CONTIGS.db -m 20 -o PNPSratio_HIMB1437_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB1437-CONTIGS.db -o HIMB1437-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1437_PNPS/HIMB1437-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1437_PNPS/HIMB1437_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1437_PNPS/PNPSratio_HIMB1437_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/

clusterize "anvi-summarize -c HIMB1437-CONTIGS.db -p HIMB1437-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1437-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1437_PNPS/HIMB1437-KByT-SUMMARY/bin_by_bin/HIMB1437/HIMB1437-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1437_PNPS/HIMB1437-KByT-SUMMARY/bin_by_bin/HIMB1437/HIMB1437-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1437_PNPS/
```



```{r}
clusterize "bowtie2-build HIMB1483.fa HIMB1483" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 35 \
            -x HIMB1483\
            -1  /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /gss/work/nand5072/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 35 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /gss/work/nand5072/PROJECTS/SAR11_PNPS/samples_QCed_Coastal.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1483-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 40 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 40 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1483-MERGED -c HIMB1483-CONTIGS.db" -n 100 -o profile.txt -j profile




for split_name in `sqlite3 HIMB1483-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1483-GENOME-COLLECTION.txt


anvi-import-collection HIMB1483-GENOME-COLLECTION.txt -c HIMB1483-CONTIGS.db -p HIMB1483-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1483-CONTIGS.db -p HIMB1483-MERGED/PROFILE.db -C Default -b HIMB1483

clusterize "anvi-gen-variability-profile -c HIMB1483-CONTIGS.db -p HIMB1483-MERGED/PROFILE.db -C Default -b HIMB1483 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1483_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB1483_SCVs_KByT.txt -c HIMB1483-CONTIGS.db -m 20 -o PNPSratio_HIMB1483_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB1483-CONTIGS.db -o HIMB1483-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1483_PNPS/HIMB1483-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1483_PNPS/HIMB1483_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1483_PNPS/PNPSratio_HIMB1483_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/

clusterize "anvi-summarize -c HIMB1483-CONTIGS.db -p HIMB1483-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1483-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1483_PNPS/HIMB1483-KByT-SUMMARY/bin_by_bin/HIMB1483/HIMB1483-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1483_PNPS/HIMB1483-KByT-SUMMARY/bin_by_bin/HIMB1483/HIMB1483-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1483_PNPS/
```


```{r}
clusterize "bowtie2-build HIMB1556.fa HIMB1556" --job-name my_job --num-nodes 1 --num-tasks-per-node 10

for sample in `awk '{print $1}' /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/samples_QCed_Coastal2.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 105 \
            -x HIMB1556\
            -1  /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R1.fastq \
            -2 /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/01_QC/$sample-QUALITY_PASSED_R2.fastq \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done

chmod u+x bowtie.sh


clusterize "./bowtie.sh" -n 105 -o bowtie.txt -j bowtie


for sample in `awk '{print $1}' /fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/samples_QCed_Coastal2.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c HIMB1556-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 104 \
                 -o $sample
done
chmod u+x profile.sh
ls -l profile.sh

clusterize "./profile.sh" -n 104 -o profile.txt -j profile



clusterize "anvi-merge */PROFILE.db -o HIMB1556-MERGED -c HIMB1556-CONTIGS.db" -n 104 -o profile.txt -j profile


for split_name in `sqlite3 HIMB1556-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="-"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > HIMB1556-GENOME-COLLECTION.txt


anvi-import-collection HIMB1556-GENOME-COLLECTION.txt -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Genomes



anvi-script-add-default-collection -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default -b HIMB1556

clusterize "anvi-gen-variability-profile -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default -b HIMB1556 --engine CDN --include-site-pnps --kiefl-mode -o HIMB1556_SCVs_KByT.txt " --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

clusterize "anvi-get-pn-ps-ratio -V HIMB1556_SCVs_KByT.txt -c HIMB1556-CONTIGS.db -m 20 -o PNPSratio_HIMB1556_KByT" --job-name bowtie --num-nodes 1 --num-tasks-per-node 100

anvi-export-functions -c HIMB1556-CONTIGS.db -o HIMB1556-functions.txt

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-functions.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556_SCVs_KByT.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/PNPSratio_HIMB1556_KByT/pNpS.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

clusterize "anvi-summarize -c HIMB1556-CONTIGS.db -p HIMB1556-MERGED/PROFILE.db -C Default --init-gene-coverages -o HIMB1556-KByT-SUMMARY" --job-name bowtie --num-nodes 1 --num-tasks-per-node 10


scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-KByT-SUMMARY/bin_by_bin/HIMB1556/HIMB1556-gene_calls.txt /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/

scp rosa:/fs/dss/groups/agecodatasci/PROJECTS/SAR11_PNPS/HIMB1556_PNPS/HIMB1556-KByT-SUMMARY/bin_by_bin/HIMB1556/HIMB1556-gene_non_outlier_coverages.txt  /Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_2024/PN_PS/HIMB1556_PNPS/
```


